diff a/3dge/PKGBUILD b/3dge/PKGBUILD
--- a/3dge/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/3dge/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,67 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=3dge
+pkgver=2.1.0pre.r260.gb981893
+pkgrel=1
+pkgdesc='Advanced OpenGL DOOM engine source port.'
+arch=(x86_64)
+url=http://3dfxdev.net/edgewiki
+license=(BSD-1-Clause GPL-3.0-only LGPL-3.0-only)
+depends=(bash hicolor-icon-theme perl-rename physfs sdl2 xorg-fonts-misc)
+makedepends=(cmake desktop-file-utils fltk ghc-libs
+             gendesk glew inkscape libjpeg-turbo
+             libogg mesa png++ sdl2_net)
+optdepends=('soundfont-fluid: FluidSynth soundfonts'
+            'freepats-general-midi: midi soundfonts')
+options=(!buildflags)
+source=("$pkgname"::git+https://github.com/nakane1chome/EDGE.git#branch=dual_joystick
+        https://www.rarlab.com/rar/unrarsrc-7.2.6.tar.gz
+        "$pkgname".sh)
+provides=("$pkgname")
+replaces=("$pkgname")
+sha256sums=('SKIP'
+            'd1afa67ef4121ebc5986815699e05db0ce8648499e5dca854f282a4c3f72c003'
+            'a3cd5305e3e21b92365e33b917364e9873807b9a14bde472e932a53a50c07362')
+
+pkgver() {
+  cd "$pkgname"
+  git describe --long --abbrev=7 --tags | sed 's/^g//;s/_/./;s/\([^-]*-g\)/r\1/;s/-/./g'
+}
+
+prepare() {
+  sed -e '/CXXFLAGS=/d' -e '/LDFLAGS=/d' -i unrar/makefile
+  make -C unrar clean
+  gendesk -f -n --name='3DGE' --comment="${pkgdesc/.}" --custom='Keywords=game;engine;' --exec="/usr/share/$pkgname/$pkgname.sh" --icon="$pkgname" --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname="$pkgname"
+  [[ -f "$pkgname.svg" ]] || inkscape "$pkgname"/data/iconfile.gif --export-plain-svg --export-filename="$pkgname".svg
+  [[ ! -d "build" ]] || rm -rf build
+  cd "$pkgname"
+  perl -0777 -pi -e 's/(int W_GetNumForName2.*(?s).+?int i\;(?-s).*)/\1\n        if (strcmp(name, "PAL") == 0)\n        {\n            i = W_CheckNumForName2("PLAYPAL");\n            if (i != -1) return i;\n        }/gi' src/w_wad.cc
+  perl -pi -e 's/CMAKE_HOME_DIRECTORY/CMAKE_CURRENT_SOURCE_DIR/gi' deps/libgme/CMakeLists.txt
+}
+
+build() {
+  LDFLAGS+=' -pthread' make -C unrar lib -j$(nproc)
+  cmake -S "$pkgname" -B build \
+    -DCMAKE_POLICY_VERSION_MINIMUM=3.5 \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_INSTALL_PREFIX=/usr \
+    -DCMAKE_PREFIX_PATH="$srcdir/unrar" \
+    -DINSTALL_PK3_PATH=share/$pkgname \
+    -DCMAKE_CXX_FLAGS="$CXXFLAGS -include cstdint" \
+    -Wno-dev
+  cmake --build build -j$(nproc)
+}
+
+package() {
+  install -dm 755 "$pkgdir"/usr/{bin,share/{"$pkgname"/soundfont,applications,icons/hicolor/scalable/apps}}
+#  DESTDIR="$pkgdir" cmake --build build --target install
+  install -m 755 build/EDGE "$pkgdir"/usr/share/"$pkgname/$pkgname"
+  install -m 644 build/edge.epk "$pkgdir"/usr/share/"$pkgname"/
+  install -m 644 build/soundfont/default.sf2 "$pkgdir"/usr/share/"$pkgname"/soundfont/
+  cp -dr --no-preserve=ownership build/base "$pkgdir"/usr/share/"$pkgname"/
+  install -m 755 "$pkgname".sh "$pkgdir"/usr/share/"$pkgname"/
+  install -m 644 "$pkgname".desktop "$pkgdir"/usr/share/applications/
+  install -m 644 "$pkgname".svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+#  ln -rsf "$pkgdir"/usr/share/"$pkgname/EDGE" "$pkgdir"/usr/bin/
+  ln -rsf "$pkgdir"/usr/share/"$pkgname/$pkgname" "$pkgdir"/usr/bin/
+}
diff a/3dge/3dge.sh b/3dge/3dge.sh
--- a/3dge/3dge.sh	1969-12-31 17:00:00.000000000 -0700
+++ b/3dge/3dge.sh	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,63 @@
+#!/bin/bash
+gmdir="$HOME/games/doom"
+gmcfg="$HOME/.edge"
+gmtgt="doom.wad"
+gmprm="-file nerve.wad"
+glnch="3dge"
+
+gconf() {
+mkdir -p "$gmcfg"
+printf 'gamepath=%s\ngamemod=%s\ngameparm=%s\n' "$gmdir" "$gmtgt" "$gmprm" > "$gmcfg/basedir"
+}
+
+gstart () {
+while IFS= read -r -d '' file; do
+  [[ -h "$gmbdr/$file" ]] || ln -sf "/usr/share/3dge/$file" "$gmbdr/$file"
+done < <(find /usr/share/3dge -maxdepth 1 -type f,d -regextype posix-extended ! -iregex '.*[^/]*3dge[^/]*' -printf '%P\0')
+echo "cd $gmbdr && DOOMWADDIR=$gmbdr $glnch -config $gmbdr/EDGE.ini $gmgnm $gmpar" | xargs -i -r sh -c "{}"
+}
+
+gtest () {
+gmbse="" gmbdr="" gmgme="" gmgnm="" gmgpm="" gmpar=""
+
+if [[ -f "$gmcfg/basedir" ]]; then
+  while IFS='=' read -r key val || [ -n "$key" ]; do
+    case "$key" in
+      gamepath) gmbse="gamepath=$val"; gmbdr="$val" ;;
+      gamemod)  gmgme="gamemod=$val";  gmgnm="$val" ;;
+      gameparm) gmgpm="gameparm=$val"; gmpar="$val" ;;
+    esac
+  done < "$gmcfg/basedir"
+  gmgnm="${gmgnm,,}"
+fi
+}
+
+[[ -f "$gmcfg/basedir" ]] || gconf
+gtest
+if [[ -z "$gmbdr" ]]; then
+  gconf
+  gtest
+fi
+if [[ -d "$gmbdr" ]]; then
+  if [[ ! -f "$gmbdr/$gmgnm" ]]; then
+    gmchk="$(find -H "$gmbdr" -mindepth 1 -maxdepth 1 -type f -regextype posix-extended -iregex "$gmbdr/$gmgnm" -printf '%p\n' | grep -Pic "$gmbdr/$gmgnm")"
+    if (( $gmchk == 1 )) ; then
+      echo -e "File system is case sensitive\nFile/folder names must be lowercase\n Rename subdirs of\n  $gmbdr\n    containing\n  $gmtgt?\nRequires perl-rename command" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      (( $? == 101 )) && find -H "$gmbdr" -depth -exec perl-rename 's|[^/]*\Z|\L$&|' {} +
+    elif (( $gmchk > 1 )) ; then
+      echo -e "File system is case sensitive\nFile/folder names must be lowercase\n Cannot rename subdirs of\n  $gmbdr\n    containing\n  $gmtgt\n\nBecause more than one exists" | xmessage -file - -buttons Ok:0 -center -timeout 30
+    elif [[ -z "$(echo "$gmgnm" | grep -Pio '\w+')" && -n "$(find -H "$gmbdr" -mindepth 1 -maxdepth 1 -type f -iname '*.wad' -printf '%f')" ]]; then
+      gmgnm=""
+      gstart
+    else
+      echo -e "Config:\n $gmcfg/basedir\n  $gmbse\n found, but not\n  $gmgme\n ($gmbdr/$gmgnm)\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      (( $? == 101 )) && gconf
+    fi
+  else
+    gmgnm="-iwad $gmgnm"
+    gstart
+  fi
+else
+  echo -e "Config:\n $gmcfg/basedir\n  $gmbse\n not found\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+  (( $? == 101 )) && gconf
+fi
