diff a/ares-emu/PKGBUILD b/ares-emu/PKGBUILD
--- a/ares-emu/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/ares-emu/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,47 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=ares-emu
+_pkgname=ares
+pkgver=20250215.r0.gc9285b528
+pkgrel=1
+pkgdesc='Multi-system emulator by Near with experimental Nintendo 64 and PlayStation support.'
+arch=(x86_64)
+url=https://ares-emu.net
+license=(ISC)
+depends=(alsa-lib cairo gcc-libs gdk-pixbuf2 glib2 glibc gtk3
+         hicolor-icon-theme libao libgl libpulse librashader
+         libretro-shaders libx11 libxrandr openal pango
+         sdl2 vulkan-driver vulkan-icd-loader zlib)
+makedepends=(cmake git mesa ninja pkgconf)
+options=(!emptydirs)
+conflicts=("$_pkgname"-emu-git)
+source=(git+https://github.com/higan-emu/"$_pkgname".git)
+sha256sums=('SKIP')
+
+pkgver() {
+  cd "$_pkgname"
+#  git describe --long --abbrev=7 --tags | sed 's/^v//;s/\([^-]*-g\)/r\1/;s/-/./g'
+  printf "%s.r%s.g%s" "$(date +%Y%m%d)" "$(git rev-list "$(git rev-list --tags --no-walk --max-count=1)"..HEAD --count)" "$(git rev-parse --short HEAD)"
+}
+
+prepare() {
+  test ! -d build || rm -rf build
+  cd "$_pkgname"
+#  perl -pi -e 's|(= \{Path::)user(Data)|\1shared\2|gi ; s|(create\(\{Path::user)Data|\1Settings|gi ; s|(return \{Path::user)Data|\1Settings|gi' desktop-ui/desktop-ui.cpp
+  perl -pi -e 's|(?<=-march=)native|x86-64|gi' cmake/linux/compilerconfig.cmake
+}
+
+build() {
+  cmake -S "$_pkgname" -B build -G Ninja \
+  -DCMAKE_INSTALL_PREFIX=/usr \
+  -DCMAKE_BUILD_TYPE=None \
+  -DARES_BUILD_LOCAL=OFF \
+  -DARES_BUNDLE_SHADERS=OFF \
+  -DARES_SKIP_DEPS=ON \
+  -Wno-dev
+  cmake --build build -j$(nproc)
+}
+
+package() {
+  DESTDIR="$pkgdir" cmake --build build --target install
+}
