diff a/chtrfmtr/PKGBUILD b/chtrfmtr/PKGBUILD
--- a/chtrfmtr/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/chtrfmtr/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,64 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgbase=chtrfmtr
+pkgname=("$pkgbase"-dotnet "$pkgbase"-qt6 python-"$pkgbase")
+pkgver=0.1
+pkgrel=1
+pkgdesc="Multi-emulator cheat reformatter."
+url=https://github.com/noabody/chtrfmtr
+license=(Unlicense)
+arch=(x86_64)
+depends=(hicolor-icon-theme)
+makedepends=(dotnet-sdk gendesk git python-build python-flit python-installer python-wheel qt6-base)
+source=(git+https://github.com/noabody/chtrfmtr.git)
+sha256sums=('SKIP')
+
+prepare() {
+  rm -rf "$pkgbase"/{bin,dist,build-*,obj,*.desktop}
+}
+
+build() {
+  cd "$pkgbase"
+  python -m build --wheel --no-isolation
+  dotnet publish -c Release -r linux-x64 --self-contained -p:PublishReadyToRun=true\;PublishSingleFile=true\;EnableCompressionInSingleFile=true "$pkgbase".csproj -o build-net10
+  cmake -S . -B build-qt6
+  cmake --build build-qt6 --config Release
+}
+
+package_python-chtrfmtr() {
+  pkgdesc="Multi-emulator cheat reformatter. (pyside6)"
+  arch=(any)
+  depends+=(python-regex pyside6)
+  cd "$pkgbase"
+  gendesk -f -n --name='Cheat Refomatter (PySide6)' --comment="${pkgdesc/.}" --custom='Keywords=game;console;editor;' --exec="$pkgbase"-pyside6 --icon="$pkgname".svg --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname="$pkgname"
+  install -dm 755 "$pkgdir"/usr/{bin,share/{applications,icons/hicolor/scalable/apps}}
+  install -m 644 "$pkgname".desktop "$pkgdir"/usr/share/applications/
+  install -m 644 "$pkgbase".svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/"$pkgname".svg
+  python -m installer --destdir="$pkgdir" dist/*.whl
+}
+
+package_chtrfmtr-qt6() {
+  pkgdesc="Multi-emulator cheat reformatter. (Qt6)"
+  depends+=(qt6-base)
+  cd "$pkgbase"
+  gendesk -f -n --name='Cheat Refomatter (Qt)' --comment="${pkgdesc/.}" --custom='Keywords=game;console;editor;' --exec="$pkgname" --icon="$pkgname".svg --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname="$pkgname"
+  install -dm 755 "$pkgdir"/usr/{bin,share/{applications,icons/hicolor/scalable/apps}}
+  install -m 644 "$pkgname".desktop "$pkgdir"/usr/share/applications/
+  install -m 644 "$pkgbase".svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/"$pkgname".svg
+  install -m 755 build-qt6/"$pkgbase" "$pkgdir"/usr/bin/"$pkgname"
+}
+
+package_chtrfmtr-dotnet() {
+  pkgdesc="Multi-emulator cheat reformatter. (DotNet10)"
+  depends+=(fontconfig)
+  options=(!strip)
+  cd "$pkgbase"
+  gendesk -f -n --name='Cheat Refomatter (DotNet10)' --comment="${pkgdesc/.}" --custom='Keywords=game;console;editor;' --exec="$pkgname" --icon="$pkgname".svg --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname="$pkgname"
+  install -dm 755 "$pkgdir"/{opt/"$pkgname",usr/{bin,share/{applications,icons/hicolor/scalable/apps}}}
+  install -m 644 "$pkgname".desktop "$pkgdir"/usr/share/applications/
+  install -m 644 "$pkgbase".svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/"$pkgname".svg
+  install -m 755 build-net10/"$pkgbase" "$pkgdir"/opt/"$pkgname"/"$pkgname"
+  install -m 644 build-net10/*.so "$pkgdir"/opt/"$pkgname"/
+  strip --strip-unneeded "$pkgdir"/opt/"$pkgname"/*.so
+  ln -rsf "$pkgdir"/opt/"$pkgname/$pkgname" "$pkgdir"/usr/bin/
+}
