diff a/chtrfmtr/PKGBUILD b/chtrfmtr/PKGBUILD
--- a/chtrfmtr/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/chtrfmtr/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,82 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgbase=chtrfmtr
+pkgname=("$pkgbase"-dotnet "$pkgbase"-qt6 "$pkgbase"-ps1 python-"$pkgbase")
+pkgver=0.1
+pkgrel=1
+pkgdesc="Multi-emulator cheat reformatter."
+url=https://github.com/noabody/chtrfmtr
+license=(Unlicense)
+arch=(x86_64)
+depends=(hicolor-icon-theme)
+makedepends=(dotnet-sdk gendesk git python-build python-flit python-installer python-wheel qt6-base)
+source=(git+https://github.com/noabody/chtrfmtr.git
+        "$pkgbase".patch
+        "$pkgbase".sh)
+sha256sums=('SKIP'
+            'e790b736adf0e563474a3cacac5e8b6dba497d0a624bb3bccd47a6b4df774828'
+            '11d5210411ce896a743e6e1697d7fa73ce4b22908fd4bcab6e52a285ecff6439')
+
+prepare() {
+  patch -Np1 -i "$pkgbase".patch
+  rm -rf "$pkgbase"/{bin,dist,build-*,obj,*.desktop}
+}
+
+build() {
+  cd "$pkgbase"
+  python -m build --wheel --no-isolation
+  cmake -S . -B build-qt6
+  cmake --build build-qt6 --config Release
+  dotnet publish -c Release -r linux-x64 --self-contained -p:PublishReadyToRun=true\;PublishSingleFile=true\;EnableCompressionInSingleFile=true "$pkgbase".csproj -o build-net10
+}
+
+package_python-chtrfmtr() {
+  pkgdesc="Multi-emulator cheat reformatter. (pyside6)"
+  arch=(any)
+  depends+=(python-regex pyside6)
+  cd "$pkgbase"
+  gendesk -f -n --name='Cheat Refomatter (PySide6)' --comment="${pkgdesc/.}" --custom='Keywords=game;console;editor;' --exec="$pkgbase"-pyside6 --icon="$pkgname".svg --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname="$pkgname"
+  install -dm 755 "$pkgdir"/usr/{bin,share/{applications,icons/hicolor/scalable/apps}}
+  install -m 644 "$pkgname".desktop "$pkgdir"/usr/share/applications/
+  install -m 644 "$pkgbase".svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/"$pkgname".svg
+  python -m installer --destdir="$pkgdir" dist/*.whl
+}
+
+package_chtrfmtr-qt6() {
+  pkgdesc="Multi-emulator cheat reformatter. (Qt6)"
+  depends+=(qt6-base)
+  cd "$pkgbase"
+  gendesk -f -n --name='Cheat Refomatter (Qt)' --comment="${pkgdesc/.}" --custom='Keywords=game;console;editor;' --exec="$pkgname" --icon="$pkgname".svg --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname="$pkgname"
+  install -dm 755 "$pkgdir"/usr/{bin,share/{applications,icons/hicolor/scalable/apps}}
+  install -m 644 "$pkgname".desktop "$pkgdir"/usr/share/applications/
+  install -m 644 "$pkgbase".svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/"$pkgname".svg
+  install -m 755 build-qt6/"$pkgbase" "$pkgdir"/usr/bin/"$pkgname"
+}
+
+package_chtrfmtr-ps1() {
+  pkgdesc="Multi-emulator cheat reformatter. (Powershell)"
+  arch=(any)
+  depends+=(wine winetricks)
+  install -dm 755 "$pkgdir"/{opt/"$pkgname",usr/{bin,share/{applications,icons/hicolor/scalable/apps}}}
+  install -m 755 "$pkgbase".sh "$pkgdir"/usr/bin/"$pkgname"
+  cd "$pkgbase"
+  gendesk -f -n --name='Cheat Refomatter (Powershell)' --comment="${pkgdesc/.}" --custom='Keywords=game;console;editor;' --exec="$pkgname" --icon="$pkgname".svg --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname="$pkgname"
+  install -m 644 "$pkgname".desktop "$pkgdir"/usr/share/applications/
+  install -m 644 "$pkgbase".svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/"$pkgname".svg
+  install -m 644 "$pkgbase".{ps1,vbs} "$pkgdir"/opt/"$pkgname"/
+}
+
+package_chtrfmtr-dotnet() {
+  pkgdesc="Multi-emulator cheat reformatter. (DotNet10)"
+  depends+=(fontconfig)
+  options=(!strip)
+  cd "$pkgbase"
+  gendesk -f -n --name='Cheat Refomatter (DotNet10)' --comment="${pkgdesc/.}" --custom='Keywords=game;console;editor;' --exec="$pkgname" --icon="$pkgname".svg --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname="$pkgname"
+  install -dm 755 "$pkgdir"/{opt/"$pkgname",usr/{bin,share/{applications,icons/hicolor/scalable/apps}}}
+  install -m 644 "$pkgname".desktop "$pkgdir"/usr/share/applications/
+  install -m 644 "$pkgbase".svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/"$pkgname".svg
+  install -m 755 build-net10/"$pkgbase" "$pkgdir"/opt/"$pkgname"/"$pkgname"
+  install -m 644 build-net10/*.so "$pkgdir"/opt/"$pkgname"/
+  strip --strip-unneeded "$pkgdir"/opt/"$pkgname"/*.so
+  ln -rsf "$pkgdir"/opt/"$pkgname/$pkgname" "$pkgdir"/usr/bin/
+}
diff a/chtrfmtr/chtrfmtr.sh b/chtrfmtr/chtrfmtr.sh
index e69de29..e223692 100755
--- a/chtrfmtr/chtrfmtr.sh	1969-12-31 17:00:00.000000000 -0700
+++ b/chtrfmtr/chtrfmtr.sh	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,3 @@
+#!/bin/bash
+winetricks list-installed 2>/dev/null | grep -Piq 'powershell' || winetricks -q powershell
+exec wine /opt/chtrfmtr-ps1/chtrfmtr.vbs
diff a/chtrfmtr/chtrfmtr.patch b/chtrfmtr/chtrfmtr.patch
index e69de29..feb733a 100644
--- a/chtrfmtr/chtrfmtr.patch	1969-12-31 17:00:00.000000000 -0700
+++ b/chtrfmtr/chtrfmtr.patch	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,27 @@
+diff --git a/chtrfmtr/main.cpp b/chtrfmtr/main.cpp
+--- a/chtrfmtr/main.cpp
++++ b/chtrfmtr/main.cpp
+@@ -497,16 +497,16 @@ class MainWindow : public QMainWindow {
+     void importMyBoyChtEngine(const QString &path) {
+         QFile f(path); if (!f.open(QIODevice::ReadOnly)) throw std::runtime_error(f.errorString().toStdString());
+         QDomDocument doc;
+-        QString err;
+-        int line = 0;
+-        int col = 0;
++
++        QDomDocument::ParseResult parseResult =
++            doc.setContent(&f, QDomDocument::ParseOption::UseNamespaceProcessing);
+         
+-        if (!doc.setContent(&f, true, &err, &line, &col)) {
++        if (!parseResult) {
+             throw std::runtime_error(
+                 QString("Failed parsing MyBoy XML target: %1 (line %2, column %3)")
+-                    .arg(err)
+-                    .arg(line)
+-                    .arg(col)
++                    .arg(parseResult.errorMessage)
++                    .arg(parseResult.errorLine)
++                    .arg(parseResult.errorColumn)
+                     .toStdString()
+             );
+         }
