diff a/crispy-doom/PKGBUILD b/crispy-doom/PKGBUILD
--- a/crispy-doom/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/crispy-doom/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,92 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgbase=crispy-doom
+pkgname=("$pkgbase" crispy-heretic crispy-server)
+pkgver=5.10.1.r7.g40fd021c
+pkgrel=1
+arch=(x86_64)
+url=https://www.chocolate-doom.org
+license=(GPL2)
+depends=(libpng libsamplerate sdl2_net)
+makedepends=(libpng libsamplerate sdl2_mixer sdl2_net)
+source=("$pkgbase"::git+https://github.com/fabiangreffrath/"$pkgbase".git
+        "$pkgbase".sh
+        crispy-heretic.sh)
+sha256sums=('SKIP'
+            'd6fdca705200b405403a187219902c080f87fd4dd0e4e90f0b4027dc4bd25b72'
+            '33d0d695b6a0ff97eb2788008ca7a5dc6dc93c84e98cadfd665bcd30d7ecd5ba')
+
+pkgver() {
+  cd "$pkgbase"
+  git describe --long --tags | sed 's/^crispy-doom-//;s/_/./;s/\([^-]*-g\)/r\1/;s/-/./g'
+}
+
+prepare() {
+  test -f "$pkgbase.svg" || inkscape "$pkgbase"/data/doom.png --export-plain-svg --export-filename="$pkgbase".svg
+  test -f "crispy-setup.svg" || inkscape "$pkgbase"/data/setup.png --export-plain-svg --export-filename=crispy-setup.svg
+  cd "$pkgbase"
+  ./autogen.sh --prefix=/usr
+  make clean
+}
+
+build() {
+  cd "$pkgbase"
+  make -j$(nproc)
+}
+
+package_crispy-doom() {
+  pkgdesc='Enhanced Chocolate DOOM source port.'
+  depends+=(hicolor-icon-theme perl-rename sdl2_mixer xorg-fonts-misc)
+  conflicts=("$pkgbase")
+  replaces=("$pkgbase")
+  options=(!emptydirs)
+  install -dm 755 "$pkgdir"/usr/{bin,share/{"$pkgbase",applications,icons/hicolor/scalable/apps}}
+  make -C "$pkgbase" install DESTDIR="$pkgdir" execgamesdir=/usr/share/"$pkgbase"
+  install -m 755 "$pkgbase".sh "$pkgdir"/usr/share/"$pkgbase"/
+  install -m 644 "$pkgbase".svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  install -m 644 crispy-setup.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/"$pkgbase"-setup.svg
+  ln -rsf "$pkgdir"/usr/share/"$pkgbase"/crispy-{doom,doom-setup} "$pkgdir"/usr/bin/
+
+  mv "$pkgdir"/usr/share/applications/io.github.fabiangreffrath.Doom.desktop "$pkgdir"/usr/share/applications/"$pkgbase".desktop
+  mv "$pkgdir"/usr/share/applications/io.github.fabiangreffrath.Setup.desktop "$pkgdir"/usr/share/applications/"$pkgbase"-setup.desktop
+  perl -pi -e "s|(^Exec=).*|\\1/usr/share/$pkgbase/$pkgbase.sh|g" "$pkgdir"/usr/share/applications/"$pkgbase".desktop
+  perl -pi -e 's/(^(Exec|Icon)=crispy)(.)(setup)/\1\3doom\3\4/gi ; s/(^Name=crispy)(.)(setup)/\1\2Doom\2\3/gi' "$pkgdir"/usr/share/applications/"$pkgbase"-setup.desktop
+  find "$pkgdir" \( -iname '*server*' -o -iname '*heretic*' -o -iname '*hexen*' -o -iname '*strife*' \) -exec rm -rf {} +
+  rm -f "$pkgdir"/usr/share/man/man5/default.cfg.5
+  rm -rf "$pkgdir"/usr/share/icons/hicolor/128x128
+}
+package_crispy-heretic() {
+  pkgdesc='Enhanced Chocolate Heretic source port.'
+  depends+=(hicolor-icon-theme perl-rename sdl2_mixer xorg-fonts-misc)
+  conflicts=(crispy-heretic)
+  replaces=(crispy-heretic)
+  options=(!emptydirs)
+  install -dm 755 "$pkgdir"/usr/{bin,share/{crispy-heretic,applications,icons/hicolor/scalable/apps}}
+  make -C "$pkgbase" install DESTDIR="$pkgdir" execgamesdir=/usr/share/crispy-heretic
+  install -m 755 crispy-heretic.sh "$pkgdir"/usr/share/crispy-heretic/
+  install -m 644 "$pkgbase".svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/crispy-heretic.svg
+  install -m 644 crispy-setup.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/crispy-heretic-setup.svg
+  ln -rsf "$pkgdir"/usr/share/crispy-heretic/crispy-{heretic,heretic-setup} "$pkgdir"/usr/bin/
+
+  mv "$pkgdir"/usr/share/applications/io.github.fabiangreffrath.Heretic.desktop "$pkgdir"/usr/share/applications/crispy-heretic.desktop
+  mv "$pkgdir"/usr/share/applications/io.github.fabiangreffrath.Setup.desktop "$pkgdir"/usr/share/applications/crispy-heretic-setup.desktop
+  perl -pi -e "s|(^Exec=).*|\\1/usr/share/crispy-heretic/crispy-heretic.sh|g ; s/(^Icon=crispy)-doom/\1-heretic/g" "$pkgdir"/usr/share/applications/crispy-heretic.desktop
+  perl -pi -e 's/(^(Exec|Icon)=crispy)(.)(setup)/\1\3heretic\3\4/gi ; s/(^Name=crispy)(.)(setup)/\1\2Heretic\2\3/gi' "$pkgdir"/usr/share/applications/crispy-heretic-setup.desktop
+  find "$pkgdir" \( -iname '*doom*' -o -iname '*server*' -o -iname '*hexen*' -o -iname '*strife*' \) -exec rm -rf {} +
+  rm -f "$pkgdir"/usr/share/man/man5/{default.cfg.5,heretic.cfg.5}
+  rm -rf "$pkgdir"/usr/share/icons/hicolor/128x128
+}
+package_crispy-server() {
+  pkgdesc='Enhanced Chocoloate Server source port.'
+  conflicts=(crispy-server)
+  replaces=(crispy-server)
+  options=(!emptydirs)
+  install -dm 755 "$pkgdir"/usr/{bin,share/crispy-server}
+  make -C "$pkgbase" install DESTDIR="$pkgdir" execgamesdir=/usr/share/crispy-server
+  ln -rsf "$pkgdir"/usr/share/crispy-server/crispy-server "$pkgdir"/usr/bin/
+
+  mv "$pkgdir"/usr/share/applications/io.github.fabiangreffrath.Setup.desktop "$pkgdir"/usr/share/applications/"$pkgbase"-setup.desktop
+  find "$pkgdir" \( -iname '*doom*' -o -iname '*heretic*' -o -iname '*hexen*' -o -iname '*strife*' \) -exec rm -rf {} +
+  rm -f "$pkgdir"/usr/share/man/man5/default.cfg.5
+  rm -rf "$pkgdir"/usr/share/icons/hicolor/128x128
+}
diff a/crispy-doom/crispy-doom.sh b/crispy-doom/crispy-doom.sh
--- a/crispy-doom/crispy-doom.sh	1969-12-31 17:00:00.000000000 -0700
+++ b/crispy-doom/crispy-doom.sh	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,55 @@
+#!/bin/bash
+gmdir="$HOME/games/doom"
+gmcfg="$HOME/.local/share/crispy-doom"
+gmtgt="doom.wad"
+gmprm="-file nerve.wad"
+glnch="crispy-doom"
+
+gconf () {
+test -d "$gmcfg" || mkdir -p "$gmcfg"
+echo -e "gamepath= $gmdir\ngamemod= $gmtgt\ngameparm= $gmprm" | tee "$gmcfg/basedm"
+}
+
+gstart () {
+(cd "$gmbdr" && echo "$glnch $gmgnm $gmpar" | xargs -i -r sh -c "{}")
+}
+
+gtest () {
+gmbse="$(grep -Pi '^gamepath=.*' "$gmcfg/basedm" | head -1)"
+gmgme="$(grep -Pi '^gamemod=.*' "$gmcfg/basedm" | head -1)"
+gmgpm="$(grep -Pi '^gameparm=.*' "$gmcfg/basedm" | head -1)"
+gmbdr="${gmbse#gamepath= }"
+gmgnm="${gmgme#gamemod= }"
+gmpar="${gmgpm#gameparm= }"
+gmgnm="${gmgnm,,}"
+}
+
+test -f "$gmcfg/basedm" || gconf
+gtest
+if [ -z "$gmbdr" ]; then
+  gconf
+  gtest
+fi
+if [ -d "$gmbdr" ]; then
+  if [ ! -f "$gmbdr/$gmgnm" ]; then
+    gmchk="$(find -H "$gmbdr" -mindepth 1 -maxdepth 1 -type f -regextype posix-extended -iregex "$gmbdr/$gmgnm" -printf '%p\n' | grep -Pic "$gmbdr/$gmgnm")"
+    if [ $gmchk -eq 1 ]; then
+      echo -e "File system is case sensitive\nFile/folder names must be lowercase\n Rename subdirs of\n  $gmbdr\n    containing\n  $gmtgt?\nRequires perl-rename command" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      test $? -eq 101 && find -H "$gmbdr" -depth -exec perl-rename 's|([^/]*\Z)|lc($1)|e' {} +
+    elif [ $gmchk -gt 1 ]; then
+      echo -e "File system is case sensitive\nFile/folder names must be lowercase\n Cannot rename subdirs of\n  $gmbdr\n    containing\n  $gmtgt\n\nBecause more than one exists" | xmessage -file - -buttons Ok:0 -center -timeout 30
+    elif [ -z "$(echo "$gmgnm" | grep -Pio '\w+')" ] && [ -n "$(find -H "$gmbdr" -mindepth 1 -maxdepth 1 -type f -iname '*.wad' -printf '%f')" ]; then
+      gmgnm=""
+      gstart
+    else
+      echo -e "Config:\n $gmcfg/basedm\n  $gmbse\n found, but not\n  $gmgme\n ($gmbdr/$gmgnm)\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      test $? -eq 101 && gconf
+    fi
+  else
+    gmgnm="-iwad $gmgnm"
+    gstart
+  fi
+else
+  echo -e "Config:\n $gmcfg/basedm\n  $gmbse\n not found\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+  test $? -eq 101 && gconf
+fi
diff a/crispy-doom/crispy-heretic.sh b/crispy-doom/crispy-heretic.sh
--- a/crispy-doom/crispy-heretic.sh	1969-12-31 17:00:00.000000000 -0700
+++ b/crispy-doom/crispy-heretic.sh	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,55 @@
+#!/bin/bash
+gmdir="$HOME/games/heretic"
+gmcfg="$HOME/.local/share/crispy-doom"
+gmtgt="heretic.wad"
+gmprm=""
+glnch="crispy-heretic"
+
+gconf () {
+test -d "$gmcfg" || mkdir -p "$gmcfg"
+echo -e "gamepath= $gmdir\ngamemod= $gmtgt\ngameparm= $gmprm" | tee "$gmcfg/basehc"
+}
+
+gstart () {
+(cd "$gmbdr" && echo "$glnch $gmgnm $gmpar" | xargs -i -r sh -c "{}")
+}
+
+gtest () {
+gmbse="$(grep -Pi '^gamepath=.*' "$gmcfg/basehc" | head -1)"
+gmgme="$(grep -Pi '^gamemod=.*' "$gmcfg/basehc" | head -1)"
+gmgpm="$(grep -Pi '^gameparm=.*' "$gmcfg/basehc" | head -1)"
+gmbdr="${gmbse#gamepath= }"
+gmgnm="${gmgme#gamemod= }"
+gmpar="${gmgpm#gameparm= }"
+gmgnm="${gmgnm,,}"
+}
+
+test -f "$gmcfg/basehc" || gconf
+gtest
+if [ -z "$gmbdr" ]; then
+  gconf
+  gtest
+fi
+if [ -d "$gmbdr" ]; then
+  if [ ! -f "$gmbdr/$gmgnm" ]; then
+    gmchk="$(find -H "$gmbdr" -mindepth 1 -maxdepth 1 -type f -regextype posix-extended -iregex "$gmbdr/$gmgnm" -printf '%p\n' | grep -Pic "$gmbdr/$gmgnm")"
+    if [ $gmchk -eq 1 ]; then
+      echo -e "File system is case sensitive\nFile/folder names must be lowercase\n Rename subdirs of\n  $gmbdr\n    containing\n  $gmtgt?\nRequires perl-rename command" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      test $? -eq 101 && find -H "$gmbdr" -depth -exec perl-rename 's|([^/]*\Z)|lc($1)|e' {} +
+    elif [ $gmchk -gt 1 ]; then
+      echo -e "File system is case sensitive\nFile/folder names must be lowercase\n Cannot rename subdirs of\n  $gmbdr\n    containing\n  $gmtgt\n\nBecause more than one exists" | xmessage -file - -buttons Ok:0 -center -timeout 30
+    elif [ -z "$(echo "$gmgnm" | grep -Pio '\w+')" ] && [ -n "$(find -H "$gmbdr" -mindepth 1 -maxdepth 1 -type f -iname '*.wad' -printf '%f')" ]; then
+      gmgnm=""
+      gstart
+    else
+      echo -e "Config:\n $gmcfg/basehc\n  $gmbse\n found, but not\n  $gmgme\n ($gmbdr/$gmgnm)\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      test $? -eq 101 && gconf
+    fi
+  else
+    gmgnm="-iwad $gmgnm"
+    gstart
+  fi
+else
+  echo -e "Config:\n $gmcfg/basehc\n  $gmbse\n not found\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+  test $? -eq 101 && gconf
+fi
