diff a/crispy-doom/PKGBUILD b/crispy-doom/PKGBUILD
--- a/crispy-doom/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/crispy-doom/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,125 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgbase=crispy-doom
+pkgname=("$pkgbase" crispy-heretic crispy-hexen crispy-strife crispy-server)
+pkgver=7.1.r69.g39f4d93
+pkgrel=1
+arch=(x86_64)
+url=https://www.chocolate-doom.org
+license=(GPL-2.0-only)
+depends=(sdl2-compat sdl2_net)
+makedepends=(gendesk libpng libsamplerate sdl2_mixer sdl2_net)
+source=("$pkgbase"::git+https://github.com/fabiangreffrath/"$pkgbase".git
+        "$pkgbase".sh
+        crispy-heretic.sh
+        crispy-hexen.sh
+        crispy-strife.sh)
+sha256sums=('SKIP'
+            '85ff85f5ca6416976b09904868bc36b09684117e693566d35064c4f20761a579'
+            '0eb98d0bbb359392b122a571c7d545ff1a88502d6b8e9b6ee8c80be309821083'
+            '56cbba7a164bb6c5c1d0acba5b928c1c6f8d04faf4e409af49e89b8daba8c1f4'
+            '1f401c566c2f7e9df975702135cf5e7c7dc2b879b8042a40f09daf6887b7e3d9')
+
+pkgver() {
+  cd "$pkgbase"
+  git describe --long --abbrev=7 --tags | sed 's/^crispy-doom-//;s/_/./;s/\([^-]*-g\)/r\1/;s/-/./g'
+}
+
+prepare() {
+  for _icon in doom heretic hexen strife; do
+    [[ -f "crispy-$_icon.svg" ]] || inkscape "$pkgbase"/data/"$_icon".png --export-plain-svg --export-filename=crispy-"$_icon".svg
+  done
+  [[ -f "crispy-setup.svg" ]] || inkscape "$pkgbase"/data/setup.png --export-plain-svg --export-filename=crispy-setup.svg
+  [[ ! -d "build" ]] || rm -rf build
+  cd "$pkgbase"
+  ./autogen.sh --prefix=/usr
+  make clean
+}
+
+build() {
+  cmake -S "$pkgbase" -B build -G Ninja \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_INSTALL_PREFIX=/usr
+  cmake --build build -j$(nproc)
+}
+
+package_crispy-doom() {
+  pkgdesc='Enhanced Crispy DOOM source port.'
+  depends+=(fluidsynth hicolor-icon-theme libpng libsamplerate perl-rename sdl2_mixer xorg-fonts-misc)
+  conflicts=("$pkgname")
+  replaces=("$pkgname")
+  gendesk -f -n --name='Crispy Doom' --comment="${pkgdesc%.*}" --custom='Keywords=game;engine;' --exec="/usr/share/$pkgname/$pkgname.sh" --icon="$pkgname" --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname="$pkgname"
+  gendesk -f -n --name='Crispy Doom Setup' --comment="${pkgdesc%.*} Setup file" --custom='Keywords=game;engine;' --exec="$pkgname-setup" --icon="$pkgname-setup" --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname="$pkgname-setup"
+  install -dm 755 "$pkgdir"/usr/{bin,share/{"$pkgname",applications,icons/hicolor/scalable/apps}}
+  make -C "$pkgbase"/man DESTDIR="$pkgdir" man_MANS="$pkgname.6 $pkgname-setup.6 $pkgname.cfg.5" install-man
+  make -C "$pkgbase"/man/bash-completion DESTDIR="$pkgdir" bashcompletion_DATA="$pkgname" install-bashcompletionDATA
+  install -m 755 build/src/"$pkgname" "$pkgdir"/usr/share/"$pkgname"/
+  install -m 755 build/src/crispy-setup "$pkgdir"/usr/share/"$pkgname"/"$pkgname"-setup
+  install -m 755 "$pkgname".sh "$pkgdir"/usr/share/"$pkgname"/
+  install -m 644 "$pkgname"{,-setup}.desktop "$pkgdir"/usr/share/applications/
+  install -m 644 "$pkgname".svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  install -m 644 crispy-setup.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/"$pkgname"-setup.svg
+  ln -rsf "$pkgdir"/usr/share/"$pkgname/$pkgname"{,-setup} "$pkgdir"/usr/bin/
+}
+package_crispy-heretic() {
+  pkgdesc='Enhanced Crispy Heretic source port.'
+  depends+=(fluidsynth hicolor-icon-theme libpng libsamplerate perl-rename sdl2_mixer xorg-fonts-misc)
+  conflicts=("$pkgname")
+  replaces=("$pkgname")
+  gendesk -f -n --name='Crispy Heretic' --comment="${pkgdesc%.*}" --custom='Keywords=game;engine;' --exec="/usr/share/$pkgname/$pkgname.sh" --icon="$pkgname" --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname="$pkgname"
+  gendesk -f -n --name='Crispy Heretic Setup' --comment="${pkgdesc%.*} Setup file" --custom='Keywords=game;engine;' --exec="$pkgname-setup" --icon="$pkgname-setup" --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname="$pkgname-setup"
+  install -dm 755 "$pkgdir"/usr/{bin,share/{"$pkgname",applications,icons/hicolor/scalable/apps}}
+  make -C "$pkgbase"/man DESTDIR="$pkgdir" man_MANS="$pkgname.6 $pkgname-setup.6 $pkgname.cfg.5" install-man
+  make -C "$pkgbase"/man/bash-completion DESTDIR="$pkgdir" bashcompletion_DATA="$pkgname" install-bashcompletionDATA
+  install -m 755 build/src/"$pkgname" "$pkgdir"/usr/share/"$pkgname"/
+  install -m 755 build/src/crispy-setup "$pkgdir"/usr/share/"$pkgname"/"$pkgname"-setup
+  install -m 755 "$pkgname".sh "$pkgdir"/usr/share/"$pkgname"/
+  install -m 644 "$pkgname"{,-setup}.desktop "$pkgdir"/usr/share/applications/
+  install -m 644 "$pkgname".svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/"$pkgname".svg
+  install -m 644 crispy-setup.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/"$pkgname"-setup.svg
+  ln -rsf "$pkgdir"/usr/share/"$pkgname/$pkgname"{,-setup} "$pkgdir"/usr/bin/
+}
+package_crispy-hexen() {
+  pkgdesc='Enhanced Crispy Hexen source port.'
+  depends+=(fluidsynth hicolor-icon-theme libpng libsamplerate perl-rename sdl2_mixer xorg-fonts-misc)
+  conflicts=("$pkgname")
+  replaces=("$pkgname")
+  gendesk -f -n --name='Crispy Hexen' --comment="${pkgdesc%.*}" --custom='Keywords=game;engine;' --exec="/usr/share/$pkgname/$pkgname.sh" --icon="$pkgname" --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname="$pkgname"
+  gendesk -f -n --name='Crispy Hexen Setup' --comment="${pkgdesc%.*} Setup file" --custom='Keywords=game;engine;' --exec="$pkgname-setup" --icon="$pkgname-setup" --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname="$pkgname-setup"
+  install -dm 755 "$pkgdir"/usr/{bin,share/{"$pkgname",applications,icons/hicolor/scalable/apps}}
+  make -C "$pkgbase"/man DESTDIR="$pkgdir" man_MANS="$pkgname.6 $pkgname-setup.6 $pkgname.cfg.5" install-man
+  make -C "$pkgbase"/man/bash-completion DESTDIR="$pkgdir" bashcompletion_DATA="$pkgname" install-bashcompletionDATA
+  install -m 755 build/src/"$pkgname" "$pkgdir"/usr/share/"$pkgname"/
+  install -m 755 build/src/crispy-setup "$pkgdir"/usr/share/"$pkgname"/"$pkgname"-setup
+  install -m 755 "$pkgname".sh "$pkgdir"/usr/share/"$pkgname"/
+  install -m 644 "$pkgname"{,-setup}.desktop "$pkgdir"/usr/share/applications/
+  install -m 644 "$pkgname".svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/"$pkgname".svg
+  install -m 644 crispy-setup.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/"$pkgname"-setup.svg
+  ln -rsf "$pkgdir"/usr/share/"$pkgname/$pkgname"{,-setup} "$pkgdir"/usr/bin/
+}
+package_crispy-strife() {
+  pkgdesc='Enhanced Crispy Strife source port.'
+  depends+=(fluidsynth hicolor-icon-theme libpng libsamplerate perl-rename sdl2_mixer xorg-fonts-misc)
+  conflicts=("$pkgname")
+  replaces=("$pkgname")
+  gendesk -f -n --name='Crispy Strife' --comment="${pkgdesc%.*}" --custom='Keywords=game;engine;' --exec="/usr/share/$pkgname/$pkgname.sh" --icon="$pkgname" --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname="$pkgname"
+  gendesk -f -n --name='Crispy Strife Setup' --comment="${pkgdesc%.*} Setup file" --custom='Keywords=game;engine;' --exec="$pkgname-setup" --icon="$pkgname-setup" --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname="$pkgname-setup"
+  install -dm 755 "$pkgdir"/usr/{bin,share/{"$pkgname",applications,icons/hicolor/scalable/apps}}
+  make -C "$pkgbase"/man DESTDIR="$pkgdir" man_MANS="$pkgname.6 $pkgname-setup.6 $pkgname.cfg.5" install-man
+  make -C "$pkgbase"/man/bash-completion DESTDIR="$pkgdir" bashcompletion_DATA="$pkgname" install-bashcompletionDATA
+  install -m 755 build/src/"$pkgname" "$pkgdir"/usr/share/"$pkgname"/
+  install -m 755 build/src/crispy-setup "$pkgdir"/usr/share/"$pkgname"/"$pkgname"-setup
+  install -m 755 "$pkgname".sh "$pkgdir"/usr/share/"$pkgname"/
+  install -m 644 "$pkgname"{,-setup}.desktop "$pkgdir"/usr/share/applications/
+  install -m 644 "$pkgname".svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/"$pkgname".svg
+  install -m 644 crispy-setup.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/"$pkgname"-setup.svg
+  ln -rsf "$pkgdir"/usr/share/"$pkgname/$pkgname"{,-setup} "$pkgdir"/usr/bin/
+}
+package_crispy-server() {
+  pkgdesc='Enhanced Crispy Server source port.'
+  conflicts=("$pkgname")
+  replaces=("$pkgname")
+  install -dm 755 "$pkgdir"/usr/{bin,share/"$pkgname"}
+  install -m 755 build/src/"$pkgname" "$pkgdir"/usr/share/"$pkgname"/
+  ln -rsf "$pkgdir"/usr/share/"$pkgname/$pkgname" "$pkgdir"/usr/bin/
+}
diff a/crispy-doom/crispy-doom.sh b/crispy-doom/crispy-doom.sh
--- a/crispy-doom/crispy-doom.sh	1969-12-31 17:00:00.000000000 -0700
+++ b/crispy-doom/crispy-doom.sh	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,60 @@
+#!/bin/bash
+gmdir="$HOME/games/doom"
+gmcfg="$HOME/.local/share/crispy-doom"
+gmtgt="doom.wad"
+gmprm="-file nerve.wad"
+glnch="crispy-doom"
+
+gconf() {
+mkdir -p "$gmcfg"
+printf 'gamepath=%s\ngamemod=%s\ngameparm=%s\n' "$gmdir" "$gmtgt" "$gmprm" > "$gmcfg/basedm"
+}
+
+gstart () {
+(cd "$gmbdr" && echo "$glnch $gmgnm $gmpar" | xargs -i -r sh -c "{}")
+}
+
+gtest () {
+gmbse="" gmbdr="" gmgme="" gmgnm="" gmgpm="" gmpar=""
+
+if [ -f "$gmcfg/basedm" ]; then
+  while IFS='=' read -r key val || [ -n "$key" ]; do
+    case "$key" in
+      gamepath) gmbse="gamepath=$val"; gmbdr="$val" ;;
+      gamemod)  gmgme="gamemod=$val";  gmgnm="$val" ;;
+      gameparm) gmgpm="gameparm=$val"; gmpar="$val" ;;
+    esac
+  done < "$gmcfg/basedm"
+  gmgnm="${gmgnm,,}"
+fi
+}
+
+[[ -f "$gmcfg/basedm" ]] || gconf
+gtest
+if [[ -z "$gmbdr" ]]; then
+  gconf
+  gtest
+fi
+if [[ -d "$gmbdr" ]]; then
+  if [[ ! -f "$gmbdr/$gmgnm" ]]; then
+    gmchk="$(find -H "$gmbdr" -mindepth 1 -maxdepth 1 -type f -regextype posix-extended -iregex "$gmbdr/$gmgnm" -printf '%p\n' | grep -Pic "$gmbdr/$gmgnm")"
+    if (( $gmchk == 1 )) ; then
+      echo -e "File system is case sensitive\nFile/folder names must be lowercase\n Rename subdirs of\n  $gmbdr\n    containing\n  $gmtgt?\nRequires perl-rename command" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      (( $? == 101 )) && find -H "$gmbdr" -depth -exec perl-rename 's|[^/]*\Z|\L$&|' {} +
+    elif (( $gmchk > 1 )) ; then
+      echo -e "File system is case sensitive\nFile/folder names must be lowercase\n Cannot rename subdirs of\n  $gmbdr\n    containing\n  $gmtgt\n\nBecause more than one exists" | xmessage -file - -buttons Ok:0 -center -timeout 30
+    elif [[ -z "$(echo "$gmgnm" | grep -Pio '\w+')" && -n "$(find -H "$gmbdr" -mindepth 1 -maxdepth 1 -type f -iname '*.wad' -printf '%f')" ]]; then
+      gmgnm=""
+      gstart
+    else
+      echo -e "Config:\n $gmcfg/basedm\n  $gmbse\n found, but not\n  $gmgme\n ($gmbdr/$gmgnm)\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      (( $? == 101 )) && gconf
+    fi
+  else
+    gmgnm="-iwad $gmgnm"
+    gstart
+  fi
+else
+  echo -e "Config:\n $gmcfg/basedm\n  $gmbse\n not found\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+  (( $? == 101 )) && gconf
+fi
diff a/crispy-doom/crispy-heretic.sh b/crispy-doom/crispy-heretic.sh
--- a/crispy-doom/crispy-heretic.sh	1969-12-31 17:00:00.000000000 -0700
+++ b/crispy-doom/crispy-heretic.sh	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,60 @@
+#!/bin/bash
+gmdir="$HOME/games/doom"
+gmcfg="$HOME/.local/share/crispy-doom"
+gmtgt="heretic.wad"
+gmprm=""
+glnch="crispy-heretic"
+
+gconf() {
+mkdir -p "$gmcfg"
+printf 'gamepath=%s\ngamemod=%s\ngameparm=%s\n' "$gmdir" "$gmtgt" "$gmprm" > "$gmcfg/basehc"
+}
+
+gstart () {
+(cd "$gmbdr" && echo "$glnch $gmgnm $gmpar" | xargs -i -r sh -c "{}")
+}
+
+gtest () {
+gmbse="" gmbdr="" gmgme="" gmgnm="" gmgpm="" gmpar=""
+
+if [ -f "$gmcfg/basehc" ]; then
+  while IFS='=' read -r key val || [ -n "$key" ]; do
+    case "$key" in
+      gamepath) gmbse="gamepath=$val"; gmbdr="$val" ;;
+      gamemod)  gmgme="gamemod=$val";  gmgnm="$val" ;;
+      gameparm) gmgpm="gameparm=$val"; gmpar="$val" ;;
+    esac
+  done < "$gmcfg/basehc"
+  gmgnm="${gmgnm,,}"
+fi
+}
+
+[[ -f "$gmcfg/basehc" ]] || gconf
+gtest
+if [[ -z "$gmbdr" ]]; then
+  gconf
+  gtest
+fi
+if [[ -d "$gmbdr" ]]; then
+  if [[ ! -f "$gmbdr/$gmgnm" ]]; then
+    gmchk="$(find -H "$gmbdr" -mindepth 1 -maxdepth 1 -type f -regextype posix-extended -iregex "$gmbdr/$gmgnm" -printf '%p\n' | grep -Pic "$gmbdr/$gmgnm")"
+    if (( $gmchk == 1 )) ; then
+      echo -e "File system is case sensitive\nFile/folder names must be lowercase\n Rename subdirs of\n  $gmbdr\n    containing\n  $gmtgt?\nRequires perl-rename command" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      (( $? == 101 )) && find -H "$gmbdr" -depth -exec perl-rename 's|[^/]*\Z|\L$&|' {} +
+    elif (( $gmchk > 1 )) ; then
+      echo -e "File system is case sensitive\nFile/folder names must be lowercase\n Cannot rename subdirs of\n  $gmbdr\n    containing\n  $gmtgt\n\nBecause more than one exists" | xmessage -file - -buttons Ok:0 -center -timeout 30
+    elif [[ -z "$(echo "$gmgnm" | grep -Pio '\w+')" && -n "$(find -H "$gmbdr" -mindepth 1 -maxdepth 1 -type f -iname '*.wad' -printf '%f')" ]]; then
+      gmgnm=""
+      gstart
+    else
+      echo -e "Config:\n $gmcfg/basehc\n  $gmbse\n found, but not\n  $gmgme\n ($gmbdr/$gmgnm)\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      (( $? == 101 )) && gconf
+    fi
+  else
+    gmgnm="-iwad $gmgnm"
+    gstart
+  fi
+else
+  echo -e "Config:\n $gmcfg/basehc\n  $gmbse\n not found\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+  (( $? == 101 )) && gconf
+fi
diff a/crispy-doom/crispy-hexen.sh b/crispy-doom/crispy-hexen.sh
--- a/crispy-doom/crispy-hexen.sh	1969-12-31 17:00:00.000000000 -0700
+++ b/crispy-doom/crispy-hexen.sh	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,60 @@
+#!/bin/bash
+gmdir="$HOME/games/doom"
+gmcfg="$HOME/.local/share/crispy-doom"
+gmtgt="hexen.wad"
+gmprm=""
+glnch="crispy-hexen"
+
+gconf() {
+mkdir -p "$gmcfg"
+printf 'gamepath=%s\ngamemod=%s\ngameparm=%s\n' "$gmdir" "$gmtgt" "$gmprm" > "$gmcfg/basehx"
+}
+
+gstart () {
+(cd "$gmbdr" && echo "$glnch $gmgnm $gmpar" | xargs -i -r sh -c "{}")
+}
+
+gtest () {
+gmbse="" gmbdr="" gmgme="" gmgnm="" gmgpm="" gmpar=""
+
+if [ -f "$gmcfg/basehx" ]; then
+  while IFS='=' read -r key val || [ -n "$key" ]; do
+    case "$key" in
+      gamepath) gmbse="gamepath=$val"; gmbdr="$val" ;;
+      gamemod)  gmgme="gamemod=$val";  gmgnm="$val" ;;
+      gameparm) gmgpm="gameparm=$val"; gmpar="$val" ;;
+    esac
+  done < "$gmcfg/basehx"
+  gmgnm="${gmgnm,,}"
+fi
+}
+
+[[ -f "$gmcfg/basehx" ]] || gconf
+gtest
+if [[ -z "$gmbdr" ]]; then
+  gconf
+  gtest
+fi
+if [[ -d "$gmbdr" ]]; then
+  if [[ ! -f "$gmbdr/$gmgnm" ]]; then
+    gmchk="$(find -H "$gmbdr" -mindepth 1 -maxdepth 1 -type f -regextype posix-extended -iregex "$gmbdr/$gmgnm" -printf '%p\n' | grep -Pic "$gmbdr/$gmgnm")"
+    if (( $gmchk == 1 )) ; then
+      echo -e "File system is case sensitive\nFile/folder names must be lowercase\n Rename subdirs of\n  $gmbdr\n    containing\n  $gmtgt?\nRequires perl-rename command" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      (( $? == 101 )) && find -H "$gmbdr" -depth -exec perl-rename 's|[^/]*\Z|\L$&|' {} +
+    elif (( $gmchk > 1 )) ; then
+      echo -e "File system is case sensitive\nFile/folder names must be lowercase\n Cannot rename subdirs of\n  $gmbdr\n    containing\n  $gmtgt\n\nBecause more than one exists" | xmessage -file - -buttons Ok:0 -center -timeout 30
+    elif [[ -z "$(echo "$gmgnm" | grep -Pio '\w+')" && -n "$(find -H "$gmbdr" -mindepth 1 -maxdepth 1 -type f -iname '*.wad' -printf '%f')" ]]; then
+      gmgnm=""
+      gstart
+    else
+      echo -e "Config:\n $gmcfg/basehx\n  $gmbse\n found, but not\n  $gmgme\n ($gmbdr/$gmgnm)\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      (( $? == 101 )) && gconf
+    fi
+  else
+    gmgnm="-iwad $gmgnm"
+    gstart
+  fi
+else
+  echo -e "Config:\n $gmcfg/basehx\n  $gmbse\n not found\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+  (( $? == 101 )) && gconf
+fi
diff a/crispy-doom/crispy-strife.sh b/crispy-doom/crispy-strife.sh
--- a/crispy-doom/crispy-strife.sh	1969-12-31 17:00:00.000000000 -0700
+++ b/crispy-doom/crispy-strife.sh	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,60 @@
+#!/bin/bash
+gmdir="$HOME/games/doom"
+gmcfg="$HOME/.local/share/crispy-doom"
+gmtgt="strife1.wad"
+gmprm=""
+glnch="crispy-strife"
+
+gconf() {
+mkdir -p "$gmcfg"
+printf 'gamepath=%s\ngamemod=%s\ngameparm=%s\n' "$gmdir" "$gmtgt" "$gmprm" > "$gmcfg/basest"
+}
+
+gstart () {
+(cd "$gmbdr" && echo "$glnch $gmgnm $gmpar" | xargs -i -r sh -c "{}")
+}
+
+gtest () {
+gmbse="" gmbdr="" gmgme="" gmgnm="" gmgpm="" gmpar=""
+
+if [ -f "$gmcfg/basest" ]; then
+  while IFS='=' read -r key val || [ -n "$key" ]; do
+    case "$key" in
+      gamepath) gmbse="gamepath=$val"; gmbdr="$val" ;;
+      gamemod)  gmgme="gamemod=$val";  gmgnm="$val" ;;
+      gameparm) gmgpm="gameparm=$val"; gmpar="$val" ;;
+    esac
+  done < "$gmcfg/basest"
+  gmgnm="${gmgnm,,}"
+fi
+}
+
+[[ -f "$gmcfg/basest" ]] || gconf
+gtest
+if [[ -z "$gmbdr" ]]; then
+  gconf
+  gtest
+fi
+if [[ -d "$gmbdr" ]]; then
+  if [[ ! -f "$gmbdr/$gmgnm" ]]; then
+    gmchk="$(find -H "$gmbdr" -mindepth 1 -maxdepth 1 -type f -regextype posix-extended -iregex "$gmbdr/$gmgnm" -printf '%p\n' | grep -Pic "$gmbdr/$gmgnm")"
+    if (( $gmchk == 1 )) ; then
+      echo -e "File system is case sensitive\nFile/folder names must be lowercase\n Rename subdirs of\n  $gmbdr\n    containing\n  $gmtgt?\nRequires perl-rename command" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      (( $? == 101 )) && find -H "$gmbdr" -depth -exec perl-rename 's|[^/]*\Z|\L$&|' {} +
+    elif (( $gmchk > 1 )) ; then
+      echo -e "File system is case sensitive\nFile/folder names must be lowercase\n Cannot rename subdirs of\n  $gmbdr\n    containing\n  $gmtgt\n\nBecause more than one exists" | xmessage -file - -buttons Ok:0 -center -timeout 30
+    elif [[ -z "$(echo "$gmgnm" | grep -Pio '\w+')" && -n "$(find -H "$gmbdr" -mindepth 1 -maxdepth 1 -type f -iname '*.wad' -printf '%f')" ]]; then
+      gmgnm=""
+      gstart
+    else
+      echo -e "Config:\n $gmcfg/basest\n  $gmbse\n found, but not\n  $gmgme\n ($gmbdr/$gmgnm)\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      (( $? == 101 )) && gconf
+    fi
+  else
+    gmgnm="-iwad $gmgnm"
+    gstart
+  fi
+else
+  echo -e "Config:\n $gmcfg/basest\n  $gmbse\n not found\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+  (( $? == 101 )) && gconf
+fi
