diff a/dbgl/PKGBUILD b/dbgl/PKGBUILD
--- a/dbgl/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/dbgl/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,41 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=dbgl
+pkgver=0.92
+pkgrel=2
+pkgdesc="DOSBox Game Launcher."
+arch=(any)
+url=http://members.quicknet.nl/blankendaalr/dbgl
+license=(GPL2)
+depends=(gtk2 hicolor-icon-theme 'java-runtime>=8')
+makedepends=(ant gendesk)
+optdepends=('dosbox: necessary to run DOS games')
+provides=("$pkgname")
+conflicts=("$pkgname")
+replaces=("$pkgname")
+source=(https://www.squadrablu.nl/dbgl/download/src092.zip
+        "$pkgname")
+sha256sums=('199ff3dfa4a85e177bebbe3abd40179ba06b3bf16d93597d99f4f5e2fd167dce'
+            '5e9ccd3d39525d19091466cc4e5861c466582d053a8a248892fdfe0b07530da3')
+
+prepare() {
+  gendesk -f -n --name='DBGL' --comment="${pkgdesc/.}" --custom='Keywords=game;console;' --exec="$pkgname" --icon="$pkgname" --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname="$pkgname"
+  ant clean
+}
+
+build() {
+  test -f "$pkgname.svg" || inkscape src/resources/img/ico/256.png --export-plain-svg --export-filename="$pkgname".svg
+  ant distlinux
+}
+
+package() {
+  install -dm 755 "$pkgdir"/usr/{bin,lib/"$pkgname"/{lib,templates,xsl},share/{applications,icons/hicolor/scalable/apps}}
+  install -m 755 "$pkgname" "$pkgdir"/usr/bin/
+  install -m 644 "$pkgname".desktop "$pkgdir"/usr/share/applications/
+  install -m 644 "$pkgname".svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  install -m 644 dist/"$pkgname".jar "$pkgdir"/usr/lib/"$pkgname"/
+  install -m 644 lib/*.jar "$pkgdir"/usr/lib/"$pkgname"/lib/
+  install -m 644 src/dist/shared/templates/*.xml "$pkgdir"/usr/lib/"$pkgname"/templates/
+  install -m 644 src/dist/shared/xsl/*.xsl "$pkgdir"/usr/lib/"$pkgname"/xsl/
+  install -m 644 src/dist/shared/COPYING "$pkgdir"/usr/lib/"$pkgname"/
+}
diff a/dbgl/dbgl b/dbgl/dbgl
index e69de29..e223692 100755
--- a/dbgl/dbgl	1969-12-31 17:00:00.000000000 -0700
+++ b/dbgl/dbgl	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,12 @@
+#!/bin/bash
+dboxx="$(which dosbox-x)"
+dbcfg="$HOME/.dbgl"
+for value in captures db dosbox dosroot export profiles; do
+  test -d "$dbcfg/$value" || mkdir -p "$dbcfg/$value"
+done
+if [ -f "$dboxx" ]; then
+  test -h "$dbcfg/dosbox/$dboxx" || ln -sf "$dboxx" "$dbcfg/dosbox/dosbox"
+fi
+jre8="$(find /usr/lib/jvm -maxdepth 1 -type d -ipath '*/java*8*')"
+#(cd /usr/lib/dbgl && env JAVA_HOME="$jre8" PATH="$jre8/bin:$PATH" SWT_GTK3=0 java -Djava.library.path=lib -Ddbgl.data.userhome=true -jar dbgl.jar)
+(cd /usr/lib/dbgl && java -Djava.library.path=lib -Ddbgl.data.userhome=true -jar dbgl.jar)
