diff a/dfunity/PKGBUILD b/dfunity/PKGBUILD
--- a/dfunity/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/dfunity/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,81 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+# Linux player build fix from Unity editor (black screen):
+#  Main Gui: Window > Asset Management > Addressables > Groups
+#  Expand sub window: Build > New Build > Default Build Script
+
+# On the 'Activate your license' page, if the 'Personal' option is
+#  not available, right-click on the page and choose 'Inspect' to
+#  open the browser's developer tools. Find the line of HTML that
+#  looks like this: <div class="option option-personal clear"
+#  style="display: none;">. Delete display: none; from the style
+#  attribute to make the 'Personal' option visible."
+
+pkgname=dfunity
+_uver=2019.4.41f2
+pkgver=1.1.1.r116.g81e89e90c
+pkgrel=1
+pkgdesc="Open source recreation of Daggerfall in the Unity engine."
+arch=(x86_64)
+url=https://www.dfworkshop.net
+license=(MIT)
+depends=(gtk2 hicolor-icon-theme rsync xorg-fonts-misc xorg-xmessage)
+makedepends=(gendesk git mesa gdk-pixbuf2 glib2 openssl-1.1 zlib)
+provides=($pkgname)
+options=(!buildflags)
+source=(
+  "$pkgname"::git+https://github.com/Interkarma/daggerfall-unity.git
+  https://download.unity3d.com/download_unity/6b23d448b533/UnitySetup-"$_uver"
+  https://raw.githubusercontent.com/noabody/unibuild/master/data/"$pkgname".svg
+  "$pkgname".sh
+  )
+sha256sums=('SKIP'
+            '23f51649f79f9ed3bf2cc0b6076d8613cdffa00f83509923bc66ccf123434576'
+            '0d383f70c1704e50f8d968a0fe460e259bb76c989841db475974a4a2c67cd2b4'
+            'cdaf5fb757f5faa8cc74241a6228dca57857215b447e1f9322e800e801f3396a')
+
+pkgver() {
+  cd "$pkgname"
+  local _ver="$(grep -Pio "rfallu.*\b.*\d+\.[[:alnum:]\.]+" Assets/Scripts/VersionInfo.cs | grep -Pio '\d+\.[[:alnum:]\.]+')"
+  printf "%s.r%s.g%s" "$_ver" "$(git rev-list "$(git rev-list --tags --no-walk --max-count=1)"..HEAD --count)" "$(git rev-parse --short HEAD)"
+}
+
+prepare() {
+  gendesk -f -n --name='Daggerfall Unity' --comment="${pkgdesc/.}" --custom='Keywords=game;console;' --exec="/usr/share/$pkgname/$pkgname.sh" --icon="$pkgname" --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname="$pkgname"
+  _unity="$(find -H "$HOME"/{Unity-"$_uver",Unity/Hub/Editor/"$_uver"}/Editor -maxdepth 1 -type f -iname 'Unity' 2>/dev/null | head -1)"
+  _ulf="$(find -H "$HOME"/.local/share/unity3d -maxdepth 2 -type f -iname '*.ulf' 2>/dev/null)"
+  if [ -z "$_unity" ]; then
+    echo -e "Unity $_uver Does not exist.\n\n  Please install using defaults.\n\nActivate as a non-commercial user.\n         If appropriate." | xmessage -file - -buttons Continue:0 -center #-timeout 30
+    chmod 755 UnitySetup"$_uver"
+    ./UnitySetup-"$_uver"
+    _unity="$HOME/Unity-$_uver/Editor/Unity"
+  fi
+  if [ -z "$_ulf" ]; then
+    export DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=1 
+    TERM=xterm "$_unity" -batchmode -logFile - -createManualActivationFile -quit || true
+    xdg-open https://license.unity3d.com/manual #upload file, f12 personal for non-commercial
+    echo -e "Unity $_uver Has no license.\n\n  Activation as non-commercial user.\n\n Upload alf from $startdir.\n\nInspect hidden personal radio button.\n\n Save to $startdir." | xmessage -file - -buttons Continue:0 -center #-timeout 30
+    TERM=xterm "$_unity" -batchmode -logFile - -manualLicenseFile "$startdir"/Unity*.ulf -quit || true
+  fi
+  cd "$pkgname"
+  perl -pi -e 's|(?=  m_BuildTargetVRSettings)|  - m_BuildTarget: LinuxStandaloneSupport\n    m_APIs: 1500000011000000\n    m_Automatic: 0\n|gi' ProjectSettings/ProjectSettings.asset
+  test ! -d "build" || rm -rf build
+  mkdir -p build
+}
+
+build() {
+  cd "$pkgname"
+  export DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=1
+  TERM=xterm "$_unity" -batchmode -logFile - -projectPath . -buildTarget linux64 -buildLinux64Player "build/DaggerfallUnity.x86_64" -quit
+}
+
+package() {
+  install -dm 755 "$pkgdir"/usr/{bin,share/{"$pkgname"/DaggerfallUnity_Data,applications,icons/hicolor/scalable/apps}}
+  install -m 755 "$pkgname"/build/DaggerfallUnity.x86_64 "$pkgdir"/usr/share/"$pkgname"/
+  install -m 755 "$pkgname"/build/UnityPlayer.so "$pkgdir"/usr/share/"$pkgname"/
+  install -m 755 "$pkgname".sh "$pkgdir"/usr/share/"$pkgname"/
+  cp -dr --no-preserve=ownership "$pkgname"/build/DaggerfallUnity_Data "$pkgdir"/usr/share/"$pkgname"/
+#  install -m 644 ""$pkgname"/build/Daggerfall Unity Manual.pdf" "$pkgdir"/usr/share/"$pkgname"/
+  install -m 644 "$pkgname".desktop "$pkgdir"/usr/share/applications/
+  install -m 644 "$pkgname".svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  ln -rsf "$pkgdir"/usr/share/"$pkgname"/DaggerfallUnity.x86_64 "$pkgdir"/usr/bin/
+}
diff a/dfunity.sh b/dfunity/dfunity.sh
--- a/dfunity/dfunity.sh	1969-12-31 17:00:00.000000000 -0700
+++ b/dfunity/dfunity.sh	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,14 @@
+#!/bin/bash
+gmcfg="$HOME/.config/unity3d/Daggerfall Workshop/Daggerfall Unity"
+glnch="DaggerfallUnity.x86_64"
+i_syms=()
+
+test -d "$gmcfg/DaggerfallUnity_Data" || mkdir -p "$gmcfg/DaggerfallUnity_Data"
+readarray -t i_syms < <(find /usr/share/dfunity -mindepth 1 -maxdepth 2 ! \( -iname '*dfunity.sh' -o -iname "*$glnch" -o -ipath '*StreamingAssets*' -o -ipath '*DaggerfallUnity_Data' \) -printf '%P\n')
+for i in ${!i_syms[@]}; do
+  test -h "$gmcfg/${i_syms[$i]}" || ln -sf "/usr/share/dfunity/${i_syms[$i]}" "$gmcfg/${i_syms[$i]}"
+done
+unset i_syms
+rsync -c "/usr/share/dfunity/$glnch" "$gmcfg"
+rsync -rc --ignore-existing /usr/share/dfunity/DaggerfallUnity_Data/StreamingAssets "$gmcfg/DaggerfallUnity_Data"
+(cd "$gmcfg" && "$gmcfg/$glnch")
