diff a/doomsday/PKGBUILD b/doomsday/PKGBUILD
--- a/doomsday/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/doomsday/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,45 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=doomsday
+pkgver=3.0.0.r1126.gdb4bb26129
+pkgrel=1
+pkgdesc='DOOM variant enhanced graphics port.'
+arch=(x86_64)
+url=http://dengine.net
+license=(BSD)
+depends=(fluidsynth glu hicolor-icon-theme libxrandr minizip
+         openal pcre qt5-x11extras sdl2_mixer soundfont-fluid)
+makedepends=(assimp cmake glu imagemagick python)
+source=("$pkgname"::git+https://github.com/skyjake/Doomsday-Engine.git
+        https://github.com/noabody/unibuild/raw/master/data/denglibs.tar.xz)
+provides=("$pkgname")
+replaces=("$pkgname" "$pkgname"-bin)
+conflicts=("$pkgname" "$pkgname"-bin)
+sha256sums=('SKIP'
+            '521a0bd2279527bf2a595928648b18045943e4fe4065597bebb1441156aa78d0')
+
+pkgver() {
+  cd "$pkgname"
+  local _ver="$(grep -Pio "ver.*_(ma|mi|pa).*\b.*\d+" "$pkgname"/cmake/Version.cmake | grep -Pio '\d+$' | perl -pe 's|\s|.|' | perl -pe 's|(.*)\.$|\1|')"
+  printf "%s.r%s.g%s" "$_ver" "$(git rev-list "$(git rev-list --tags --no-walk --max-count=1)"..HEAD --count)" "$(git rev-parse --short HEAD)"
+}
+
+prepare() {
+  test ! -d "build" || rm -rf build
+  cd "$pkgname"
+  perl -pi -e "s/(.*DTFDN_ENABLE_TLSREQUEST=NO.*)$/\1\n         '-DTFDN_ENABLE_SSE41=NO',/gi" "$pkgname"/build/scripts/build_deps.py
+  perl -pi -e 's|(.*lib_Foundation.so.*)$|\1\n                   \${the_Foundation_DIR}/../../lib_Foundation.so.1.4\n                   \${the_Foundation_DIR}/../../lib_Foundation.so.1.4.0|gi' "$pkgname"/libs/core/CMakeLists.txt
+  python "$pkgname"/build/scripts/build_deps.py build
+}
+
+build() {
+  cmake -S "$pkgname/$pkgname" -B build -G Ninja \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_INSTALL_PREFIX=/usr \
+    -DFMOD_DIR="$srcdir"/fmod
+  cmake --build build -j$(nproc)
+}
+
+package() {
+  DESTDIR="$pkgdir" cmake --build build --target install
+}
