diff a/dosbox-x/PKGBUILD b/dosbox-x/PKGBUILD
--- a/dosbox-x/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/dosbox-x/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,41 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=dosbox-x
+pkgver=0.83.18.r205.g1e3d36adf
+pkgrel=1
+pkgdesc='Fork of DOSBox, an x86 emulator with DOS.'
+arch=(x86_64)
+url=http://dosbox-x.com
+license=(GPL2)
+depends=(ffmpeg4.4 fluidsynth hicolor-icon-theme
+         libpng libslirp libxkbfile libxrandr sdl2_net tslib)
+makedepends=(git glu physfs)
+source=("$pkgname"::git+https://github.com/joncampbell123/"$pkgname".git)
+provides=("$pkgname")
+replaces=("$pkgname"-git "$pkgname"-fast-bios-git "$pkgname"-sdl2)
+conflicts=("$pkgname"-git "$pkgname"-fast-bios-git "$pkgname"-sdl2)
+sha256sums=('SKIP')
+
+pkgver() {
+  cd "$pkgname"
+  git describe --long --tags | sed 's/.*-v//;s/_/./;s/\([^-]*-g\)/r\1/;s/-/./g'
+}
+
+prepare() {
+  gendesk -f -n --name='DOSBox-X' --comment="${pkgdesc/.}" --custom='Keywords=game;emulator;' --exec="$pkgname" --icon="$pkgname" --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname="$pkgname"
+}
+
+build() {
+  cd "$pkgname"
+  sed -i -e 's/-j3/-j$(nproc)/g' build
+  sed -i 's|"$LIBS -lavcodec -lavformat -lavutil -lswscale "`pkg-config libavcodec --libs`|`pkg-config libavcodec libavformat libavutil libswscale libswresample --libs`"$LIBS"|' configure.ac
+  PKG_CONFIG_PATH='/usr/lib/ffmpeg4.4/pkgconfig' ./build
+}
+
+package() {
+  make -C "$pkgname" DESTDIR="$pkgdir" install
+  rm -f "$pkgdir"/usr/share/applications/*.desktop
+  install -dm 755 "$pkgdir"/usr/share/{applications,icons/hicolor/scalable/apps}
+  install -m 644 "$srcdir/$pkgname"/contrib/icons/"$pkgname".svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  install -m 644 "$srcdir/$pkgname".desktop "$pkgdir"/usr/share/applications/
+}
