diff a/dosbox-x/PKGBUILD b/dosbox-x/PKGBUILD
--- a/dosbox-x/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/dosbox-x/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,42 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=dosbox-x
+pkgver=0.84.3.r236.g982c44176
+pkgrel=1
+pkgdesc='Fork of DOSBox, an x86 emulator with DOS.'
+arch=(x86_64)
+url=http://dosbox-x.com
+license=(GPL2)
+depends=(ffmpeg fluidsynth hicolor-icon-theme libpcap
+         libpng libslirp libxkbfile libxrandr mesa sdl2_net)
+makedepends=(git glu)
+source=("$pkgname"::git+https://github.com/joncampbell123/"$pkgname".git)
+provides=("$pkgname")
+replaces=("$pkgname"-git "$pkgname"-fast-bios-git "$pkgname"-sdl2)
+conflicts=("$pkgname"-git "$pkgname"-fast-bios-git "$pkgname"-sdl2)
+sha256sums=('SKIP')
+
+pkgver() {
+  cd "$pkgname"
+  git describe --long --tags | sed 's/.*-v//;s/_/./;s/\([^-]*-g\)/r\1/;s/-/./g'
+}
+
+prepare() {
+  gendesk -f -n --name='DOSBox-X' --comment="${pkgdesc/.}" --custom='Keywords=game;emulator;' --exec="$pkgname" --icon="$pkgname" --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname="$pkgname"
+  cd "$pkgname"
+  test ! -f "Makefile" || make distclean
+  ./autogen.sh
+  ./configure --enable-avcodec --enable-debug --enable-sdl2 --prefix=/usr
+}
+
+build() {
+  make -C "$pkgname" -j$(nproc)
+}
+
+package() {
+  make -C "$pkgname" DESTDIR="$pkgdir" install
+  rm -f "$pkgdir"/usr/share/applications/*.desktop
+  install -dm 755 "$pkgdir"/usr/share/{applications,icons/hicolor/scalable/apps}
+  install -m 644 "$srcdir/$pkgname"/contrib/icons/"$pkgname".svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  install -m 644 "$srcdir/$pkgname".desktop "$pkgdir"/usr/share/applications/
+}
