diff a/dxx-rebirth/PKGBUILD b/dxx-rebirth/PKGBUILD
--- a/dxx-rebirth/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/dxx-rebirth/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,92 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgbase=dxx-rebirth
+pkgname=(d1x-rebirth d2x-rebirth)
+pkgver=0.60.0.r1310.g1ebb73843
+pkgrel=1
+epoch=1
+arch=(x86_64)
+url=https://www.dxx-rebirth.com
+license=(GPL-3.0-only)
+depends=(glu hicolor-icon-theme perl-rename
+         physfs sdl2_image sdl2_mixer xorg-fonts-misc)
+makedepends=(gendesk git inkscape mesa
+             scons sdl_image unzip)
+source=("$pkgbase"::git+https://github.com/"$pkgbase/$pkgbase".git
+        https://www."$pkgbase".com/d1x-rebirth_addons.zip
+        https://www."$pkgbase".com/d2x-rebirth_addons.zip
+        d1x-rebirth.sh
+        d2x-rebirth.sh)
+noextract=(d1x-rebirth_addons.zip
+           d2x-rebirth_addons.zip)
+sha256sums=('SKIP'
+            '68eab65a757c060e4588acd7d1cb72ef96505f5b0c86d27c546a4a7bb6113b97'
+            '6b70773148f6241901e1b7831e1044cfb734c53095570c9c96bb994362a6fa6b'
+            'b27cd69e0b56615ee986f74c961bb4cdf5cf601a43d7a3d1440f66003edea66a'
+            'dcfecaf94a9cd3d12a11b3d59b307b8766bd40f0f25b854e1701804e058ef029')
+_music=sc55 # (sc55/opl3)
+
+pkgver() {
+  cd "$pkgbase"
+  git describe --long --abbrev=7 --tags | sed 's/-beta*[^-]//;s/_/./;s/\([^-]*-g\)/r\1/;s/-/./g'
+}
+
+prepare() {
+  unzip -ju d1x-rebirth_addons.zip -x "*/editor.dxa"
+  unzip -ju d2x-rebirth_addons.zip -x "*/editor.dxa"
+  gendesk -f -n --name='D1X-Rebirth' --comment='A source port of the Descent engine' --custom='Keywords=game;engine;' --exec='/usr/share/d1x-rebirth/d1x-rebirth.sh' --icon=d1x-rebirth --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname=d1x-rebirth
+  gendesk -f -n --name='D2X-Rebirth' --comment='A source port of the Descent 2 engine' --custom='Keywords=game;engine;' --exec='/usr/share/d2x-rebirth/d2x-rebirth.sh' --icon=d2x-rebirth --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname=d2x-rebirth
+  [[ -f "d1x-rebirth.svg" ]] || inkscape "$pkgbase"/d1x-rebirth/d1x-rebirth.xpm --export-plain-svg --export-filename=d1x-rebirth.svg
+  [[ -f "d2x-rebirth.svg" ]] || inkscape "$pkgbase"/d2x-rebirth/d2x-rebirth.xpm --export-plain-svg --export-filename=d2x-rebirth.svg
+  cd "$pkgbase"
+  scons -c
+  rm -rf .scon*
+  [[ ! -d "build" ]] || rm -rf build
+}
+
+build() {
+  local -a _common_opts=(
+    "$MAKEFLAGS"
+    '-Cdxx-rebirth'
+    'builddir=./build'
+    'prefix=/usr'
+    'opengl=yes'
+    'sdl2=yes'
+    'sdlmixer=yes'
+    'ipv6=yes'
+    'use_udp=yes'
+    'use_tracker=yes'
+    'screenshot=png')
+  export CXXFLAGS="${CXXFLAGS/-Wp,-D_GLIBCXX_ASSERTIONS/}"
+  scons "${_common_opts[@]}" 'd1x=1' 'd2x=0' 'sharepath=/usr/share/d1x-rebirth'
+  scons "${_common_opts[@]}" 'd1x=0' 'd2x=1' 'sharepath=/usr/share/d2x-rebirth'
+}
+
+package_d1x-rebirth() {
+  pkgdesc='A source port of the Descent engine.'
+  optdepends=('d1x-data: addon data/music')
+  conflicts=("$pkgname" "$pkgname"-git)
+  replaces=("$pkgname" "$pkgname"-git)
+  install -dm 755 "$pkgdir"/usr/{bin,share/{"$pkgname",applications,icons/hicolor/scalable/apps}}
+  install -m 755 "$pkgbase"/build/"$pkgname/$pkgname" "$pkgdir"/usr/share/"$pkgname"/
+  install -m 755 "$pkgname".sh "$pkgdir"/usr/share/"$pkgname"/
+  install -m 644 "$pkgname".desktop "$pkgdir"/usr/share/applications/
+  install -m 644 "$pkgname".svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  install -m 644 d1xr-hires.dxa "$pkgdir"/usr/share/"$pkgname"/
+  install -m 644 d1xr-"$_music"-music.dxa "$pkgdir"/usr/share/"$pkgname"/
+  ln -rsf "$pkgdir"/usr/share/"$pkgname/$pkgname" "$pkgdir"/usr/bin/
+}
+
+package_d2x-rebirth() {
+  pkgdesc='A source port of the Descent 2 engine.'
+  optdepends=('d2x-data: addon music')
+  conflicts=("$pkgname" "$pkgname"-git)
+  replaces=("$pkgname" "$pkgname"-git)
+  install -dm 755 "$pkgdir"/usr/{bin,share/{"$pkgname",applications,icons/hicolor/scalable/apps}}
+  install -m 755 "$pkgbase"/build/"$pkgname/$pkgname" "$pkgdir"/usr/share/"$pkgname"/
+  install -m 755 "$pkgname".sh "$pkgdir"/usr/share/"$pkgname"/
+  install -m 644 "$pkgname".desktop "$pkgdir"/usr/share/applications/
+  install -m 644 "$pkgname".svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  install -m 644 d2xr-"$_music"-music.dxa "$pkgdir"/usr/share/"$pkgname"/
+  ln -rsf "$pkgdir"/usr/share/"$pkgname/$pkgname" "$pkgdir"/usr/bin/
+}
diff a/dxx-rebirth/d1x-rebirth.sh b/dxx-rebirth/d1x-rebirth.sh
--- a/dxx-rebirth/d1x-rebirth.sh	1969-12-31 17:00:00.000000000 -0700
+++ b/dxx-rebirth/d1x-rebirth.sh	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,60 @@
+#!/bin/bash
+gmdir="$HOME/games/descent"
+gmcfg="$HOME/.d1x-rebirth"
+gmtgt="descent.hog"
+gmprm=""
+glnch="d1x-rebirth"
+
+gconf() {
+mkdir -p "$gmcfg"
+printf 'gamepath=%s\ngamemod=%s\ngameparm=%s\n' "$gmdir" "$gmtgt" "$gmprm" > "$gmcfg/basedir"
+}
+
+gstart () {
+while IFS= read -r -d '' file; do
+  [[ -h "$gmcfg/$file" ]] || ln -sf "/usr/share/d1x-rebirth/$file" "$gmcfg/$file"
+done < <(find /usr/share/d1x-rebirth -type f \( -iname '*.dxa' \) -printf '%P\0')
+echo "$glnch $gmbdr $gmpar" | xargs -i -r sh -c "{}"
+}
+
+gtest () {
+gmbse="" gmbdr="" gmgme="" gmgnm="" gmgpm="" gmpar=""
+
+if [[ -f "$gmcfg/basedir" ]]; then
+  while IFS='=' read -r key val || [ -n "$key" ]; do
+    case "$key" in
+      gamepath) gmbse="gamepath=$val"; gmbdr="$val" ;;
+      gamemod)  gmgme="gamemod=$val";  gmgnm="$val" ;;
+      gameparm) gmgpm="gameparm=$val"; gmpar="$val" ;;
+    esac
+  done < "$gmcfg/basedir"
+  gmgnm="${gmgnm,,}"
+fi
+}
+
+[[ -f "$gmcfg/basedir" ]] || gconf
+gtest
+if [[ -z "$gmbdr" ]]; then
+  gconf
+  gtest
+fi
+if [[ -d "$gmbdr" ]]; then
+  if [[ ! -f "$gmbdr/$gmgnm" ]]; then
+    gmchk="$(find -H "$gmbdr" -mindepth 1 -maxdepth 1 -type f -regextype posix-extended -iregex "$gmbdr/$gmgnm" -printf '%p\n' | grep -Pic "$gmbdr/$gmgnm")"
+    if (( $gmchk == 1 )) ; then
+      echo -e "File system is case sensitive\nFile/folder names must be lowercase\n Rename subdirs of\n  $gmbdr\n    containing\n  $gmtgt?\nRequires perl-rename command" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      (( $? == 101 )) && find -H "$gmbdr" -depth -exec perl-rename 's|[^/]*\Z|\L$&|' {} +
+    elif (( $gmchk > 1 )) ; then
+      echo -e "File system is case sensitive\nFile/folder names must be lowercase\n Cannot rename subdirs of\n  $gmbdr\n    containing\n  $gmtgt\n\nBecause more than one exists" | xmessage -file - -buttons Ok:0 -center -timeout 30
+    else
+      echo -e "Config:\n $gmcfg/basedir\n  $gmbse\n found, but not\n  $gmgme\n ($gmbdr/$gmgnm)\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      (( $? == 101 )) && gconf
+    fi
+  else
+    gmbdr="-hogdir $gmbdr"
+    gstart
+  fi
+else
+  echo -e "Config:\n $gmcfg/basedir\n  $gmbse\n not found\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+  (( $? == 101 )) && gconf
+fi
diff a/dxx-rebirth/d2x-rebirth.sh b/dxx-rebirth/d2x-rebirth.sh
--- a/dxx-rebirth/d2x-rebirth.sh	1969-12-31 17:00:00.000000000 -0700
+++ b/dxx-rebirth/d2x-rebirth.sh	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,60 @@
+#!/bin/bash
+gmdir="$HOME/games/descent2"
+gmcfg="$HOME/.d2x-rebirth"
+gmtgt="descent2.hog"
+gmprm=""
+glnch="d2x-rebirth"
+
+gconf() {
+mkdir -p "$gmcfg"
+printf 'gamepath=%s\ngamemod=%s\ngameparm=%s\n' "$gmdir" "$gmtgt" "$gmprm" > "$gmcfg/basedir"
+}
+
+gstart () {
+while IFS= read -r -d '' file; do
+  [[ -h "$gmcfg/$file" ]] || ln -sf "/usr/share/d2x-rebirth/$file" "$gmcfg/$file"
+done < <(find /usr/share/d2x-rebirth -type f \( -iname '*.dxa' \) -printf '%P\0')
+echo "$glnch $gmbdr $gmpar" | xargs -i -r sh -c "{}"
+}
+
+gtest () {
+gmbse="" gmbdr="" gmgme="" gmgnm="" gmgpm="" gmpar=""
+
+if [[ -f "$gmcfg/basedir" ]]; then
+  while IFS='=' read -r key val || [ -n "$key" ]; do
+    case "$key" in
+      gamepath) gmbse="gamepath=$val"; gmbdr="$val" ;;
+      gamemod)  gmgme="gamemod=$val";  gmgnm="$val" ;;
+      gameparm) gmgpm="gameparm=$val"; gmpar="$val" ;;
+    esac
+  done < "$gmcfg/basedir"
+  gmgnm="${gmgnm,,}"
+fi
+}
+
+[[ -f "$gmcfg/basedir" ]] || gconf
+gtest
+if [[ -z "$gmbdr" ]]; then
+  gconf
+  gtest
+fi
+if [[ -d "$gmbdr" ]]; then
+  if [[ ! -f "$gmbdr/$gmgnm" ]]; then
+    gmchk="$(find -H "$gmbdr" -mindepth 1 -maxdepth 1 -type f -regextype posix-extended -iregex "$gmbdr/$gmgnm" -printf '%p\n' | grep -Pic "$gmbdr/$gmgnm")"
+    if (( $gmchk == 1 )) ; then
+      echo -e "File system is case sensitive\nFile/folder names must be lowercase\n Rename subdirs of\n  $gmbdr\n    containing\n  $gmtgt?\nRequires perl-rename command" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      (( $? == 101 )) && find -H "$gmbdr" -depth -exec perl-rename 's|[^/]*\Z|\L$&|' {} +
+    elif (( $gmchk > 1 )) ; then
+      echo -e "File system is case sensitive\nFile/folder names must be lowercase\n Cannot rename subdirs of\n  $gmbdr\n    containing\n  $gmtgt\n\nBecause more than one exists" | xmessage -file - -buttons Ok:0 -center -timeout 30
+    else
+      echo -e "Config:\n $gmcfg/basedir\n  $gmbse\n found, but not\n  $gmgme\n ($gmbdr/$gmgnm)\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      (( $? == 101 )) && gconf
+    fi
+  else
+    gmbdr="-hogdir $gmbdr"
+    gstart
+  fi
+else
+  echo -e "Config:\n $gmcfg/basedir\n  $gmbse\n not found\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+  (( $? == 101 )) && gconf
+fi
