diff a/emu-ex-plus-alpha/PKGBUILD b/emu-ex-plus-alpha/PKGBUILD
--- a/emu-ex-plus-alpha/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/emu-ex-plus-alpha/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,190 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgbase=emu-ex-plus-alpha
+pkgname=(2600-emu c64-emu gba-emu gbc-emu md-emu msx-emu neo-emu nes-emu ngp-emu pce-emu snes9x-ex)
+pkgver=1.5.61.r477.gfae7143e
+pkgrel=1
+arch=(x86_64)
+url=https://www.explusalpha.com
+license=(GPL3)
+depends=(alsa-lib bluez-libs fontconfig freetype2 hicolor-icon-theme libglvnd libpulse libxi libxrandr)
+makedepends=(gendesk git inkscape)
+install="$pkgbase".install
+source=("$pkgbase"::git+https://github.com/Rakashazi/"$pkgbase".git
+        https://raw.githubusercontent.com/archlinux/svntogit-packages/packages/libarchive/trunk/0003-libarchive-Do-not-include-sys-mount.h-when-linux-fs..patch
+        https://sourceforge.net/projects/libpng/files/libpng16/1.6.37/libpng-1.6.37.tar.xz
+        "$pkgbase".install)
+sha256sums=('SKIP'
+            '99c85a9b8e6c16131cefaf1040d5fa15cd565ecf8c71c1f644c8fa47fe1306dc'
+            '505e70834d35383537b6491e7ae8641f1a4bed1876dbfe361201fc80868d88ca'
+            'c00d324c7cf7a6d73300b3da8fdf74f729078c1e8f2275c905e136e34b20cd52')
+noextract=(libpng-1.6.37.tar.xz)
+
+pkgver() {
+  cd "$pkgbase"
+  local _ver="$(grep -Pio "vers.*\b.*\d+\.[[:alnum:]\.]+" imagine/include/imagine/config/version.h | grep -Pio '\d+\.[[:alnum:]\.]+')"
+  printf "%s.r%s.g%s" "$_ver" "$(git rev-list --count HEAD)" "$(git rev-parse --short HEAD)"
+}
+
+prepare() {
+  cp 0003-libarchive-Do-not-include-sys-mount.h-when-linux-fs..patch "$pkgbase"/imagine/bundle/all/src/libarchive/libarchive-sys-mount.h-linux-fs.patch
+  cp libpng-1.6.37.tar.xz "$pkgbase"/imagine/bundle/all/src/libpng
+  perl -pi -e 's|(\ttar \-mxJf.*)|\1\n\tpatch -d \$(libarchiveSrcDir) -p1 \< libarchive-sys-mount.h-linux-fs.patch|gi' "$pkgbase"/imagine/bundle/all/src/libarchive/common.mk
+  perl -pi -e 's|1.6.13|1.6.37|gi' "$pkgbase"/imagine/bundle/all/src/libpng/common.mk
+  ln -rsf "$pkgbase"/imagine/make/shortcut/common-builds/linux-x86_64-release-static.mk "$pkgbase"/EmuFramework/
+  ln -rsf "$pkgbase"/imagine/make/shortcut/common-builds/linux-x86_64-release.mk "$pkgbase"/C64.emu/
+  for _emu in 2600.emu C64.emu GBA.emu GBC.emu MD.emu MSX.emu NEO.emu NES.emu NGP.emu PCE.emu Snes9x; do
+  _sub="${_emu//./}"
+  _sub="${_sub,,}"
+  if [ ! -f "$_sub.svg" ]; then
+    convert -thumbnail 256x256 -strip "$pkgbase/$_emu"/res/icons/icon-144.png "$_sub".png
+    inkscape "$_sub".png --export-plain-svg --export-filename="$_sub".svg
+    rm "$_sub".png
+  fi
+  done
+  for _emu in . imagine EmuFramework 2600.emu C64.emu GBA.emu GBC.emu MD.emu MSX.emu NEO.emu NES.emu NGP.emu PCE.emu Snes9x; do
+    test ! -d "$pkgbase/$_emu/build" || rm -rf "$pkgbase/$_emu"/build
+    test ! -d "$pkgbase/$_emu/target" || rm -rf "$pkgbase/$_emu"/target
+  done
+  test ! -d "$pkgbase/temp" || rm -rf "$pkgbase"/temp
+  test ! -d "$pkgbase/imagine-sdk" || rm -rf "$pkgbase"/imagine-sdk
+}
+
+build() {
+  export IMAGINE_PATH="$startdir"/src/"$pkgbase"/imagine
+  export IMAGINE_SDK_PATH="$IMAGINE_PATH"-sdk
+  export PKG_CONFIG_PATH="$IMAGINE_SDK_PATH"/lib/pkgconfig
+  cd "$pkgbase"/imagine/bundle/all
+  bash makeAll-linux-x86_64-static.sh tempDir="$startdir"/src/"$pkgbase"/temp installDir="$IMAGINE_SDK_PATH" install
+  for _emu in imagine EmuFramework 2600.emu GBA.emu GBC.emu MD.emu MSX.emu NEO.emu NES.emu NGP.emu PCE.emu Snes9x; do
+    cd "$startdir"/src/"$pkgbase/$_emu"
+    (echo "$_emu" | grep -Pioq 'imagine|EmuFramework' && make -f linux-x86_64-release-static.mk -j4 install prefix="$IMAGINE_SDK_PATH" || make -f linux-x86_64-release-static.mk -j4)
+  done
+  cd ../C64.emu
+  make -f linux-x86_64-release.mk -j4 CFLAGS_OPTIMIZE_LEVEL_RELEASE_DEFAULT=-O1
+  cd ..
+}
+
+package_2600-emu() {
+  pkgdesc='2600.emu from the emu-ex-plus-alpha suite.'
+  conflicts=(2600-emu)
+  replaces=(2600-emu)
+  gendesk -f -n --name=2600-emu --comment="${pkgdesc%.*}" --custom='Keywords=game;engine;' --exec='/usr/share/2600emu/2600emu' --icon=2600emu --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname=2600emu
+  install -dm 755 "$pkgdir"/usr/{bin,share/{2600emu,applications,icons/hicolor/scalable/apps}}
+  cp -rL --no-preserve='ownership' "$pkgbase"/2600.emu/{res/android/assets,target/linux}/* "$pkgdir"/usr/share/2600emu/
+  install -m 644 2600emu.desktop "$pkgdir"/usr/share/applications/
+  install -m 644 2600emu.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  ln -rsf "$pkgdir"/usr/share/2600emu/2600emu "$pkgdir"/usr/bin/
+}
+package_c64-emu() {
+  pkgdesc='C64.emu from the emu-ex-plus-alpha suite.'
+  conflicts=(c64-emu)
+  replaces=(c64-emu)
+  gendesk -f -n --name=C64-emu --comment="${pkgdesc%.*}" --custom='Keywords=game;engine;' --exec='/usr/share/c64emu/c64emu' --icon=c64emu --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname=c64emu
+  install -dm 755 "$pkgdir"/usr/{bin,share/{c64emu,applications,icons/hicolor/scalable/apps}}
+  cp -rL --no-preserve='ownership' "$pkgbase"/C64.emu/{res/android/assets,target/linux}/* "$pkgdir"/usr/share/c64emu/
+  install -m 644 c64emu.desktop "$pkgdir"/usr/share/applications/
+  install -m 644 c64emu.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  ln -rsf "$pkgdir"/usr/share/c64emu/c64emu "$pkgdir"/usr/bin/
+}
+package_gba-emu() {
+  pkgdesc='GBA.emu from the emu-ex-plus-alpha suite.'
+  conflicts=(gba-emu)
+  replaces=(gba-emu)
+  gendesk -f -n --name=GBA-emu --comment="${pkgdesc%.*}" --custom='Keywords=game;engine;' --exec='/usr/share/gbaemu/gbaemu' --icon=gbaemu --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname=gbaemu
+  install -dm 755 "$pkgdir"/usr/{bin,share/{gbaemu,applications,icons/hicolor/scalable/apps}}
+  cp -rL --no-preserve='ownership' "$pkgbase"/GBA.emu/{res/android/assets,target/linux}/* "$pkgdir"/usr/share/gbaemu/
+  install -m 644 gbaemu.desktop "$pkgdir"/usr/share/applications/
+  install -m 644 gbaemu.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  ln -rsf "$pkgdir"/usr/share/gbaemu/gbaemu "$pkgdir"/usr/bin/
+}
+package_gbc-emu() {
+  pkgdesc='GBC.emu from the emu-ex-plus-alpha suite.'
+  conflicts=(gbc-emu)
+  replaces=(gbc-emu)
+  gendesk -f -n --name=GBC-emu --comment="${pkgdesc%.*}" --custom='Keywords=game;engine;' --exec='/usr/share/gbcemu/gbcemu' --icon=gbcemu --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname=gbcemu
+  install -dm 755 "$pkgdir"/usr/{bin,share/{gbcemu,applications,icons/hicolor/scalable/apps}}
+  cp -rL --no-preserve='ownership' "$pkgbase"/GBC.emu/{res/android/assets,target/linux}/* "$pkgdir"/usr/share/gbcemu/
+  install -m 644 gbcemu.desktop "$pkgdir"/usr/share/applications/
+  install -m 644 gbcemu.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  ln -rsf "$pkgdir"/usr/share/gbcemu/gbcemu "$pkgdir"/usr/bin/
+}
+package_md-emu() {
+  pkgdesc='MD.emu from the emu-ex-plus-alpha suite.'
+  depends+=(libarchive)
+  conflicts=(md-emu)
+  replaces=(md-emu)
+  gendesk -f -n --name=MD-emu --comment="${pkgdesc%.*}" --custom='Keywords=game;engine;' --exec='/usr/share/mdemu/mdemu' --icon=mdemu --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname=mdemu
+  install -dm 755 "$pkgdir"/usr/{bin,share/{mdemu,applications,icons/hicolor/scalable/apps}}
+  cp -rL --no-preserve='ownership' "$pkgbase"/MD.emu/{res/android/assets,target/linux}/* "$pkgdir"/usr/share/mdemu/
+  install -m 644 mdemu.desktop "$pkgdir"/usr/share/applications/
+  install -m 644 mdemu.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  ln -rsf "$pkgdir"/usr/share/mdemu/mdemu "$pkgdir"/usr/bin/
+}
+package_msx-emu() {
+  pkgdesc='MSX.emu from the emu-ex-plus-alpha suite.'
+  depends+=(libarchive)
+  conflicts=(msx-emu)
+  replaces=(msx-emu)
+  gendesk -f -n --name=MSX-emu --comment="${pkgdesc%.*}" --custom='Keywords=game;engine;' --exec='/usr/share/msxemu/msxemu' --icon=msxemu --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname=msxemu
+  install -dm 755 "$pkgdir"/usr/{bin,share/{msxemu,applications,icons/hicolor/scalable/apps}}
+  cp -rL --no-preserve='ownership' "$pkgbase"/MSX.emu/{res/android/assets,target/linux}/* "$pkgdir"/usr/share/msxemu/
+  install -m 644 msxemu.desktop "$pkgdir"/usr/share/applications/
+  install -m 644 msxemu.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  ln -rsf "$pkgdir"/usr/share/msxemu/msxemu "$pkgdir"/usr/bin/
+}
+package_neo-emu() {
+  pkgdesc='NEO.emu from the emu-ex-plus-alpha suite.'
+  conflicts=(neo-emu)
+  replaces=(neo-emu)
+  gendesk -f -n --name=NEO-emu --comment="${pkgdesc%.*}" --custom='Keywords=game;engine;' --exec='/usr/share/neoemu/neoemu' --icon=neoemu --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname=neoemu
+  install -dm 755 "$pkgdir"/usr/{bin,share/{neoemu,applications,icons/hicolor/scalable/apps}}
+  cp -rL --no-preserve='ownership' "$pkgbase"/NEO.emu/{res/android/assets,target/linux}/* "$pkgdir"/usr/share/neoemu/
+  install -m 644 neoemu.desktop "$pkgdir"/usr/share/applications/
+  install -m 644 neoemu.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  ln -rsf "$pkgdir"/usr/share/neoemu/neoemu "$pkgdir"/usr/bin/
+}
+package_nes-emu() {
+  pkgdesc='NES.emu from the emu-ex-plus-alpha suite.'
+  conflicts=(nes-emu)
+  replaces=(nes-emu)
+  gendesk -f -n --name=NES-emu --comment="${pkgdesc%.*}" --custom='Keywords=game;engine;' --exec='/usr/share/nesemu/nesemu' --icon=nesemu --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname=nesemu
+  install -dm 755 "$pkgdir"/usr/{bin,share/{nesemu,applications,icons/hicolor/scalable/apps}}
+  cp -rL --no-preserve='ownership' "$pkgbase"/NES.emu/{res/android/assets,target/linux}/* "$pkgdir"/usr/share/nesemu/
+  install -m 644 nesemu.desktop "$pkgdir"/usr/share/applications/
+  install -m 644 nesemu.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  ln -rsf "$pkgdir"/usr/share/nesemu/nesemu "$pkgdir"/usr/bin/
+}
+package_ngp-emu() {
+  pkgdesc='NGP.emu from the emu-ex-plus-alpha suite.'
+  conflicts=(ngp-emu)
+  replaces=(ngp-emu)
+  gendesk -f -n --name=NGP-emu --comment="${pkgdesc%.*}" --custom='Keywords=game;engine;' --exec='/usr/share/ngpemu/ngpemu' --icon=ngpemu --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname=ngpemu
+  install -dm 755 "$pkgdir"/usr/{bin,share/{ngpemu,applications,icons/hicolor/scalable/apps}}
+  cp -rL --no-preserve='ownership' "$pkgbase"/NGP.emu/{res/android/assets,target/linux}/* "$pkgdir"/usr/share/ngpemu/
+  install -m 644 ngpemu.desktop "$pkgdir"/usr/share/applications/
+  install -m 644 ngpemu.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  ln -rsf "$pkgdir"/usr/share/ngpemu/ngpemu "$pkgdir"/usr/bin/
+}
+package_pce-emu() {
+  pkgdesc='PCE.emu from the emu-ex-plus-alpha suite.'
+  conflicts=(pce-emu)
+  replaces=(pce-emu)
+  gendesk -f -n --name=PCE-emu --comment="${pkgdesc%.*}" --custom='Keywords=game;engine;' --exec='/usr/share/pceemu/pceemu' --icon=pceemu --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname=pceemu
+  install -dm 755 "$pkgdir"/usr/{bin,share/{pceemu,applications,icons/hicolor/scalable/apps}}
+  cp -rL --no-preserve='ownership' "$pkgbase"/PCE.emu/{res/android/assets,target/linux}/* "$pkgdir"/usr/share/pceemu/
+  install -m 644 pceemu.desktop "$pkgdir"/usr/share/applications/
+  install -m 644 pceemu.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  ln -rsf "$pkgdir"/usr/share/pceemu/pceemu "$pkgdir"/usr/bin/
+}
+package_snes9x-ex() {
+  pkgdesc='Snes9x EX+ from the emu-ex-plus-alpha suite.'
+  conflicts=(snes9x-ex)
+  replaces=(snes9x-ex)
+  gendesk -f -n --name=Snes9x-Ex --comment="${pkgdesc%.*}" --custom='Keywords=game;engine;' --exec='/usr/share/snes9xex/s9xp' --icon=snes9xex --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname=snes9xex
+  install -dm 755 "$pkgdir"/usr/{bin,share/{snes9xex,applications,icons/hicolor/scalable/apps}}
+  cp -rL --no-preserve='ownership' "$pkgbase"/Snes9x/{res/android/assets,target/linux}/* "$pkgdir"/usr/share/snes9xex/
+  install -m 644 snes9xex.desktop "$pkgdir"/usr/share/applications/
+  install -m 644 snes9x.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/snes9xex.svg
+  ln -rsf "$pkgdir"/usr/share/snes9xex/s9xp "$pkgdir"/usr/bin/
+}
diff a/emu-ex-plus-alpha/emu-ex-plus-alpha.install b/emu-ex-plus-alpha/emu-ex-plus-alpha.install
--- a/emu-ex-plus-alpha/emu-ex-plus-alpha.install	1969-12-31 17:00:00.000000000 -0700
+++ b/emu-ex-plus-alpha/emu-ex-plus-alpha.install	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,21 @@
+# Colored makepkg-like functions
+msg_blue() {
+    printf "${blue}==>${bold} $1${all_off}\n"
+}
+
+note() {
+    printf "${blue}==>${yellow} NOTE:${bold} $1${all_off}\n"
+}
+
+all_off="$(tput sgr0)"
+bold="${all_off}$(tput bold)"
+blue="${bold}$(tput setaf 4)"
+yellow="${bold}$(tput setaf 3)"
+
+post_install() {
+    note "Ensure xrandr has a primary output:  xrandr | grep primary"
+    note "set with:  xrandr --output (output name) --primary"
+}
+post_upgrade() {
+    post_install
+}
