diff a/emu-ex-plus-alpha/PKGBUILD b/emu-ex-plus-alpha/PKGBUILD
--- a/emu-ex-plus-alpha/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/emu-ex-plus-alpha/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,203 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgbase=emu-ex-plus-alpha
+pkgname=(2600-emu c64-emu gba-emu gbc-emu lynx-emu md-emu msx-emu neo-emu nes-emu ngp-emu pce-emu snes9x-ex swan-emu)
+pkgver=1.5.65.r492.g8f236c1c
+pkgrel=1
+arch=(x86_64)
+url=https://www.explusalpha.com
+license=(GPL3)
+depends=(alsa-lib bluez-libs fontconfig freetype2 hicolor-icon-theme libglvnd libpulse libxi libxrandr)
+makedepends=('gcc>=14' gendesk git inkscape mold)
+source=("$pkgbase"::git+https://github.com/Rakashazi/"$pkgbase".git
+        C64.emu.zip
+        MSX.emu.zip)
+sha256sums=('SKIP'
+            '97d8b545c950b6a01a8876d84dca9cd5405895ca3b0403f1f13070b54b03c16e'
+            '1e7fc01cb3ad0edd48c9021a227ff8997f54e3cdcdda1328512cbe6ad0764e32')
+
+pkgver() {
+  cd "$pkgbase"
+  local _ver="$(grep -Pio "vers.*\b.*\d+\.[[:alnum:]\.]+" imagine/include/imagine/config/version.h | grep -Pio '\d+\.[[:alnum:]\.]+')"
+  printf "%s.r%s.g%s" "$_ver" "$(git rev-list --count HEAD)" "$(git rev-parse --short HEAD)"
+}
+
+prepare() {
+  perl -pi -e 's/(?<=g..)-14//gi' "$pkgbase"/imagine/make/linux*gcc.mk
+  perl -pi -e 's/(?<=x86-64)-v3//gi' "$pkgbase"/imagine/make/linux-x86_64-gcc.mk
+  for _emu in 2600.emu C64.emu GBA.emu GBC.emu Lynx.emu MD.emu MSX.emu NEO.emu NES.emu NGP.emu PCE.emu Snes9x Swan.emu; do
+  _sub="${_emu//./}"
+  _sub="${_sub,,}"
+  if [ ! -f "$_sub.svg" ]; then
+    magick -thumbnail 256x256 -strip "$pkgbase/$_emu"/res/icons/icon-144.png "$_sub".png
+    inkscape "$_sub".png --export-plain-svg --export-filename="$_sub".svg
+    rm -f "$_sub".png
+  fi
+  done
+  cd "$pkgbase"
+  rm -rf {imagine,EmuFramework,2600.emu,C64.emu,GBA.emu,GBC.emu,Lynx.emu,MD.emu,MSX.emu,NEO.emu,NES.emu,NGP.emu,PCE.emu,Snes9x,Swan.emu}/{build,target} build imagine-sdk
+}
+
+build() {
+  _imagine="$(realpath "$pkgbase")"
+  export IMAGINE_PATH="$_imagine"/imagine
+  export IMAGINE_SDK_PATH="$IMAGINE_PATH"-sdk
+  export PKG_CONFIG_PATH="$IMAGINE_SDK_PATH"/lib/pkgconfig
+  cd "$pkgbase"/imagine/bundle/all
+  bash makeAll-linux-x86_64-static.sh tempDir="$_imagine"/build installDir="$IMAGINE_SDK_PATH" install
+  for _emu in imagine EmuFramework 2600.emu C64.emu GBA.emu GBC.emu Lynx.emu MD.emu MSX.emu NEO.emu NES.emu NGP.emu PCE.emu Snes9x Swan.emu; do
+    cd "$_imagine/$_emu"
+    echo "$_emu" | grep -Pioq 'imagine|EmuFramework' && make -f linux-x86_64-release.mk -j$(nproc) install prefix="$IMAGINE_SDK_PATH" || make -f linux-x86_64-release.mk -j$(nproc)
+  done
+#  cd ../C64.emu
+#  make -f linux-x86_64-release.mk -j$(nproc) CFLAGS_OPTIMIZE_LEVEL_RELEASE_DEFAULT=-O3
+}
+
+package_2600-emu() {
+  pkgdesc='2600.emu from the emu-ex-plus-alpha suite.'
+  conflicts=("$pkgname")
+  replaces=("$pkgname")
+  gendesk -f -n --name=2600.emu --comment="${pkgdesc%.*}" --custom='Keywords=game;engine;' --exec='/usr/share/2600emu/2600emu' --icon=2600emu --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname=2600emu
+  install -dm 755 "$pkgdir"/usr/{bin,share/{2600emu,applications,icons/hicolor/scalable/apps}}
+  cp -rL --no-preserve='ownership' "$pkgbase"/2600.emu/{res/android/assets,target/linux}/* "$pkgdir"/usr/share/2600emu/
+  install -m 644 2600emu.desktop "$pkgdir"/usr/share/applications/
+  install -m 644 2600emu.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  ln -rsf "$pkgdir"/usr/share/2600emu/2600emu "$pkgdir"/usr/bin/
+}
+package_c64-emu() {
+  pkgdesc='C64.emu from the emu-ex-plus-alpha suite.'
+  conflicts=("$pkgname")
+  replaces=("$pkgname")
+  gendesk -f -n --name=C64.emu --comment="${pkgdesc%.*}" --custom='Keywords=game;engine;' --exec='/usr/share/c64emu/c64emu' --icon=c64emu --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname=c64emu
+  install -dm 755 "$pkgdir"/usr/{bin,share/{c64emu,applications,icons/hicolor/scalable/apps}}
+  cp -rL --no-preserve='ownership' "$pkgbase"/C64.emu/{res/android/assets,target/linux}/* "$pkgdir"/usr/share/c64emu/
+  cp -dfr --no-preserve='ownership' "$srcdir"/C64.emu/* "$pkgdir"/usr/share/c64emu/
+  install -m 644 c64emu.desktop "$pkgdir"/usr/share/applications/
+  install -m 644 c64emu.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  ln -rsf "$pkgdir"/usr/share/c64emu/c64emu "$pkgdir"/usr/bin/
+}
+package_gba-emu() {
+  pkgdesc='GBA.emu from the emu-ex-plus-alpha suite.'
+  conflicts=("$pkgname")
+  replaces=("$pkgname")
+  gendesk -f -n --name=GBA.emu --comment="${pkgdesc%.*}" --custom='Keywords=game;engine;' --exec='/usr/share/gbaemu/gbaemu' --icon=gbaemu --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname=gbaemu
+  install -dm 755 "$pkgdir"/usr/{bin,share/{gbaemu,applications,icons/hicolor/scalable/apps}}
+  cp -rL --no-preserve='ownership' "$pkgbase"/GBA.emu/{res/android/assets,target/linux}/* "$pkgdir"/usr/share/gbaemu/
+  install -m 644 gbaemu.desktop "$pkgdir"/usr/share/applications/
+  install -m 644 gbaemu.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  ln -rsf "$pkgdir"/usr/share/gbaemu/gbaemu "$pkgdir"/usr/bin/
+}
+package_gbc-emu() {
+  pkgdesc='GBC.emu from the emu-ex-plus-alpha suite.'
+  conflicts=("$pkgname")
+  replaces=("$pkgname")
+  gendesk -f -n --name=GBC.emu --comment="${pkgdesc%.*}" --custom='Keywords=game;engine;' --exec='/usr/share/gbcemu/gbcemu' --icon=gbcemu --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname=gbcemu
+  install -dm 755 "$pkgdir"/usr/{bin,share/{gbcemu,applications,icons/hicolor/scalable/apps}}
+  cp -rL --no-preserve='ownership' "$pkgbase"/GBC.emu/{res/android/assets,target/linux}/* "$pkgdir"/usr/share/gbcemu/
+  install -m 644 gbcemu.desktop "$pkgdir"/usr/share/applications/
+  install -m 644 gbcemu.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  ln -rsf "$pkgdir"/usr/share/gbcemu/gbcemu "$pkgdir"/usr/bin/
+}
+package_lynx-emu() {
+  pkgdesc='Lynx.emu from the emu-ex-plus-alpha suite.'
+  conflicts=("$pkgname")
+  replaces=("$pkgname")
+  gendesk -f -n --name=Lynx.emu --comment="${pkgdesc%.*}" --custom='Keywords=game;engine;' --exec='/usr/share/lynxemu/lynxemu' --icon=lynxemu --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname=lynxemu
+  install -dm 755 "$pkgdir"/usr/{bin,share/{lynxemu,applications,icons/hicolor/scalable/apps}}
+  cp -rL --no-preserve='ownership' "$pkgbase"/Lynx.emu/{res/android/assets,target/linux}/* "$pkgdir"/usr/share/lynxemu/
+  install -m 644 lynxemu.desktop "$pkgdir"/usr/share/applications/
+  install -m 644 lynxemu.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  ln -rsf "$pkgdir"/usr/share/lynxemu/lynxemu "$pkgdir"/usr/bin/
+}
+package_md-emu() {
+  pkgdesc='MD.emu from the emu-ex-plus-alpha suite.'
+  depends+=(libarchive)
+  conflicts=("$pkgname")
+  replaces=("$pkgname")
+  gendesk -f -n --name=MD.emu --comment="${pkgdesc%.*}" --custom='Keywords=game;engine;' --exec='/usr/share/mdemu/mdemu' --icon=mdemu --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname=mdemu
+  install -dm 755 "$pkgdir"/usr/{bin,share/{mdemu,applications,icons/hicolor/scalable/apps}}
+  cp -rL --no-preserve='ownership' "$pkgbase"/MD.emu/{res/android/assets,target/linux}/* "$pkgdir"/usr/share/mdemu/
+  install -m 644 mdemu.desktop "$pkgdir"/usr/share/applications/
+  install -m 644 mdemu.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  ln -rsf "$pkgdir"/usr/share/mdemu/mdemu "$pkgdir"/usr/bin/
+}
+package_msx-emu() {
+  pkgdesc='MSX.emu from the emu-ex-plus-alpha suite.'
+  depends+=(libarchive)
+  conflicts=("$pkgname")
+  replaces=("$pkgname")
+  gendesk -f -n --name=MSX.emu --comment="${pkgdesc%.*}" --custom='Keywords=game;engine;' --exec='/usr/share/msxemu/msxemu' --icon=msxemu --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname=msxemu
+  install -dm 755 "$pkgdir"/usr/{bin,share/{msxemu,applications,icons/hicolor/scalable/apps}}
+  cp -rL --no-preserve='ownership' "$pkgbase"/MSX.emu/{res/android/assets,target/linux}/* "$pkgdir"/usr/share/msxemu/
+  cp -dfr --no-preserve='ownership' "$srcdir"/MSX.emu/* "$pkgdir"/usr/share/msxemu/
+  chmod 755 "$pkgdir"/usr/share/msxemu/Machines/{'COL - ColecoVision',MSXturboR}
+  install -m 644 msxemu.desktop "$pkgdir"/usr/share/applications/
+  install -m 644 msxemu.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  ln -rsf "$pkgdir"/usr/share/msxemu/msxemu "$pkgdir"/usr/bin/
+}
+package_neo-emu() {
+  pkgdesc='NEO.emu from the emu-ex-plus-alpha suite.'
+  conflicts=("$pkgname")
+  replaces=("$pkgname")
+  gendesk -f -n --name=NEO.emu --comment="${pkgdesc%.*}" --custom='Keywords=game;engine;' --exec='/usr/share/neoemu/neoemu' --icon=neoemu --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname=neoemu
+  install -dm 755 "$pkgdir"/usr/{bin,share/{neoemu,applications,icons/hicolor/scalable/apps}}
+  cp -rL --no-preserve='ownership' "$pkgbase"/NEO.emu/{res/android/assets,target/linux}/* "$pkgdir"/usr/share/neoemu/
+  install -m 644 neoemu.desktop "$pkgdir"/usr/share/applications/
+  install -m 644 neoemu.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  ln -rsf "$pkgdir"/usr/share/neoemu/neoemu "$pkgdir"/usr/bin/
+}
+package_nes-emu() {
+  pkgdesc='NES.emu from the emu-ex-plus-alpha suite.'
+  conflicts=("$pkgname")
+  replaces=("$pkgname")
+  gendesk -f -n --name=NES.emu --comment="${pkgdesc%.*}" --custom='Keywords=game;engine;' --exec='/usr/share/nesemu/nesemu' --icon=nesemu --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname=nesemu
+  install -dm 755 "$pkgdir"/usr/{bin,share/{nesemu,applications,icons/hicolor/scalable/apps}}
+  cp -rL --no-preserve='ownership' "$pkgbase"/NES.emu/{res/android/assets,target/linux}/* "$pkgdir"/usr/share/nesemu/
+  install -m 644 nesemu.desktop "$pkgdir"/usr/share/applications/
+  install -m 644 nesemu.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  ln -rsf "$pkgdir"/usr/share/nesemu/nesemu "$pkgdir"/usr/bin/
+}
+package_ngp-emu() {
+  pkgdesc='NGP.emu from the emu-ex-plus-alpha suite.'
+  conflicts=("$pkgname")
+  replaces=("$pkgname")
+  gendesk -f -n --name=NGP.emu --comment="${pkgdesc%.*}" --custom='Keywords=game;engine;' --exec='/usr/share/ngpemu/ngpemu' --icon=ngpemu --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname=ngpemu
+  install -dm 755 "$pkgdir"/usr/{bin,share/{ngpemu,applications,icons/hicolor/scalable/apps}}
+  cp -rL --no-preserve='ownership' "$pkgbase"/NGP.emu/{res/android/assets,target/linux}/* "$pkgdir"/usr/share/ngpemu/
+  install -m 644 ngpemu.desktop "$pkgdir"/usr/share/applications/
+  install -m 644 ngpemu.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  ln -rsf "$pkgdir"/usr/share/ngpemu/ngpemu "$pkgdir"/usr/bin/
+}
+package_pce-emu() {
+  pkgdesc='PCE.emu from the emu-ex-plus-alpha suite.'
+  conflicts=("$pkgname")
+  replaces=("$pkgname")
+  gendesk -f -n --name=PCE.emu --comment="${pkgdesc%.*}" --custom='Keywords=game;engine;' --exec='/usr/share/pceemu/pceemu' --icon=pceemu --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname=pceemu
+  install -dm 755 "$pkgdir"/usr/{bin,share/{pceemu,applications,icons/hicolor/scalable/apps}}
+  cp -rL --no-preserve='ownership' "$pkgbase"/PCE.emu/{res/android/assets,target/linux}/* "$pkgdir"/usr/share/pceemu/
+  install -m 644 pceemu.desktop "$pkgdir"/usr/share/applications/
+  install -m 644 pceemu.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  ln -rsf "$pkgdir"/usr/share/pceemu/pceemu "$pkgdir"/usr/bin/
+}
+package_snes9x-ex() {
+  pkgdesc='Snes9x EX+ from the emu-ex-plus-alpha suite.'
+  conflicts=("$pkgname")
+  replaces=("$pkgname")
+  gendesk -f -n --name=Snes9x-Ex --comment="${pkgdesc%.*}" --custom='Keywords=game;engine;' --exec='/usr/share/snes9xex/s9xp' --icon=snes9xex --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname=snes9xex
+  install -dm 755 "$pkgdir"/usr/{bin,share/{snes9xex,applications,icons/hicolor/scalable/apps}}
+  cp -rL --no-preserve='ownership' "$pkgbase"/Snes9x/{res/android/assets,target/linux}/* "$pkgdir"/usr/share/snes9xex/
+  install -m 644 snes9xex.desktop "$pkgdir"/usr/share/applications/
+  install -m 644 snes9x.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/snes9xex.svg
+  ln -rsf "$pkgdir"/usr/share/snes9xex/s9xp "$pkgdir"/usr/bin/
+}
+package_swan-emu() {
+  pkgdesc='Swan.emu from the emu-ex-plus-alpha suite.'
+  conflicts=("$pkgname")
+  replaces=("$pkgname")
+  gendesk -f -n --name=Swan.emu --comment="${pkgdesc%.*}" --custom='Keywords=game;engine;' --exec='/usr/share/swanemu/swanemu' --icon=swanemu --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname=swanemu
+  install -dm 755 "$pkgdir"/usr/{bin,share/{swanemu,applications,icons/hicolor/scalable/apps}}
+  cp -rL --no-preserve='ownership' "$pkgbase"/Swan.emu/{res/android/assets,target/linux}/* "$pkgdir"/usr/share/swanemu/
+  install -m 644 swanemu.desktop "$pkgdir"/usr/share/applications/
+  install -m 644 swanemu.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  ln -rsf "$pkgdir"/usr/share/swanemu/swanemu "$pkgdir"/usr/bin/
+}
