diff a/ezquake/PKGBUILD b/ezquake/PKGBUILD
--- a/ezquake/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/ezquake/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,64 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=ezquake
+pkgver=3.6.9.r119.ge61b3a6
+pkgrel=1
+pkgdesc="Modern QuakeWorld focused on competitive online play."
+arch=(x86_64)
+url=http://ezquake.github.io
+license=(GPL-2.0-only)
+depends=(curl expat freetype2 hicolor-icon-theme jansson libjpeg-turbo
+         libpng openssl perl-rename sdl2 speex xorg-fonts-misc)
+makedepends=(gendesk git icoutils inkscape zip)
+provides=("$pkgname")
+conflicts=("$pkgname"-git quake fuhquake)
+replaces=("$pkgname"-git quake fuhquake)
+source=(
+  "$pkgname"::git+https://github.com/ezQuake/"$pkgname"-source.git
+  "media"::git+https://github.com/ezQuake/"$pkgname"-media.git
+   https://github.com/noabody/unibuild/raw/master/data/progs_dat.tar.xz
+  "$pkgname".sh
+)
+sha256sums=('SKIP'
+            'SKIP'
+            '033674c70d20f87a61165974c08f205fc7718516548312ed7fe1e2319bb39d3f'
+            '87091a7aa3a1119af6c692716c4a8c30aa01150dc5f891e63f1b89a4088e122f')
+
+pkgver() {
+  cd "$pkgname"
+  git describe --long --abbrev=7 --tags | sed 's/\([^-]*-g\)/r\1/;s/-/./g'
+}
+
+prepare() {
+  if [[ ! -f "$pkgname.svg" ]]; then
+    inkscape media/installer/"$pkgname"_logo.png --export-plain-svg --export-filename="$pkgname".svg
+  fi
+  git -C "$pkgname" submodule update --init --recursive
+  [[ -h "media/game/cfg" ]] || ln -sf "$pkgname"/misc/cfg media/game/
+  [[ -h "media/game/keymaps" ]] || ln -sf "$pkgname"/misc/keymaps media/game/
+  [[ -d "media/game/sb" ]] || mkdir -p media/game/sb
+  [[ -f "media/game/sb/sources.txt" ]] || cp "$pkgname"/misc/sb/sources.txt media/game/sb/
+  [[ ! -d build ]] || rm -rf build
+}
+
+build() {
+  cmake -S "$pkgname" -B build \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_SKIP_RPATH=ON
+  cmake --build build -j$(nproc)
+
+  cd media/game
+  [[ -f ezquake.pk3 ]] && rm -f ezquake.pk3
+  zip -Z deflate -u -r ezquake.pk3 cfg gfx keymaps locs progs radars sb textures *progs.dat
+}
+
+package() {
+  gendesk -f -n --name='ezQuake' --comment="${pkgdesc/.}" --custom='Keywords=game;engine;' --exec="/usr/share/$pkgname/$pkgname.sh" --icon="$pkgname" --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname="$pkgname"
+  install -dm 755 "$pkgdir"/usr/{bin,share/{"$pkgname",applications,icons/hicolor/scalable/apps}}
+  install -m 755 build/"$pkgname"-linux-"$(uname -m)" "$pkgdir"/usr/share/"$pkgname/$pkgname"
+  install -m 755 "$pkgname".sh "$pkgdir"/usr/share/"$pkgname"/
+  install -m 644 media/game/ezquake.pk3 "$pkgdir"/usr/share/"$pkgname"/
+  install -m 644 "$pkgname".desktop "$pkgdir"/usr/share/applications/
+  install -m 644 "$pkgname".svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  ln -rsf "$pkgdir"/usr/share/"$pkgname/$pkgname" "$pkgdir"/usr/bin/
+}
diff a/ezquake/ezquake.sh b/ezquake/ezquake.sh
--- a/ezquake/ezquake.sh	1969-12-31 17:00:00.000000000 -0700
+++ b/ezquake/ezquake.sh	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,80 @@
+#!/bin/bash
+gmdir="$HOME/games/quake"
+qwdir="$HOME/games/nquake"
+gmcfg="$HOME/.ezquake"
+gmtgt="id1/pak0.pak"
+gmprm="+vid_restart +gamma 0.8 +contrast 1.0"
+glnch="ezquake"
+i_syms=()
+
+gconf() {
+mkdir -p "$gmcfg"
+printf 'gamepath=%s\ngamemod=%s\ngameparm=%s\n' "$gmdir" "id1" "$gmprm" > "$gmcfg/basedir"
+}
+
+gstart () {
+mkdir -p "$qwdir/ezquake"
+for value in id1 hipnotic rogue; do
+  if [[ -d "$gmbdr/$value" ]]; then
+    mkdir -p "$qwdir/$value"
+    while IFS= read -r -d '' file; do
+      [[ -h "$qwdir/$value/$file" ]] || ln -rsf "$gmbdr/$value/$file" "$qwdir/$value/$file"
+    done < <(find "$gmbdr/$value" -type f -iname '*.pak' -printf '%P\0')
+  fi
+done
+while IFS= read -r -d '' file; do
+  [[ -h "$qwdir/ezquake/$file" ]] || ln -rsf "/usr/share/ezquake/$file" "$qwdir/ezquake/$file"
+done < <(find /usr/share/ezquake -mindepth 1 -type f -iname '*.pk3' -printf '%P\0')
+echo "$glnch +cfg_use_home 1 -basedir $qwdir $gmgnm $gmpar" | xargs -i -r sh -c "{}"
+}
+
+gtest () {
+gmbse="" gmbdr="" gmgme="" gmgnm="" gmgpm="" gmpar=""
+
+if [ -f "$gmcfg/basedir" ]; then
+  while IFS='=' read -r key val || [ -n "$key" ]; do
+    case "$key" in
+      gamepath) gmbse="gamepath=$val"; gmbdr="$val" ;;
+      gamemod)  gmgme="gamemod=$val";  gmgnm="$val" ;;
+      gameparm) gmgpm="gameparm=$val"; gmpar="$val" ;;
+    esac
+  done < "$gmcfg/basedir"
+fi
+}
+
+[[ -f "$gmcfg/basedir" ]] || gconf
+gtest
+if [[ -z "$gmbdr" ]]; then
+  gconf
+  gtest
+fi
+if [[ -d "$gmbdr" ]]; then
+  if [[ ! -f "$gmbdr/$gmtgt" ]]; then
+    gmchk="$(find -H "$gmbdr" -mindepth 2 -maxdepth 2 -type f -regextype posix-extended -iregex "$gmbdr/$gmtgt" -printf '%p\n' | grep -Pic "$gmbdr/$gmtgt")"
+    if (( $gmchk == 1 )) ; then
+      echo -e "File system is case sensitive\nFile/folder names must be lowercase\n Rename subdirs of\n  $gmbdr\n    containing\n  $gmtgt?\nRequires perl-rename command" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      (( $? == 101 )) && find -H "$gmbdr" -depth -exec perl-rename 's|[^/]*\Z|\L$&|' {} +
+    elif (( $gmchk > 1 )) ; then
+      echo -e "File system is case sensitive\nFile/folder names must be lowercase\n Cannot rename subdirs of\n  $gmbdr\n    containing\n  $gmtgt\n\nBecause more than one exists" | xmessage -file - -buttons Ok:0 -center -timeout 30
+    else
+      echo -e "Config:\n $gmcfg/basedir\n  $gmbse\n found, but not\n  $gmgme\n ($gmbdr/$gmtgt)\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      (( $? == 101 )) && gconf
+    fi
+  else
+    if [[ ! -d "$gmbdr/$gmgnm" ]]; then
+      echo -e "Config:\n $gmcfg/basedir\n  $gmbse\n found, but not\n  $gmgme\n ($gmbdr/$gmgnm)\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      (( $? == 101 )) && gconf
+    else
+      if [[ -z "$(echo "$gmgnm" | grep -Pio '\w+')" ]]; then
+        gmgnm=""
+        gstart
+      else
+        gmgnm="-game $gmgnm"
+        gstart
+      fi
+    fi
+  fi
+else
+  echo -e "Config:\n $gmcfg/basedir\n  $gmbse\n not found\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+  (( $? == 101 )) && gconf
+fi
