diff a/freepats-legacy/PKGBUILD b/freepats-legacy/PKGBUILD
--- a/freepats-legacy/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/freepats-legacy/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,43 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=freepats-legacy
+pkgver=20060219
+pkgrel=1
+pkgdesc="A free and open set of instrument patches for use with softsynths (old version)"
+arch=('any')
+url="https://web.archive.org/web/20241004222803/https://freepats.zenvoid.org/SoundSets/general-midi.html"
+license=(GPL-1.0-only)
+provides=("timidity-freepats=$pkgver")
+conflicts=("timidity-freepats=$pkgver")
+optdepends=('timidity++: MIDI Sequencer and converter'
+            'wildmidi: Another MIDI Synthesizer')
+install=$pkgname.install
+source=("https://web.archive.org/web/20241004222803/https://freepats.zenvoid.org/freepats-$pkgver.tar.xz"
+        'freepats-legacy.install'
+        'freepats.cfg')
+sha256sums=('500c61782ff4b22de6887c0a32e68dd98b511c4396ddf89e8cab482c7dcea89e'
+            'e18c39623ed6f68d60177be90cda599d46548512b5f4a4bc67a4721f445f98e4'
+            '7a37808753321ac31554b2f2ad9b063f73a9a9bf64799cbb2165bd9247140fc2')
+
+prepare() {
+  # wildmidi does not support soundfont settings, so remove them
+  tail -n3 freepats.cfg > wildmidi-freepats.cfg
+}
+
+package() {
+  install -d "$pkgdir"/usr/share/freepats/{Drum,Tone}_000
+  install -m644 freepats.cfg "$pkgdir"/usr/share/freepats/timidity-freepats.cfg
+  install -m644 wildmidi-freepats.cfg "$pkgdir"/usr/share/freepats/wildmidi-freepats.cfg
+
+  cd freepats
+  install -m644 Drum_000/*.{pat,txt} "$pkgdir"/usr/share/freepats/Drum_000
+  install -m644 Tone_000/*.{pat,txt} "$pkgdir"/usr/share/freepats/Tone_000
+  install -m644 crude.cfg "$pkgdir"/usr/share/freepats/freepats.cfg
+  install -Dm644 README "$pkgdir"/usr/share/doc/freepats/README
+
+  # support old paths (from old timidity-freepats package)
+  install -d "$pkgdir"/usr/share/timidity
+  ln -s /usr/share/freepats "$pkgdir"/usr/share/timidity/freepats
+  install -d "$pkgdir"/etc/timidity++/freepats
+  ln -s /usr/share/freepats/freepats.cfg "$pkgdir"/etc/timidity++/freepats/freepats.cfg
+}
diff a/freepats-legacy/freepats-legacy.install b/freepats-legacy/freepats-legacy.install
--- a/freepats-legacy/freepats-legacy.install	1969-12-31 17:00:00.000000000 -0700
+++ b/freepats-legacy/freepats-legacy.install	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,9 @@
+post_install() {
+  echo "=> To make TiMidity++ or WildMidi use the FreePats patches, copy"
+  echo "   /usr/share/freepats/timidity-freepats.cfg to /etc/timidity++/timidity.cfg"
+  echo "   or /usr/share/freepats/wildmidi-freepats.cfg to /etc/wildmidi/wildmidi.cfg"
+}
+
+post_upgrade() {
+  post_install
+}
diff a/freepats-legacy/freepats.cfg b/freepats-legacy/freepats.cfg
--- a/freepats-legacy/freepats.cfg	1969-12-31 17:00:00.000000000 -0700
+++ b/freepats-legacy/freepats.cfg	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,32 @@
+# details on this file and other examples see here:
+# http://www.onicos.com/staff/iz/timidity/dist/cfg/sndfont.cfg
+# http://www.onicos.com/staff/iz/timidity/dist/cfg/
+#----------------------------------------------------------------
+# SoundFont extension configuration
+#
+# soundfont <filename> [order={0|1}] [remove]
+#    <filename> is the path of SoundFont file.
+#    order: 0(preload) or 1(load after GUS).
+#    remove: disable specified soundfont.
+#
+# font exclude <tonebank> [<preset> [<keynote>]]
+# font order {0|1} <tonebank> [<preset> [<keynote>]]
+#    <tonebank>: 0-127
+#
+# font exclude 128 [<drumset> [<key>]]
+# font order {0|1} 128 [<drumset> [<key>]]
+#----------------------------------------------------------------
+
+# change the next line to point to a soundfont you have
+# soundfonts can be found e.g. here: http://www.hammersound.net/
+# soundfont /usr/share/soundfonts/DX7Piano.SF2
+
+# you can do manipulations of the soundfont you load like this:
+# e.g. removing drumset 99
+# font exclude 128 99
+
+# load drum samples after GUS patches
+# font order 1 128
+
+dir /usr/share/freepats
+source /usr/share/freepats/freepats.cfg
