diff a/fteqw/PKGBUILD b/fteqw/PKGBUILD
--- a/fteqw/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/fteqw/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,52 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=fteqw
+pkgver=r6343
+pkgrel=1
+pkgdesc="Highly versatile evolution of QuakeWorld."
+arch=(x86_64)
+url=http://fte.triptohell.info
+license=(GPL-2.0-only)
+depends=(hicolor-icon-theme freetype2 libjpeg-turbo libpng libvorbis
+         opus perl-rename speex speexdsp xorg-fonts-misc zlib)
+makedepends=(ecasound gendesk icoutils inkscape libogg libpng
+             mesa subversion vulkan-validation-layers zlib)
+conflicts=(ftequake ftequake-svn)
+source=(
+  "$pkgname"::svn+http://svn.code.sf.net/p/"$pkgname"/code/trunk
+  "$pkgname".sh
+)
+sha256sums=('SKIP'
+            'b029b258fefa19e59f577cc98e394e771eb1e5c06a12193c73564d1e12713454')
+
+pkgver() {
+  cd "$pkgname"
+  local ver="$(svnversion)"
+  printf "r%s" "${ver//[[:alpha:]]}"
+}
+
+prepare() {
+  gendesk -f -n --name='FTEQW' --comment="${pkgdesc/.}" --custom='Keywords=game;engine;' --exec="/usr/share/$pkgname/$pkgname.sh" --icon="$pkgname" --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname="$pkgname"
+  cp "$pkgname".{sh,desktop} "$pkgname"/engine/
+  make -C "$pkgname"/engine clean
+}
+
+build() {
+  cd "$pkgname"/engine
+  if [[ ! -f "$pkgname.svg" ]]; then
+    icotool -i 8 -x client/fte_eukara.ico -o "$pkgname".png
+    inkscape "$pkgname".png  --export-plain-svg --export-filename="$pkgname".svg
+    rm -f "$pkgname".png
+  fi
+  make -j$(nproc) NATIVE_PLUGINS='ezhud qi hl2 openssl quake3 ode xmpp irc openxr' gl-rel sv-rel vk-rel plugins-rel CFLAGS+=' -std=gnu17'
+}
+
+package() {
+  install -dm 755 "$pkgdir"/usr/{bin,share/{"$pkgname",applications,icons/hicolor/scalable/apps}}
+  install -m 755 "$pkgname"/engine/release/"$pkgname"-{gl,sv,vk} "$pkgdir"/usr/share/"$pkgname"/
+  install -m 755 "$pkgname"/engine/"$pkgname".sh "$pkgdir"/usr/share/"$pkgname"/
+  install -m 644 "$pkgname"/engine/release/fteplug_*.so "$pkgdir"/usr/share/"$pkgname"/
+  install -m 644 "$pkgname"/engine/"$pkgname".desktop "$pkgdir"/usr/share/applications/
+  install -m 644 "$pkgname"/engine/"$pkgname".svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  ln -rsf "$pkgdir"/usr/share/"$pkgname/$pkgname"-{gl,sv,vk} "$pkgdir"/usr/bin/
+}
diff a/fteqw/fteqw.sh b/fteqw/fteqw.sh
--- a/fteqw/fteqw.sh	1969-12-31 17:00:00.000000000 -0700
+++ b/fteqw/fteqw.sh	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,75 @@
+#!/bin/bash
+gmdir="$HOME/games/quake"
+qwdir="$HOME/games/aquake"
+gmcfg="$HOME/.local/share/fte"
+gmtgt="id1/pak0.pak"
+gmprm="+vid_restart"
+glnch="fteqw-vk"
+
+gconf() {
+mkdir -p "$gmcfg"
+printf 'gamepath=%s\ngamemod=%s\ngameparm=%s\n' "$gmdir" "id1" "$gmprm" > "$gmcfg/basedir"
+}
+
+gstart () {
+for value in id1 hipnotic rogue; do
+  if [[ -d "$gmbdr/$value" ]]; then
+    mkdir -p "$qwdir/$value"
+    while IFS= read -r -d '' file; do
+      [[ -h "$qwdir/$value/$file" ]] || ln -rsf "$gmbdr/$value/$file" "$qwdir/$value/$file"
+    done < <(find "$gmbdr/$value" -type f -iname '*.pak' -printf '%P\0')
+  fi
+done
+echo "$glnch -homedir $gmcfg/ -basedir $qwdir $gmgnm $gmpar" | xargs -i -r sh -c "{}"
+}
+
+gtest () {
+gmbse="" gmbdr="" gmgme="" gmgnm="" gmgpm="" gmpar=""
+
+if [[ -f "$gmcfg/basedir" ]]; then
+  while IFS='=' read -r key val || [[ -n "$key" ]]; do
+    case "$key" in
+      gamepath) gmbse="gamepath=$val"; gmbdr="$val" ;;
+      gamemod)  gmgme="gamemod=$val";  gmgnm="$val" ;;
+      gameparm) gmgpm="gameparm=$val"; gmpar="$val" ;;
+    esac
+  done < "$gmcfg/basedir"
+fi
+}
+
+[[ -f "$gmcfg/basedir" ]] || gconf
+gtest
+if [[ -z "$gmbdr" ]]; then
+  gconf
+  gtest
+fi
+if [[ -d "$gmbdr" ]]; then
+  if [[ ! -f "$gmbdr/$gmtgt" ]]; then
+    gmchk="$(find -H "$gmbdr" -mindepth 2 -maxdepth 2 -type f -regextype posix-extended -iregex "$gmbdr/$gmtgt" -printf '%p\n' | grep -Pic "$gmbdr/$gmtgt")"
+    if (( $gmchk == 1 )) ; then
+      echo -e "File system is case sensitive\nFile/folder names must be lowercase\n Rename subdirs of\n  $gmbdr\n    containing\n  $gmtgt?\nRequires perl-rename command" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      (( $? == 101 )) && find -H "$gmbdr" -depth -exec perl-rename 's|[^/]*\Z|\L$&|' {} +
+    elif (( $gmchk > 1 )) ; then
+      echo -e "File system is case sensitive\nFile/folder names must be lowercase\n Cannot rename subdirs of\n  $gmbdr\n    containing\n  $gmtgt\n\nBecause more than one exists" | xmessage -file - -buttons Ok:0 -center -timeout 30
+    else
+      echo -e "Config:\n $gmcfg/basedir\n  $gmbse\n found, but not\n  $gmgme\n ($gmbdr/$gmtgt)\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      (( $? == 101 )) && gconf
+    fi
+  else
+    if [[ ! -d "$gmbdr/$gmgnm" ]]; then
+      echo -e "Config:\n $gmcfg/basedir\n  $gmbse\n found, but not\n  $gmgme\n ($gmbdr/$gmgnm)\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      (( $? == 101 )) && gconf
+    else
+      if [[ -z "$(echo "$gmgnm" | grep -Pio '\w+')" ]]; then
+        gmgnm=""
+        gstart
+      else
+        gmgnm="-game $gmgnm"
+        gstart
+      fi
+    fi
+  fi
+else
+  echo -e "Config:\n $gmcfg/basedir\n  $gmbse\n not found\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+  (( $? == 101 )) && gconf
+fi
