diff a/gearsystem/PKGBUILD b/gearsystem/PKGBUILD
--- a/gearsystem/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/gearsystem/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,43 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname="gearsystem"
+pkgver=3.4.1.r38
+pkgrel=1
+pkgdesc='Sega Master System, Game Gear, SG-1000 Emulator.'
+url=https://github.com/drhelius/Gearsystem
+arch=(x86_64)
+license=(GPL3)
+depends=(glew hicolor-icon-theme sdl2)
+makedepends=(gendesk git imagemagick)
+source=("$pkgname"::git+https://github.com/drhelius/Gearsystem.git
+        https://raw.githubusercontent.com/gabomdq/SDL_GameControllerDB/master/gamecontrollerdb.txt)
+sha256sums=("SKIP"
+            "SKIP")
+
+pkgver() {
+  cd "$pkgname"
+  git describe --long --tags | sed 's/^gearsystem-//;s/\([^-]*\)-g.*/r\1/;s/-/./g'
+}
+
+prepare() {
+  gendesk -f -n --name='Gearsystem' --comment="${pkgdesc/.}" --custom='Keywords=game;engine;' --exec="$pkgname" --icon="$pkgname" --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname="$pkgname"
+  if [ ! -f "$pkgname.svg" ]; then
+    convert -fuzz 3.6% -fill transparent -draw 'color 0,0 floodfill' -transverse -fuzz 6.4% -draw 'color 0,0 floodfill' -transverse -gravity center -extent 768x768 -background none -thumbnail 256x256 -strip "$srcdir/$pkgname"/platforms/ios/Gearsystem/Images.xcassets/LaunchImage.launchimage/Default_1024.png "$pkgname".png
+    inkscape "$pkgname".png --export-plain-svg --export-filename="$pkgname".svg
+    rm "$pkgname".png
+  fi
+  make -C "$srcdir/$pkgname"/platforms/linux clean
+}
+
+build() {
+  make -C "$srcdir/$pkgname"/platforms/linux -j4
+}
+
+package() {
+  install -dm 755 "$pkgdir"/usr/{bin,share/{"$pkgname",applications,icons/hicolor/scalable/apps}}
+  install -m 755 "$srcdir/$pkgname"/platforms/linux/"$pkgname" "$pkgdir"/usr/share/"$pkgname/$pkgname"
+  install -m 644 gamecontrollerdb.txt "$pkgdir"/usr/share/"$pkgname"/
+  install -m 644 "$pkgname".desktop "$pkgdir"/usr/share/applications/
+  install -m 644 "$pkgname".svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  ln -rsf "$pkgdir"/usr/share/"$pkgname/$pkgname" "$pkgdir"/usr/bin/
+}
