diff a/iortcw/PKGBUILD b/iortcw/PKGBUILD
--- a/iortcw/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/iortcw/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,55 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=iortcw
+pkgver=1.51c.r129.g438e7d4
+pkgrel=1
+pkgdesc='ioQuake3 Return to Castle Wolfenstein.'
+arch=(x86_64)
+url=https://iortcw.org
+license=(GPL-2.0-only)
+depends=(freetype2 graphite harfbuzz hicolor-icon-theme
+         libjpeg-turbo libogg mesa openal opus opusfile
+         perl-rename pcre sdl2 xorg-fonts-misc zlib)
+makedepends=(gendesk git pcre)
+source=(
+  "$pkgname"::git+https://github.com/"$pkgname/$pkgname".git
+  rtcwsp.sh
+)
+provides=("$pkgname")
+conflicts=("$pkgname"-svn)
+sha256sums=('SKIP'
+            '8fa6e0e1c6b5bfbf2d87cf319983a333165be659fd020a85b90f84884a064098')
+
+pkgver() {
+  cd "$pkgname"
+  git describe --long --abbrev=7 --tags | sed 's/^v//;s/_/./;s/\([^-]*-g\)/r\1/;s/-/./g'
+}
+
+prepare() {
+  gendesk -f -n --name='ioRTCW SP' --comment="${pkgdesc/.}" --custom='Keywords=game;engine;' --exec="/usr/share/$pkgname/rtcwsp.sh" --icon="$pkgname" --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname=rtcwsp
+  gendesk -f -n --name='ioRTCW MP' --comment="${pkgdesc/.}" --custom='Keywords=game;engine;' --exec="/usr/share/$pkgname/rtcwmp.sh" --icon="$pkgname" --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname=rtcwmp
+  cd "$pkgname"
+  make -C MP clean
+  make -C SP clean
+  perl -pi -e 's/(?<= )(?=constexpr)/c_/gi' ?P/code/tools/lcc/src/*
+}
+
+build() {
+  cd "$pkgname"
+#  make -C SP -j$(nproc) USE_OPENAL_DLOPEN=0 USE_INTERNAL_LIBS=0
+#  make -C MP -j$(nproc) USE_OPENAL_DLOPEN=0 USE_INTERNAL_LIBS=0
+}
+
+package() {
+  install -dm 755 "$pkgdir"/usr/{bin,share/{"$pkgname"/main,applications,icons/hicolor/scalable/apps}}
+  make -C "$pkgname"/SP -j$(nproc) USE_OPENAL_DLOPEN=0 USE_INTERNAL_LIBS=0 COPYDIR="$pkgdir"/usr/share/"$pkgname" copyfiles
+  make -C "$pkgname"/MP -j$(nproc) USE_OPENAL_DLOPEN=0 USE_INTERNAL_LIBS=0 COPYDIR="$pkgdir"/usr/share/"$pkgname" copyfiles
+#  install -m 755 "$pkgname"/[MS]P/build/release-linux*/*iowolf*.x86_64 "$pkgdir"/usr/share/"$pkgname"/
+#  install -m 644 "$pkgname"/[MS]P/build/release-linux*/renderer*.so "$pkgdir"/usr/share/"$pkgname"/
+#  install -m 644 "$pkgname"/[MS]P/build/release-linux*/main/*[ms]p.*.so "$pkgdir"/usr/share/"$pkgname"/main/
+  install -m 644 rtcw*.desktop "$pkgdir"/usr/share/applications/
+  perl -pe 's/(?<=glnch=).*/"iowolfmp.x86_64"/i' rtcwsp.sh > rtcwmp.sh
+  install -m 755 rtcw*.sh "$pkgdir"/usr/share/"$pkgname"/
+  install -m 644 "$pkgname"/SP/misc/"$pkgname".svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  ln -rsf "$pkgdir"/usr/share/"$pkgname"/iowolf*.x86_64 "$pkgdir"/usr/bin/
+}
diff a/iortcw/rtcwsp.sh b/iortcw/rtcwsp.sh
--- a/iortcw/rtcwsp.sh	1969-12-31 17:00:00.000000000 -0700
+++ b/iortcw/rtcwsp.sh	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,72 @@
+#!/bin/bash
+gmdir="$HOME/games/rtcw"
+gmcfg="$HOME/.wolf"
+gmtgt="main/pak0.pk3"
+gmprm=""
+glnch="iowolfsp.x86_64"
+
+gconf() {
+mkdir -p "$gmcfg"
+printf 'gamepath=%s\ngamemod=%s\ngameparm=%s\n' "$gmdir" "main" "$gmprm" > "$gmcfg/basedir"
+}
+
+gstart () {
+while IFS= read -r -d '' file; do
+  [[ -d "$gmbdr/$file" ]] || mkdir -p "$gmbdr/$file"
+done < <(find /usr/share/iortcw -mindepth 1 -type d -printf '%P\0')
+while IFS= read -r -d '' file; do
+  [[ -h "$gmbdr/$file" ]] || ln -sf "/usr/share/iortcw/$file" "$gmbdr/$file"
+done < <(find /usr/share/iortcw -type f -iname '*.so' -printf '%P\0')
+echo "$glnch +set fs_basepath $gmbdr $gmgnm $gmpar" | xargs -i -r sh -c "{}"
+}
+
+gtest () {
+gmbse="" gmbdr="" gmgme="" gmgnm="" gmgpm="" gmpar=""
+
+if [[ -f "$gmcfg/basedir" ]]; then
+  while IFS='=' read -r key val || [[ -n "$key" ]]; do
+    case "$key" in
+      gamepath) gmbse="gamepath=$val"; gmbdr="$val" ;;
+      gamemod)  gmgme="gamemod=$val";  gmgnm="$val" ;;
+      gameparm) gmgpm="gameparm=$val"; gmpar="$val" ;;
+    esac
+  done < "$gmcfg/basedir"
+fi
+}
+
+[[ -f "$gmcfg/basedir" ]] || gconf
+gtest
+if [[ -z "$gmbdr" ]]; then
+  gconf
+  gtest
+fi
+if [[ -d "$gmbdr" ]]; then
+  if [[ ! -f "$gmbdr/$gmtgt" ]]; then
+    gmchk="$(find -H "$gmbdr" -mindepth 2 -maxdepth 2 -type f -regextype posix-extended -iregex "$gmbdr/$gmtgt" -printf '%p\n' | grep -Pic "$gmbdr/$gmtgt")"
+    if (( $gmchk == 1 )) ; then
+      echo -e "File system is case sensitive\nFile/folder names must be lowercase\n Rename subdirs of\n  $gmbdr\n    containing\n  $gmtgt?\nRequires perl-rename command" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      (( $? == 101 )) && find -H "$gmbdr" -depth -exec perl-rename 's|[^/]*\Z|\L$&|' {} +
+    elif (( $gmchk > 1 )) ; then
+      echo -e "File system is case sensitive\nFile/folder names must be lowercase\n Cannot rename subdirs of\n  $gmbdr\n    containing\n  $gmtgt\n\nBecause more than one exists" | xmessage -file - -buttons Ok:0 -center -timeout 30
+    else
+      echo -e "Config:\n $gmcfg/basedir\n  $gmbse\n found, but not\n  $gmgme\n ($gmbdr/$gmtgt)\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      (( $? == 101 )) && gconf
+    fi
+  else
+    if [[ ! -d "$gmbdr/$gmgnm" ]]; then
+      echo -e "Config:\n $gmcfg/basedir\n  $gmbse\n found, but not\n  $gmgme\n ($gmbdr/$gmgnm)\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      (( $? == 101 )) && gconf
+    else
+      if [[ -z "$(echo "$gmgnm" | grep -Pio '\w+')" ]]; then
+        gmgnm=""
+        gstart
+      else
+        gmgnm="+set fs_game $gmgnm"
+        gstart
+      fi
+    fi
+  fi
+else
+  echo -e "Config:\n $gmcfg/basedir\n  $gmbse\n not found\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+  (( $? == 101 )) && gconf
+fi
