diff a/kronos/PKGBUILD b/kronos/PKGBUILD
--- a/kronos/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/kronos/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,41 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=kronos
+_pkgname=yabause
+pkgver=2.3.1.r7072.48b93ceca
+pkgrel=1
+pkgdesc='Sega Saturn emulator'
+arch=(x86_64)
+url=https://fcare.github.io
+license=(GPL2)
+provides=("pkgname")
+conflicts=("pkgname"-git)
+depends=(openal qt5-base qt5-multimedia sdl2)
+makedepends=(cmake clang)
+source=("$pkgname"::git+https://github.com/FCare/Kronos)
+md5sums=('SKIP')
+
+pkgver() {
+  cd "$pkgname/$_pkgname"
+  _ver="$(cat CMakeLists.txt | grep -m3 "YAB_VERSION" | grep -o "[[:digit:]]*" | paste -sd'.')"
+  echo "${_ver}.r$(git rev-list --count HEAD).$(git rev-parse --short HEAD)"
+}
+
+prepare() {
+  cd "$pkgname/$_pkgname"
+  perl -pi -e 's| -march=native||gi' src/CMakeLists.txt
+  test ! -d build || rm -rf build
+}
+
+build() {
+  export CFLAGS="${CFLAGS// -Werror=format-security}"
+  export CXXFLAGS="$CFLAGS"
+  cmake -S "$pkgname/$_pkgname" -B build \
+    -DCMAKE_INSTALL_PREFIX=/usr \
+    -DYAB_USE_QT5=ON
+  cmake --build build -j$(nproc)
+}
+
+package() {
+  DESTDIR="$pkgdir" cmake --build build --target install
+}
