diff a/labelife-cups/PKGBUILD b/labelife-cups/PKGBUILD
--- a/labelife-cups/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/labelife-cups/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,19 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=labelife-cups
+pkgver=1.2.3
+pkgrel=1
+pkgdesc='CUPS printer driver for the labelife series'
+arch=(any)
+url=https://labelife.net
+license=(GPL)
+depends=(cups)
+source=(https://oss.qu-in.ltd/Labelife/Label_Printer_Driver_Linux.zip)
+sha256sums=(9f1817a5a14da53a95ffe7ba2840ecbf5ac4785b647c482f0d29d9e47545cdff)
+
+package() {
+  tar -xzvf "$srcdir"/*Driver*/*.tar.gz --strip-components=1 -C "$srcdir"
+  install -Dpm644 "$srcdir"/ppds/*.ppd* -t "$pkgdir"/usr/share/cups/model/Labelife
+  install -Dpm755 "$srcdir"/x86_64/rasterto* -t "$pkgdir"/usr/lib/cups/filter
+  perl -pi -e 's/(\*manufacturer: \").*/\1Labelife"/gi' "$pkgdir"/usr/share/cups/model/Labelife/*.ppd
+}
