diff a/memcardrex/PKGBUILD b/memcardrex/PKGBUILD
--- a/memcardrex/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/memcardrex/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,42 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=memcardrex
+pkgver=2.0.r40.gef794a0
+pkgrel=1
+pkgdesc="Advanced PlayStation 1 Memory Card editor."
+arch=(x86_64)
+url=https://github.com/ShendoXT/memcardrex
+license=(GPL-3.0-only)
+depends=(glibc hicolor-icon-theme zlib)
+makedepends=(dotnet-sdk gendesk git icoutils inkscape)
+provides=("$pkgname")
+conflicts=("$pkgname")
+replaces=("$pkgname")
+options=(!strip)
+source=("$pkgname"::git+https://github.com/ShendoXT/"$pkgname".git)
+sha256sums=('SKIP')
+
+pkgver() {
+  cd "$pkgname"
+  local _ver="$(grep -Pio "vers.*\b.*\d+\.[[:alnum:]\.]+" MemcardRex.Linux/GUI/AboutDialog.cs | grep -Pio '\d+\.[[:alnum:]\.]+')"
+  printf "%s.r%s.g%s" "$_ver" "$(git rev-list "$(git rev-list --tags --no-walk --max-count=1)"..HEAD --count)" "$(git rev-parse --short HEAD)"
+}
+
+prepare() {
+  gendesk -f -n --name='MemcardRex' --comment="${pkgdesc/.}" --custom='Keywords=game;console;' --exec="$pkgname %f" --icon="$pkgname" --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname="$pkgname"
+  rm -rf build "$pkgname"/MemcardRex.Linux/{bin,obj}
+}
+
+build() {
+  dotnet publish -f net8.0 -c Release -r linux-x64 --self-contained -p:PublishReadyToRun=true\;EnableCompressionInSingleFile=true "$pkgname"/MemcardRex.Linux -o "$srcdir"/build
+}
+
+package() {
+  install -dm 755 "$pkgdir"/{opt/"$pkgname",usr/{bin,share/{applications,icons/hicolor/scalable/apps}}}
+  install -m 644 "$pkgname".desktop "$pkgdir"/usr/share/applications/
+  install -m 644 "$pkgname"/MemcardRex.Linux/Resources/io.gitlab.robxnano."$pkgname".svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/"$pkgname".svg
+  install -m 755 build/"$pkgname" "$pkgdir"/opt/"$pkgname"/
+  install -m 644 build/*.so "$pkgdir"/opt/"$pkgname"/
+  strip --strip-unneeded "$pkgdir"/opt/"$pkgname"/*.so
+  ln -rsf "$pkgdir"/opt/"$pkgname/$pkgname" "$pkgdir"/usr/bin/
+}
