diff a/mgba/PKGBUILD b/mgba/PKGBUILD
--- a/mgba/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/mgba/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,75 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgbase=mgba
+pkgname=(lib"$pkgbase" "$pkgbase"-qt)
+pkgver=0.11.0.r1267.gb827eaf27
+pkgrel=1
+arch=(x86_64)
+url=http://mgba.io
+license=(MPL-2.0)
+makedepends=(cmake ffmpeg git json-c libedit libelf libepoxy libglvnd libpng
+             libzip lua ninja qt6-base qt6-multimedia qt6-tools sdl3 zlib)
+source=(git+https://github.com/"$pkgbase"-emu/"$pkgbase".git)
+sha1sums=('SKIP')
+
+pkgver() {
+  cd "$pkgbase"
+  local _ver="$(grep -Pio "ver.*_(ma|mi|pa).*\b.*\d+" version.cmake | grep -Pio '\d+$' | perl -pe 's|\s|.|' | perl -pe 's|(.*)\.$|\1|')"
+  printf "%s.r%s.g%s" "$_ver" "$(git rev-list "$(git rev-list --tags --no-walk --max-count=1)"..HEAD --count)" "$(git rev-parse --short HEAD)"
+}
+
+prepare() {
+  test ! -d "build" || rm -rf build
+  cd "$pkgbase"
+#  perl -pi -e 's|io\.mgba\.||gi' {,src/platform/qt/}CMakeLists.txt
+  export GITHEAD="$startdir/$pkgbase"
+  perl -pi -e 's|(?<=short HEAD WORKING_DIRECTORY \"\$)\{CMAKE_CURRENT_SOURCE_DIR|ENV{GITHEAD|gi ; s| --dirty||gi' version.cmake
+}
+
+build() {
+  cmake -S "$pkgbase" -G Ninja -B build \
+    -DCMAKE_INSTALL_LIBDIR=lib \
+    -DCMAKE_INSTALL_PREFIX=/usr \
+    -DUSE_MINIZIP=OFF \
+    -DCMAKE_BUILD_TYPE=Release
+  cmake --build build --config Release -j$(nproc)
+}
+
+package_libmgba() {
+  pkgdesc='mGBA Shared library.'
+  depends=(ffmpeg freetype2 json-c libedit libelf libepoxy libpng libzip lua sqlite zlib)
+  conflicts=("$pkgname"-git)
+  provides=("$pkgname")
+
+  cmake -DCOMPONENT="$pkgname" "$pkgbase" -DCMAKE_INSTALL_PREFIX="$pkgdir"/usr \
+    -P build/cmake_install.cmake
+  cmake -DCOMPONENT="$pkgbase"-dev "$pkgbase" -DCMAKE_INSTALL_PREFIX="$pkgdir"/usr \
+    -P build/cmake_install.cmake
+  install -Dm644 "$pkgbase"/LICENSE "$pkgdir"/usr/share/licenses/"$pkgname"/LICENSE
+}
+
+package_mgba-sdl() {
+  pkgdesc='Gameboy Advance Emulator focused on speed and accuracy.'
+  depends=(lib"$pkgbase" sdl3)
+  conflicts=("$pkgname"-git)
+  provides=("$pkgname")
+
+  cmake -DCOMPONENT="$pkgname" "$pkgbase" -DCMAKE_INSTALL_PREFIX="$pkgdir"/usr \
+    -P build/cmake_install.cmake
+  install -d "$pkgdir"/usr/share/licenses/"$pkgname"
+  ln -s /usr/share/licenses/lib"$pkgbase"/LICENSE "$pkgdir"/usr/share/licenses/"$pkgname"/LICENSE
+}
+
+package_mgba-qt() {
+  pkgdesc='Qt6 UI Gameboy Advance Emulator focused on speed and accuracy.'
+  depends=(hicolor-icon-theme ffmpeg libglvnd lib"$pkgbase" qt6-base qt6-multimedia sdl3)
+  conflicts=("$pkgname"-git)
+  provides=("$pkgname")
+
+  cmake -DCOMPONENT="$pkgname" "$pkgbase" -DCMAKE_INSTALL_PREFIX="$pkgdir"/usr \
+    -P build/cmake_install.cmake
+  cmake -DCOMPONENT="$pkgbase" "$pkgbase" -DCMAKE_INSTALL_PREFIX="$pkgdir"/usr \
+    -P build/cmake_install.cmake
+  install -d "$pkgdir"/usr/share/licenses/"$pkgname"
+  ln -s /usr/share/licenses/lib"$pkgbase"/LICENSE "$pkgdir"/usr/share/licenses/"$pkgname"/LICENSE
+}
