diff a/mhvtl/PKGBUILD b/mhvtl/PKGBUILD
--- a/mhvtl/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/mhvtl/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,55 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgbase=mhvtl
+pkgname=("$pkgbase" "$pkgbase"-dkms)
+pkgver=1.7.2
+pkgrel=1
+pkgdesc='A Virtual Tape & Library system'
+arch=(x86_64)
+url=http://sites.google.com/site/linuxvtl2
+license=(GPL-2.0-or-later)
+makedepends=(git)
+source=(git+https://github.com/markh794/mhvtl.git
+        dkms.conf)
+sha256sums=('SKIP'
+            '3b9288cdd6ab82aa34657e1e74bffd0b7778f4e8f475cdf6ae9e44a5b0e34999')
+
+pkgver() {
+  cd "$pkgbase"
+  git describe --long --abbrev=7 --tags | sed 's/^//;s/_release//;s/_/./;s/\([^-]*-g\)/r\1/;s/-/./g'
+}
+
+prepare() {
+  cd "$pkgbase"
+  perl -pi -e 's/YES/NO/g' Makefile
+  perl -pi -e 's|(?<=\?\= )(/lib)|/usr\1|gi' config.mk
+  make clean
+}
+
+build() {
+  cd "$pkgbase"
+  make -j4
+}
+
+package_mhvtl() {
+  depends=("$pkgname"-dkms bash glibc lsscsi lzo mtx perl sg3_utils zlib)
+  provides=("$pkgname")
+
+  cd "$pkgname"
+  make DESTDIR="$pkgdir" install
+  chmod 755 "$pkgdir"/{usr/bin/*.conf,opt/"$pkgname"/*}
+  chmod 644 "$pkgdir"/opt/"$pkgname"/*/*
+}
+package_mhvtl-dkms() {
+  depends=(dkms linux-headers)
+  provides=("$pkgname")
+
+  cd "$pkgbase"
+  install -Dm 644 kernel/{Makefile,*.{c,h}} -t "$pkgdir"/usr/src/"$pkgname"
+  install -Dm 755 kernel/config.sh "$pkgdir"/usr/src/"$pkgname"/
+  install -Dm 644 "$startdir"/dkms.conf "$pkgdir"/usr/src/"$pkgname"/
+
+  sed -e "s/@PKGNAME@/$pkgbase/" \
+      -e "s/@PKGVER@/$pkgver/" \
+      -i "$pkgdir/usr/src/$pkgname/dkms.conf"
+}
+diff a/mhvtl/dkms.conf b/mhvtl/dkms.conf
--- a/mhvtl/dkms.conf	1969-12-31 17:00:00.000000000 -0700
+++ b/mhvtl/dkms.conf	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,7 @@
+PACKAGE_NAME="@PKGNAME@"
+PACKAGE_VERSION="@PKGVER@"
+MAKE[0]="make V=$kernelver"
+CLEAN="make clean"
+BUILT_MODULE_NAME[0]="@PKGNAME@"
+DEST_MODULE_LOCATION[0]="/kernel/drivers/scsi"
+AUTOINSTALL="yes"
