diff a/mhvtl/PKGBUILD b/mhvtl/PKGBUILD
--- a/mhvtl/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/mhvtl/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,40 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=mhvtl
+pkgver=1.7.2
+pkgrel=1
+pkgdesc='A Virtual Tape & Library system'
+arch=(x86_64)
+url=https://github.com/dymosoftwarehttp://sites.google.com/site/linuxvtl2
+license=(GPL-2.0-or-later)
+depends=(bash glibc lsscsi lzo mtx perl sg3_utils zlib)
+makedepends=(git)
+source=(git+https://github.com/markh794/mhvtl.git)
+sha256sums=('SKIP')
+
+pkgver() {
+  cd "$pkgname"
+  git describe --long --abbrev=7 --tags | sed 's/^//;s/_release//;s/_/./;s/\([^-]*-g\)/r\1/;s/-/./g'
+}
+
+prepare() {
+  cd "$pkgname"
+  perl -pi -e 's/YES/NO/g' Makefile
+  perl -pi -e 's|(?<=\?\= )(/lib)|/usr\1|gi' config.mk
+  make clean
+}
+
+build() {
+  cd "$pkgname"
+  make -j4
+  make kernel
+}
+
+package() {
+  cd "$pkgname"
+  make DESTDIR="$pkgdir" install
+  chmod 755 "$pkgdir"/{usr/bin/*.conf,opt/"$pkgname"/*}
+  chmod 644 "$pkgdir"/opt/"$pkgname"/*/*
+  install -Dm 644 kernel/"$pkgname".ko -t "$pkgdir"/usr/lib/modules/"$(uname -r)"/extramodules/
+  zstd --rm -19 "$pkgdir"/usr/lib/modules/"$(uname -r)"/extramodules/"$pkgname".ko
+}
