diff a/mupen64plus/PKGBUILD b/mupen64plus/PKGBUILD
--- a/mupen64plus/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/mupen64plus/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,109 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=mupen64plus
+pkgver=20221229.r0.gba9a5248
+pkgrel=1
+pkgdesc="A cross-platform plugin-based N64 emulator."
+arch=(x86_64)
+url=http://www.mupen64plus.org
+license=(GPL2)
+depends=(boost-libs desktop-file-utils freetype2 glew glu
+         hicolor-icon-theme libgl libpng libsamplerate minizip
+         python-pyqt5 python-pysdl2 qt5-base sdl2 speexdsp zlib)
+makedepends=(boost git mesa nasm)
+optdepends=(python-opengl python-opengl-accelerate python-pylzma python-rarfile)
+provides=("$pkgname")
+conflicts=(gom64p m64py m64py-git "$pkgname"-audio-sdl2 "$pkgname"-extraplugins "$pkgname"-extraplugins-git "$pkgname"-git "$pkgname"-input-raphnetraw "$pkgname"-rsp-cxd4-git "$pkgname"-rsp-parallel-git "$pkgname"-video-angrylion-plus-git "$pkgname"-video-gliden64 "$pkgname"-video-gliden64-git "$pkgname"-video-parallel-git mupengui-git simple64 simple64-git)
+replaces=(gom64p m64py m64py-git "$pkgname"-audio-sdl2 "$pkgname"-extraplugins "$pkgname"-extraplugins-git "$pkgname"-git "$pkgname"-input-raphnetraw "$pkgname"-rsp-cxd4-git "$pkgname"-rsp-parallel-git "$pkgname"-video-angrylion-plus-git "$pkgname"-video-gliden64 "$pkgname"-video-gliden64-git "$pkgname"-video-parallel-git mupengui-git simple64 simple64-git)
+source=(
+  git+https://github.com/"$pkgname/$pkgname"-core.git
+  git+https://github.com/"$pkgname/$pkgname"-video-glide64mk2.git
+  git+https://github.com/"$pkgname/$pkgname"-video-z64.git
+  git+https://github.com/"$pkgname/$pkgname"-rsp-hle.git
+  git+https://github.com/"$pkgname/$pkgname"-rsp-cxd4.git
+  git+https://github.com/"$pkgname/$pkgname"-ui-console.git
+  git+https://github.com/"$pkgname/$pkgname"-video-glide64.git
+  git+https://github.com/"$pkgname/$pkgname"-input-sdl.git
+  git+https://github.com/"$pkgname/$pkgname"-ui-python.git
+  git+https://github.com/"$pkgname/$pkgname"-video-rice.git
+  git+https://github.com/"$pkgname/$pkgname"-audio-sdl.git
+  git+https://github.com/"$pkgname/$pkgname"-video-arachnoid.git
+  git+https://github.com/"$pkgname/$pkgname"-rsp-z64.git
+  git+https://github.com/ata4/angrylion-rdp-plus.git
+  git+https://github.com/Mastergatto/gom64p.git
+  git+https://github.com/Mastergatto/parallel-rsp.git#branch=m64p
+  parallel-rdp::git+https://github.com/Mastergatto/parallel-rdp-standalone.git#branch=m64p
+  git+https://github.com/raphnet/mupen64plus-input-raphnetraw.git
+  mupen64.patch
+)
+sha256sums=('SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'a6272495f17e59d3063c03564a8e4c0970b3c8afc2c7fe7441f9a93b1785df34')
+
+_subs='audio-sdl core input-raphnetraw input-sdl rsp-cxd4 rsp-hle rsp-z64 ui-console video-arachnoid video-glide64 video-glide64mk2 video-rice video-z64'
+_exsubs='angrylion-rdp-plus parallel-rdp parallel-rsp'
+
+pkgver() {
+  cd "$pkgname"-core
+#  git describe --long --abbrev=7 --tags | sed 's/^v//;s/\([^-]*-g\)/r\1/;s/-/./g'
+  printf "%s.r%s.g%s" "$(date +%Y%m%d)" "$(git rev-list "$(git rev-list --tags --no-walk --max-count=1)"..HEAD --count)" "$(git rev-parse --short HEAD)"
+}
+
+prepare() {
+  patch -Np1 -i mupen64.patch
+  perl -pi -e 's|(include_directories).*(mupen64plus).*|\1(/usr/include/\2)|gi' parallel-r*/CMakeLists.txt
+  perl -pi -e 's|.*include.*config.h.*\n||gi' "$pkgname"-input-raphnetraw/src/plugin_front.c
+  perl -pi -e 's| -march=x86-64[^ )]+||gi' parallel-rsp/CMakeLists.txt
+  for _sub in $_subs; do
+    make -C "$pkgname-$_sub"/projects/unix clean
+  done
+  for _sub in $_exsubs gom64p; do
+    if [[ -d "$_sub/build" ]]; then
+      rm -rf "$_sub"/build
+    fi
+  done
+  cd "$pkgname"-ui-python
+  python setup.py clean
+}
+
+build() {
+  for _sub in $_exsubs; do
+    cmake -S "$_sub" -B "$_sub"/build \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DCMAKE_INSTALL_PREFIX=/usr
+    cmake --build "$_sub"/build -j$(nproc)
+  done
+  for _sub in $_subs; do
+    make -C "$pkgname-$_sub"/projects/unix -j$(nproc) PIE=1 PREFIX=/usr all
+  done
+#  arch-meson gom64p gom64p/build
+#  meson compile -C gom64p/build
+}
+
+package() {
+  for _sub in $_subs; do
+    make -C "$pkgname-$_sub"/projects/unix PREFIX="$pkgdir"/usr install
+  done
+  install -m 644 angrylion-rdp-plus/build/*.so "$pkgdir"/usr/lib/"$pkgname"/
+  install -m 644 parallel-r?p/build/*.so "$pkgdir"/usr/lib/"$pkgname"/
+#  DESTDIR="$pkgdir" meson install -C gom64p/build
+  cd "$pkgname"-ui-python
+  python setup.py install --optimize=1 --prefix=/usr --root="$pkgdir"
+  ln -rsf "$(find "$pkgdir"/usr/lib -type f -iname lib"$pkgname".*)" "$pkgdir"/usr/lib/lib"$pkgname".so
+}
diff a/mupen64plus/mupen64.patch b/mupen64plus/mupen64.patch
index a593ca4..0e73e2e 100644
--- a/mupen64plus/mupen64.patch	1969-12-31 17:00:00.000000000 -0700
+++ b/mupen64plus/mupen64.patch	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,66 @@
+diff a/mupen64plus-ui-python/src/m64py/core/defs.py b/mupen64plus-ui-python/src/m64py/core/defs.py
+index b202fac..5169331 100644
+--- a/mupen64plus-ui-python/src/m64py/core/defs.py	1969-12-31 17:00:00.000000000 -0700
++++ b/mupen64plus-ui-python/src/m64py/core/defs.py	1969-12-31 17:00:00.000000000 -0700
+@@ -196,7 +196,10 @@ class m64p_rom_settings(C.Structure):
+         ('rumble', C.c_ubyte),
+         ('transferpak', C.c_ubyte),
+         ('mempak', C.c_ubyte),
+-        ('biopak', C.c_ubyte)
++        ('biopak', C.c_ubyte),
++        ('disableextramem', C.c_ubyte),
++        ('countperop', C.c_uint),
++        ('sidmaduration', C.c_uint),
+     ]
+ 
+ 
+diff a/mupen64plus-ui-python/src/m64py/frontend/romreader.py b/mupen64plus-ui-python/src/m64py/frontend/romreader.py
+index 84de37a..62dde4d 100644
+--- a/mupen64plus-ui-python/src/m64py/frontend/romreader.py	1969-12-31 17:00:00.000000000 -0700
++++ b/mupen64plus-ui-python/src/m64py/frontend/romreader.py	1969-12-31 17:00:00.000000000 -0700
+@@ -18,6 +18,8 @@
+ import ctypes
+ import fnmatch
+ 
++from pathlib import Path
++
+ from PyQt5.QtCore import QThread
+ 
+ from m64py.utils import sl
+@@ -48,10 +50,8 @@ def get_files(self):
+         """Returns list of files found in path."""
+         files = []
+         types = EXT_FILTER.split()
+-        for filename in os.listdir(self.rom_path):
+-            for ext in types:
+-                if fnmatch.fnmatch(filename, ext):
+-                    files.append(filename)
++        rom_path = Path(self.rom_path)
++        files = [file for type in types for file in rom_path.rglob(type)]
+         return files
+ 
+     def get_rom_crc(self, archive, fname):
+@@ -106,8 +106,8 @@ def read_files(self):
+         self.roms = []
+         files = self.get_files()
+         num_files = len(files)
+-        for filenum, filename in enumerate(files):
+-            fullpath = os.path.join(self.rom_path, filename)
++        for filenum, path in enumerate(files):
++            fullpath = str(path)
+             try:
+                 archive = Archive(fullpath)
+                 for fname in archive.namelist:
+@@ -115,9 +115,11 @@ def read_files(self):
+                     if crc_tuple:
+                         rom_settings = self.parent.core.get_rom_settings(
+                             crc_tuple[0], crc_tuple[1])
++                        crc = "%X%X" % (crc_tuple[0], crc_tuple[1])
+                         if rom_settings:
+-                            crc = "%X%X" % (crc_tuple[0], crc_tuple[1])
+                             self.roms.append((crc, rom_settings.goodname, fullpath, fname))
++                        else:
++                            self.roms.append((crc, path.name, fullpath, fname))
+                 archive.close()
+             except Exception as err:
+                 log.warn(str(err))
