diff a/opentesarena/PKGBUILD b/opentesarena/PKGBUILD
--- a/opentesarena/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/opentesarena/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,51 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=opentesarena
+pkgver=0.14.0.r455.1599053
+pkgrel=1
+pkgdesc='Open source Elder Scrolls Arena.'
+arch=(x86_64)
+url=https://github.com/afritz1/OpenTESArena
+license=(MIT)
+depends=(freepats-legacy hicolor-icon-theme openal
+         perl-rename sdl2 wildmidi xorg-fonts-misc)
+makedepends=(cmake gendesk inkscape)
+source=("$pkgname"::git+https://github.com/afritz1/OpenTESArena.git#branch=main
+        https://raw.githubusercontent.com/noabody/unibuild/master/data/"$pkgname".svg
+        "$pkgname".sh)
+provides=("$pkgname")
+replaces=("$pkgname")
+sha256sums=('SKIP'
+            '56c333d8c7225ebe71fc90868e1b4683d8a3340e74995c67ef2f0405948865ea'
+            'a8355230123ed78f577eb41f1feadfe3240bcaccc2e23326c56d4d4ae11419ea')
+
+pkgver() {
+  cd "$pkgname"
+  git describe --abbrev=7 --tags | sed 's/^opentesarena-//; s/-/.r/; s/-g/./'
+}
+
+prepare() {
+  gendesk -f -n --name='OpenTESArena' --comment="${pkgdesc/.}" --custom='Keywords=game;engine;' --exec="/usr/share/$pkgname/$pkgname.sh" --icon="$pkgname" --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname="$pkgname"
+  test ! -d build || rm -rf build
+  perl -pi -e 's/(\#include \<vector\>)/\#include <cstdint>\n\1/g' "$pkgname"/components/utilities/VirtualHeap.h
+}
+
+build() {
+  cmake -S "$pkgname" -B build \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_INSTALL_PREFIX=/usr
+  cmake --build build -j$(nproc)
+}
+
+package() {
+  install -dm 755 "$pkgdir"/usr/{bin,share/{"$pkgname",applications,icons/hicolor/scalable/apps}}
+#  DESTDIR="$pkgdir" cmake --build build --target install 
+  install -m 755 build/otesa "$pkgdir"/usr/share/"$pkgname/$pkgname"
+  cp -dr --no-preserve=ownership build/OpenTESArena/data "$pkgdir"/usr/share/"$pkgname"/
+  cp -dr --no-preserve=ownership build/OpenTESArena/options "$pkgdir"/usr/share/"$pkgname"/
+  install -m 755 "$pkgname".sh "$pkgdir"/usr/share/"$pkgname"/
+  install -m 644 "$pkgname".desktop "$pkgdir"/usr/share/applications/
+  install -m 644 "$pkgname".svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  ln -rsf "$pkgdir"/usr/share/"$pkgname/$pkgname" "$pkgdir"/usr/bin/
+  perl -pi -e 's|^(MidiConfig=).*|\1/usr/share/freepats/wildmidi-freepats.cfg|gi' "$pkgdir"/usr/share/"$pkgname"/options/options-default.txt
+}
diff a/opentesarena/opentesarena.sh b/opentesarena/opentesarena.sh
--- a/opentesarena/opentesarena.sh	1969-12-31 17:00:00.000000000 -0700
+++ b/opentesarena/opentesarena.sh	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,65 @@
+#!/bin/bash
+gmdir="$HOME/games/arena"
+gmcfg="$HOME/.config/OpenTESArena"
+gmtgt="GLOBAL.BSA"
+gmprm="options/options-changes.txt"
+glnch="opentesarena"
+
+gconf () {
+test -d "$gmcfg" || mkdir -p "$gmcfg/options"
+echo -e "gamepath= $gmdir\ngamemod= \ngameparm= $gmprm" | tee "$gmcfg/basedir"
+}
+
+gstart () {
+if [ -f "$gmcfg/$gmpar" ]; then
+  gmpth="$(grep -Pio "(?<=ArenaPath=).*" "$gmcfg/$gmpar")"
+  gmsav="$(grep -Pio "(?<=ArenaSavesPath=).*" "$gmcfg/$gmpar")"
+  if [ -n "$gmpth" ] && [ "$gmpth" != "$gmbdr" ]; then
+    perl -pi -e "s|(ArenaPath)=.*|\\1=$gmbdr|gi" "$gmcfg/$gmpar"
+  fi
+  if [ -n "$gmsav" ] && [ "$gmsav" != "$gmbdr/saves" ]; then
+    perl -pi -e "s|(ArenaSavesPath)=.*|\\1=$gmbdr/saves|gi" "$gmcfg/$gmpar"
+  fi
+else
+#  echo -e "[Misc]\nArenaPath=$gmbdr\nArenaSavesPath=$gmbdr/saves" | tee -a "$gmcfg/$gmpar"
+  cp /usr/share/opentesarena/options/options-default.txt "$gmcfg/$gmpar"
+  perl -pi -e "s|(ArenaPath)=.*|\\1=$gmbdr|gi" "$gmcfg/$gmpar"
+  perl -pi -e "s|(ArenaSavesPath)=.*|\\1=$gmbdr/saves|gi" "$gmcfg/$gmpar"
+fi
+echo "$glnch" | xargs -i -r sh -c "{}"
+}
+
+gtest () {
+gmbse="$(grep -Pi '^gamepath=.*' "$gmcfg/basedir" | head -1)"
+gmgme="$(grep -Pi '^gamemod=.*' "$gmcfg/basedir" | head -1)"
+gmgpm="$(grep -Pi '^gameparm=.*' "$gmcfg/basedir" | head -1)"
+gmbdr="${gmbse#gamepath= }"
+gmgnm="${gmgme#gamemod= }"
+gmpar="${gmgpm#gameparm= }"
+}
+
+test -f "$gmcfg/basedir" || gconf
+gtest
+if [ -z "$gmbdr" ]; then
+  gconf
+  gtest
+fi
+if [ -d "$gmbdr" ]; then
+  if [ ! -f "$gmbdr/$gmtgt" ]; then
+    gmchk="$(find -H "$gmbdr" -mindepth 1 -maxdepth 1 -type f -regextype posix-extended -iregex "$gmbdr/$gmtgt" -printf '%p\n' | grep -Pic "$gmbdr/$gmtgt")"
+    if [ $gmchk -eq 1 ]; then
+      echo -e "File system is case sensitive\nFile/folder names must be uppercase\n Rename subdirs of\n  $gmbdr\n    containing\n  $gmtgt?\nRequires perl-rename command" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      test $? -eq 101 && find -H "$gmbdr" -depth -exec perl-rename 's|([^/]*\Z)|uc($1)|e' {} +
+    elif [ $gmchk -gt 1 ]; then
+      echo -e "File system is case sensitive\nFile/folder names must be uppercase\n Cannot rename subdirs of\n  $gmbdr\n    containing\n  $gmtgt\n\nBecause more than one exists" | xmessage -file - -buttons Ok:0 -center -timeout 30
+    else
+      echo -e "Config:\n $gmcfg/basedir\n  $gmbse\n found, but not\n ($gmbdr/$gmtgt)\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      test $? -eq 101 && gconf
+    fi
+  else
+    gstart
+  fi
+else
+  echo -e "Config:\n $gmcfg/basedir\n  $gmbse\n not found\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+  test $? -eq 101 && gconf
+fi
