diff a/pcsx2/PKGBUILD b/pcsx2/PKGBUILD
--- a/pcsx2/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/pcsx2/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,61 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=pcsx2
+pkgver=1.7.0.r1528.cd33a1758
+pkgrel=1
+pkgdesc='A Sony PlayStation 2 emulator'
+arch=(x86_64)
+url=https://www.pcsx2.net
+license=(GPL2 GPL3 LGPL2.1 LGPL3)
+depends=(libaio libjpeg-turbo libpcap libpulse libsamplerate portaudio sdl2 soundtouch wxgtk3 wayland rapidyaml-git)
+makedepends=(cmake git ninja python swig xorgproto)
+source=(git+https://github.com/PCSX2/pcsx2.git
+        git+https://github.com/mozilla/cubeb.git
+        git+https://github.com/fmtlib/fmt.git
+        git+https://github.com/KhronosGroup/glslang.git
+        git+https://github.com/google/googletest.git
+        git+https://github.com/ocornut/imgui.git
+        git+https://github.com/rtissera/libchdr.git
+        vulkan-headers::git+https://github.com/KhronosGroup/Vulkan-Headers.git)
+b2sums=(SKIP
+        SKIP
+        SKIP
+        SKIP
+        SKIP
+        SKIP
+        SKIP
+        SKIP)
+
+pkgver() {
+  cd "$pkgname"
+  git describe --tags | sed 's/^v//; s/-dev//; s/-/.r/; s/-g/./'
+}
+
+prepare() {
+  test ! -d "$pkgname/build" || rm -rf "$pkgname"/build
+  cd "$pkgname"/3rdparty
+  for submodule in cubeb/cubeb fmt/fmt glslang/glslang gtest imgui/imgui libchdr/libchdr vulkan-headers; do
+    git submodule init ${submodule}
+    git config submodule.${submodule}.url ../../${submodule#*/}
+    git submodule update ${submodule}
+  done
+}
+
+build() {
+  cmake -S "$pkgname" -B "$pkgname"/build -G Ninja \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_INSTALL_PREFIX=/usr \
+    -DUSE_SYSTEM_YAML=TRUE \
+    -DCMAKE_INTERPROCEDURAL_OPTIMIZATION=ON \
+    -DWAYLAND_API=ON \
+    -DPACKAGE_MODE=ON \
+    -DXDG_STD=TRUE \
+    -DDISABLE_SETCAP=ON \
+    -DCMAKE_SKIP_RPATH=ON \
+    -Wno-dev
+  DESTDIR="$pkgdir" ninja -C "$pkgname"/build -j$(nproc)
+}
+
+package() {
+  DESTDIR="$pkgdir" cmake --install "$pkgname"/build
+}
