diff a/pcsx2/PKGBUILD b/pcsx2/PKGBUILD
--- a/pcsx2/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/pcsx2/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,71 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=pcsx2
+pkgver=1.7.0.r4236.gcca2c38d8
+pkgrel=1
+epoch=1
+pkgdesc='A Sony PlayStation 2 emulator'
+arch=(x86_64)
+url=https://www.pcsx2.net
+license=(GPL2 GPL3 LGPL2.1 LGPL3)
+depends=(fmt libaio libgl libpcap libpulse libzip qt6-base
+         rapidyaml sdl2 soundtouch wayland xorg-xrandr zstd)
+makedepends=(cmake git ninja python swig xorgproto)
+provides=(pcsx2-qt)
+source=(git+https://github.com/kenshen112/pcsx2.git#branch=packages
+        git+https://github.com/mozilla/cubeb.git
+        git+https://github.com/KhronosGroup/glslang.git
+        git+https://github.com/ocornut/imgui.git
+        git+https://github.com/rtissera/libchdr.git
+        vulkan-headers::git+https://github.com/KhronosGroup/Vulkan-Headers.git
+        "$pkgname".patch)
+sha256sums=(SKIP
+            SKIP
+            SKIP
+            SKIP
+            SKIP
+            SKIP
+            'ee4d2475d0752b4e7465936aa9c111b34338fa1765afd7e76b1d8ad158e902b2')
+
+pkgver() {
+  cd "$pkgname"
+  git describe --long --tags | sed 's/^v//;s/-dev//;s/_/./;s/\([^-dev]*-g\)/r\1/;s/-/./g'
+}
+
+prepare() {
+#  patch -Np1 -i "$pkgname".patch
+  test ! -d "$pkgname/build" || rm -rf "$pkgname"/build
+  cd "$pkgname"/3rdparty
+  for submodule in cubeb/cubeb glslang/glslang imgui/imgui libchdr/libchdr vulkan-headers; do
+    git submodule init ${submodule}
+    git config submodule.${submodule}.url ../../${submodule#*/}
+    git submodule update ${submodule}
+  done
+}
+
+build() {
+  cmake -S "$pkgname" -B "$pkgname"/build -G Ninja \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_INSTALL_PREFIX=/usr \
+    -DCMAKE_INTERPROCEDURAL_OPTIMIZATION=ON \
+    -DDISABLE_ADVANCE_SIMD=ON \
+    -DDISABLE_BUILD_DATE=ON \
+    -DDISABLE_PCSX2_WRAPPER=ON \
+    -DDISABLE_SETCAP=ON \
+    -DENABLE_TESTS=OFF \
+    -DPACKAGE_MODE=ON \
+    -DQT_BUILD=ON \
+    -DUSE_SYSTEM_LIBS=ON \
+    -DUSE_VTUNE=OFF \
+    -DWAYLAND_API=ON \
+    -DX11_API=ON \
+    -DXDG_STD=ON \
+    -DCMAKE_SKIP_RPATH=ON \
+    -Wno-dev
+  ninja -C "$pkgname"/build -j$(nproc)
+}
+
+package() {
+  DESTDIR="$pkgdir" cmake --install "$pkgname"/build
+#  sed -i 's/Exec=env GDK_BACKEND=x11 MESA_NO_ERROR=1 pcsx2/Exec=env QT_QPA_PLATFORM=xcb MESA_NO_ERROR=1 pcsx2-qt/g' "$pkgdir"/usr/share/applications/PCSX2.desktop
+}
diff a/pcsx2/pcsx2.patch b/pcsx2/pcsx2.patch
index e69de29..feb733a 100644
--- a/pcsx2/pcsx2.patch	1969-12-31 17:00:00.000000000 -0700
+++ b/pcsx2/pcsx2.patch	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,21 @@
+diff a/pcsx2/pcsx2-qt/QtHost.cpp b/pcsx2/pcsx2-qt/QtHost.cpp
+index e7e1805c2..e862e8145 100644
+--- a/pcsx2/pcsx2-qt/QtHost.cpp	1969-12-31 17:00:00.000000000 -0700
++++ b/pcsx2/pcsx2-qt/QtHost.cpp	1969-12-31 17:00:00.000000000 -0700
+@@ -134,12 +134,14 @@ void QtHost::SetAppRoot()
+ 
+ void QtHost::SetResourcesDirectory()
+ {
+-#ifndef __APPLE__
++#ifdef __APPLE__
++	// On macOS, this is in the bundle resources directory.
++	EmuFolders::Resources = Path::Canonicalize(Path::Combine(EmuFolders::AppRoot, "../Resources"));
++#elif !defined(PCSX2_APP_DATADIR)
+ 	// On Windows/Linux, these are in the binary directory.
+ 	EmuFolders::Resources = Path::Combine(EmuFolders::AppRoot, "resources");
+ #else
+-	// On macOS, this is in the bundle resources directory.
+-	EmuFolders::Resources = Path::Canonicalize(Path::Combine(EmuFolders::AppRoot, "../Resources"));
++	EmuFolders::Resources = Path::Canonicalize(Path::Combine(EmuFolders::AppRoot, PCSX2_APP_DATADIR "/resources"));
+ #endif
+ }
