diff a/pkg2zip/PKGBUILD b/pkg2zip/PKGBUILD
--- a/pkg2zip/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/pkg2zip/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,38 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=pkg2zip
+pkgver=2.6.r2.g88fe198
+pkgrel=1
+arch=(x86_64)
+pkgdesc='Decrypts PlayStation Vita pkg file and packages to zip archive.'
+url=https://github.com/lusid1/pkg2zip
+license=(Unlicense)
+depends=(glibc python3)
+conflicts=("$pkgname"-fork "$pkgname"-git)
+source=(git+https://github.com/lusid1/"$pkgname".git)
+sha256sums=('SKIP')
+
+pkgver() {
+  cd "$pkgname"
+  git describe --long --tags --abbrev=7 | sed 's/\([^-]*-g\)/r\1/;s/-/./g'
+}
+
+prepare() {
+  cd "$pkgname"
+  make clean
+}
+
+build() {
+  cd "$pkgname"
+  make -j$(nproc) CFLAGS="${CFLAGS} -Wno-format-truncation"
+}
+
+package() {
+  cd "$pkgname"
+  install -dm 755 "$pkgdir"/usr/{bin,share/"$pkgname"}
+  install -Dm644 LICENSE "$pkgdir"/usr/share/licenses/"$pkgname"/LICENSE
+  install -Dm755 {"$pkgname",*rif.py} "$pkgdir"/usr/share/"$pkgname"
+  ln -rsf "$pkgdir"/usr/share/"$pkgname/$pkgname" "$pkgdir"/usr/bin/
+  ln -rsf "$pkgdir"/usr/share/"$pkgname"/zrif2rif.py "$pkgdir"/usr/bin/
+  ln -rsf "$pkgdir"/usr/share/"$pkgname"/rif2zrif.py "$pkgdir"/usr/bin/
+}
