diff a/pkgdecrypt/PKGBUILD b/pkgdecrypt/PKGBUILD
--- a/pkgdecrypt/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/pkgdecrypt/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,32 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=pkgdecrypt
+pkgver=1.1.0
+pkgrel=1
+arch=(x86_64)
+pkgdesc='Decrypts and extracts PS Vita PKG files.'
+url=https://github.com/St4rk/PkgDecrypt
+license=(MIT)
+depends=(glibc zlib)
+source=("$pkgname"::git+https://github.com/st4rk/PkgDecrypt.git)
+sha256sums=('SKIP')
+
+prepare() {
+  cd "$pkgname"
+  make clean
+}
+
+build() {
+  cd "$pkgname"
+  make -j$(nproc) deps
+  make -j$(nproc) pkg_dec make_key
+}
+
+package() {
+  cd "$pkgname"
+  install -dm 755 "$pkgdir"/usr/{bin,share/"$pkgname"}
+  install -Dm644 LICENSE "$pkgdir"/usr/share/licenses/"$pkgname"/LICENSE
+  install -Dm755 {pkg_dec,make_key} -t "$pkgdir"/usr/share/"$pkgname"
+  ln -rsf "$pkgdir"/usr/share/"$pkgname"/pkg_dec "$pkgdir"/usr/bin/
+  ln -rsf "$pkgdir"/usr/share/"$pkgname"/make_key "$pkgdir"/usr/bin/
+}
