diff a/ppsspp/PKGBUILD b/ppsspp/PKGBUILD
index e69de29..a84929e 100644
--- a/ppsspp/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/ppsspp/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,154 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgbase=ppsspp
+pkgname=("$pkgbase")
+pkgver=1.14.4.r455.c9e386f78e
+pkgrel=1
+pkgdesc='A PSP emulator written in C++'
+arch=(x86_64)
+url=http://www.ppsspp.org
+license=(GPL2)
+depends=(glew glibc hicolor-icon-theme libgl libzip sdl2 snappy zlib)
+makedepends=(clang cmake git libglvnd libzip python qt5-tools qt5-multimedia)
+provides=(ppsspp-headless ppsspp-qt)
+conflicts=(ppsspp-headless ppsspp-qt)
+replaces=(ppsspp-headless ppsspp-qt)
+source=(
+  git+https://github.com/hrydgard/"$pkgbase".git
+  git+https://github.com/Kingcom/armips.git
+  git+https://github.com/google/cpu_features.git
+  git+https://github.com/discordapp/discord-rpc.git
+  "$pkgbase"-ffmpeg::git+https://github.com/hrydgard/"$pkgbase"-ffmpeg.git
+  "$pkgbase"-glslang::git+https://github.com/hrydgard/glslang.git
+  git+https://github.com/hrydgard/"$pkgbase"-lang.git
+  git+https://github.com/Tencent/rapidjson.git
+  git+https://github.com/KhronosGroup/SPIRV-Cross.git
+  armips-filesystem::git+https://github.com/Kingcom/filesystem.git
+  git+https://github.com/facebook/zstd.git
+  "$pkgbase".sh
+  "$pkgbase"-headless.sh
+  "$pkgbase"-qt.sh
+  "$pkgbase".desktop
+  "$pkgbase"-qt.desktop
+)
+sha256sums=('SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            '2fb7cfbd1f36715ae325a24c13821839a72c1d34b3cbed1e18a2a2ce2a7fb106'
+            '732c7414e937eeedc4448405d673885843e947326aafd8b77408d6ec36e47e32'
+            '3c17aeed93b80aad74d24cb654073b18833afdb2fb853a6c2d891a0e253c9cc9'
+            '78321b29e6d7b531c4404f473ac7e707f2bcce87f54e2ee2d2720b7c118fcc66'
+            '568bed0874030dfd0c0034f845f80a450e03dd14bdbac464282cc77a43703be3')
+
+pkgver() {
+  cd "$pkgbase"
+  git describe --abbrev=7 --tags | sed 's/^v//; s/-/.r/; s/-g/./'
+}
+
+prepare() {
+  test ! -d build-sdl || rm -rf build-sdl
+  test ! -d build-qt || rm -rf build-qt
+  cd "$pkgbase"
+  perl -pi -e 's/-O3/-O2/gi' CMakeLists.txt
+
+  for ui in sdl qt; do
+    if [[ -d build-$ui ]]; then
+      rm -rf build-$ui
+    fi
+  done
+
+  for submodule in assets/lang ext/glslang ffmpeg; do
+    git submodule init ${submodule}
+    git config submodule.${submodule}.url ../"$pkgbase"-${submodule#*/}
+    git -c protocol.file.allow=always submodule update ${submodule}
+  done
+
+  for submodule in ext/{armips,cpu_features,discord-rpc,rapidjson,SPIRV-Cross,zstd}; do
+    git submodule init ${submodule}
+    git config submodule.${submodule}.url ../${submodule#*/}
+    git -c protocol.file.allow=always submodule update ${submodule}
+  done
+
+  git submodule init ext/miniupnp
+  git -c protocol.file.allow=always submodule update ext/miniupnp
+
+  cd ext/armips
+
+  for submodule in ext/filesystem; do
+    git submodule init ${submodule}
+    git config submodule.${submodule}.url ../../../armips-${submodule#*/}
+    git -c protocol.file.allow=always submodule update ${submodule}
+  done
+}
+
+build() {
+  cmake -S $pkgbase -B build-sdl \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_SKIP_RPATH=ON \
+    -DHEADLESS=ON \
+    -DUSE_SYSTEM_LIBZIP=ON \
+    -DUSE_SYSTEM_SNAPPY=ON \
+    -DUSING_QT_UI=OFF \
+    -DSKIP_GLSLANG_INSTALL=ON \
+    -DOpenGL_GL_PREFERENCE=GLVND \
+    -Wno-dev \
+    -DGOLD=ON
+  cmake --build build-sdl -j$(nproc)
+
+#  cmake -S "$pkgbase" -B build-qt \
+#    -DCMAKE_BUILD_TYPE=Release \
+#    -DCMAKE_SKIP_RPATH=ON \
+#    -DHEADLESS=OFF \
+#    -DUSE_SYSTEM_LIBZIP=ON \
+#    -DUSE_SYSTEM_SNAPPY=ON \
+#    -DUSING_QT_UI=ON \
+#    -DSKIP_GLSLANG_INSTALL=ON \
+#    -DOpenGL_GL_PREFERENCE=GLVND \
+#    -Wno-dev
+#  cmake --build build-qt -j$(nproc)
+}
+
+package_ppsspp() {
+  depends+=(hicolor-icon-theme libzip)
+  provides=("$pkgname")
+  conflicts=("$pkgname"-git "$pkgname"-qt "$pkgname"-qt-git)
+
+  install -dm 755 "$pkgdir"/{opt/"$pkgname",usr/{bin,share/{applications,icons,pixmaps}}}
+  install -m 755 build-sdl/PPSSPPSDL "$pkgdir"/opt/"$pkgname"/
+  cp -dr --no-preserve=ownership build-sdl/assets "$pkgdir"/opt/"$pkgname"/
+  cp -dr --no-preserve=ownership "$pkgname"/icons/hicolor "$pkgdir"/usr/share/icons/
+  install -m 644 "$pkgname"/icons/icon-512.svg "$pkgdir"/usr/share/pixmaps/"$pkgname".svg
+  install -m 755 "$pkgname".sh "$pkgdir"/usr/bin/"$pkgname"
+  install -m 644 "$pkgname".desktop "$pkgdir"/usr/share/applications/
+}
+
+package_ppsspp-headless() {
+  provides=("$pkgname")
+  conflicts=("$pkgname"-git)
+
+  install -dm 755 "$pkgdir"/{opt/"$pkgbase",usr/bin}
+  install -m 755 build-sdl/PPSSPPHeadless "$pkgdir"/opt/"$pkgbase"/
+  install -m 755 "$pkgname".sh "$pkgdir"/usr/bin/"$pkgname"
+}
+
+package_ppsspp-qt() {
+  depends+=(hicolor-icon-theme libzip qt5-base qt5-multimedia )
+  provides=("$pkgname")
+  conflicts=("$pkgbase" "$pkgbase"-git "$pkgname"-git)
+
+  install -dm 755 "$pkgdir"/{opt/"$pkgbase",usr/{bin,share/{applications,icons,pixmaps}}}
+  install -m 755 build-qt/PPSSPPQt "$pkgdir"/opt/"$pkgbase"/
+  cp -dr --no-preserve=ownership build-qt/assets "$pkgdir"/opt/"$pkgbase"/
+  cp -dr --no-preserve=ownership "$pkgbase"/icons/hicolor "$pkgdir"/usr/share/icons/
+  install -m 644 "$pkgbase"/icons/icon-512.svg "$pkgdir"/usr/share/pixmaps/"$pkgbase".svg
+  install -m 755 "$pkgname".sh "$pkgdir"/usr/bin/"$pkgname"
+  install -m 644 "$pkgname".desktop "$pkgdir"/usr/share/applications/
+}
diff a/ppsspp/ppsspp.desktop b/ppsspp/ppsspp.desktop
index e69de29..a84929e 100644
--- a/ppsspp/ppsspp.desktop	1969-12-31 17:00:00.000000000 -0700
+++ b/ppsspp/ppsspp.desktop	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Version=1.0
+Type=Application
+Name=PPSSPP
+GenericName=PSP Emulator
+Comment=PlayStation Portable Simulator Suitable for Playing Portably
+Exec=ppsspp %f
+Icon=ppsspp
+Categories=Game
diff a/ppsspp/ppsspp-qt.desktop b/ppsspp/ppsspp-qt.desktop
index e69de29..a84929e 100644
--- a/ppsspp/ppsspp-qt.desktop	1969-12-31 17:00:00.000000000 -0700
+++ b/ppsspp/ppsspp-qt.desktop	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Version=1.0
+Type=Application
+Name=PPSSPP (Qt)
+GenericName=PSP Emulator
+Comment=PlayStation Portable Simulator Suitable for Playing Portably
+Exec=ppsspp-qt %f
+Icon=ppsspp
+Categories=Game
diff a/ppsspp/ppsspp.sh b/ppsspp/ppsspp.sh
index e69de29..a84929e 100644
--- a/ppsspp/ppsspp.sh	1969-12-31 17:00:00.000000000 -0700
+++ b/ppsspp/ppsspp.sh	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,3 @@
+#!/bin/sh
+cd /opt/ppsspp
+exec ./PPSSPPSDL "$@"
diff a/ppsspp/ppsspp-headless.sh b/ppsspp/ppsspp-headless.sh
index e69de29..a84929e 100644
--- a/ppsspp/ppsspp-headless.sh	1969-12-31 17:00:00.000000000 -0700
+++ b/ppsspp/ppsspp-headless.sh	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,3 @@
+#!/bin/sh
+cd /opt/ppsspp
+exec ./PPSSPPHeadless "$@"
diff a/ppsspp/ppsspp-qt.sh b/ppsspp/ppsspp-qt.sh
index e69de29..a84929e 100644
--- a/ppsspp/ppsspp-qt.sh	1969-12-31 17:00:00.000000000 -0700
+++ b/ppsspp/ppsspp-qt.sh	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,3 @@
+#!/bin/sh
+cd /opt/ppsspp
+exec ./PPSSPPQt "$@"
