diff a/python-b3ds/PKGBUILD b/python-b3ds/PKGBUILD
--- a/python-b3ds/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/python-b3ds/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,37 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgbase=python-b3ds
+pkgname=(python-b3dsdecrypt python-3dsdecode)
+pkgver=0.1
+pkgrel=1
+arch=(any)
+url=https://github.com/jagiella
+license=(GPL)
+depends=(python python-cryptography)
+makedepends=(python-setuptools)
+source=("$pkgbase"::git+https://github.com/jagiella/b3DS.git#branch=feature/switch-from-pycrypto-to-cryptography)
+sha256sums=('SKIP')
+
+prepare() {
+  cd "$pkgbase"
+  perl -pi -e 's|(def rol.*)|def main():\n    pass\n\n\1|gi' *DSDec*.py
+}
+
+package_python-b3dsdecrypt() {
+  pkgdesc='3DS Rom Decryptor'
+  cd "$pkgbase"
+  echo -e "from distutils.core import setup\nsetup(name='b3dsdecrypt',\n      version='0.1',\n      py_modules=['b3DSDecrypt'],\n      entry_points={'console_scripts': ['b3dsdecrypt=b3DSDecrypt:main',]},\n      )\npython_requires='>=3.7, <4',\ninstall_requires=['cryptography']," > setup.py
+  rm -rf build
+  python setup.py build
+  python setup.py install --root="$pkgdir" --optimize=1 --skip-build
+}
+
+package_python-3dsdecode() {
+  pkgdesc='3DS Rom Decoder'
+  depends+=(python-opencv python-numpy)
+  cd "$pkgbase"
+  echo -e "from distutils.core import setup\nsetup(name='3dsdecode',\n      version='0.1',\n      py_modules=['3DSDecoder'],\n      entry_points={'console_scripts': ['3dsdecode=3DSDecoder:main',]},\n      )\npython_requires='>=3.7, <4',\ninstall_requires=['cryptography', 'numpy', 'opencv-python']," > setup.py
+  rm -rf build
+  python setup.py build
+  python setup.py install --root="$pkgdir" --optimize=1 --skip-build
+}
