diff a/python-repochk/PKGBUILD b/python-repochk/PKGBUILD
--- a/python-repochk/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/python-repochk/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,43 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=python-repochk
+pkgver=0.1
+pkgrel=1
+pkgdesc="Repository Checker."
+url=https://github.com/noabody
+arch=(any)
+license=(Unlicense)
+depends=(python sudo trash-cli)
+optdepends=(
+  'git: Git repository checking'
+  'mercurial: Mercurial repository checking'
+  'subversion: SVN repository checking'
+  'lxc: LXC container management (-l option)'
+  'patch: applying PKGBUILD patches during builds (part of base-devel)'
+)
+
+makedepends=(python-build python-flit python-installer python-wheel)
+provides=("${pkgname#*-}")
+conflicts=("${pkgname#*-}")
+noextract=(master.tar.gz)
+source=(https://github.com/noabody/unibuild/archive/refs/heads/master.tar.gz
+        "$pkgname".patch)
+sha256sums=('SKIP'
+            'bb0a0ed431634030b8574a4cb6eaee36f62ce7fb8549b234f4459281f26394b0')
+
+prepare() {
+  tar -xzf master.tar.gz -C "$srcdir" --strip-components=2 --wildcards "*/data/${pkgname#*-}.py"
+  rm -f "$srcdir"/*.toml
+  patch -Np1 -i "$pkgname".patch
+}
+
+build() {
+  cd "$srcdir"
+  python -m build --wheel --no-isolation
+}
+
+package() {
+  cd "$srcdir"
+  python -m installer --destdir="$pkgdir" dist/*.whl
+  rm -f "$startdir"/master.tar.gz
+}
diff a/python-repochk/python-repochk.patch b/python-repochk/python-repochk.patch
index e69de29..e223692 100755
--- a/python-repochk/python-repochk.patch	1969-12-31 17:00:00.000000000 -0700
+++ b/python-repochk/python-repochk.patch	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,19 @@
+diff a/pyproject.toml b/pyproject.toml
+index e69de29..e223692 100755
+--- a/pyproject.toml	1969-12-31 17:00:00.000000000 -0700
++++ b/pyproject.toml	1969-12-31 17:00:00.000000000 -0700
+@@ -0,0 +1,14 @@
++[build-system]
++requires = ["flit_core>=3.2"]
++build-backend = "flit_core.buildapi"
++
++[project]
++name = "repochk"
++version = "0.1.0"
++description = "Repository Checker"
++requires-python = ">=3.8"
++
++dependencies = []
++
++[project.scripts]
++repochk = "repochk:main"
