diff a/python-b3ds/PKGBUILD b/python-b3ds/PKGBUILD
--- a/python-b3ds/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/python-b3ds/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,44 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgbase=python-b3ds
+pkgname=(python-b3ds python-3dsdecoder)
+pkgver=0.1
+pkgrel=1
+arch=(any)
+url=https://github.com/b1k/b3DS
+license=(GPL)
+depends=('python>=3.7')
+makedepends=(python-setuptools)
+source=(3dsdecoder::git+https://github.com/jagiella/b3DS.git#branch=feature/switch-from-pycrypto-to-cryptography
+        b3ds::git+https://github.com/ss-d49/b3DS.git)
+sha256sums=('SKIP'
+            'SKIP')
+
+package_python-b3ds() {
+  pkgdesc='3DS Rom Decrypter/Encrypter'
+  depends+=(python-pycryptodome)
+  cd b3ds
+#  echo -e '#!/bin/bash\npython /usr/lib/python3.*/site-packages/b3DS.py "$@"' > b3ds
+#  echo -e "from distutils.core import setup\nsetup(name='b3ds', version='0.1', py_modules=['b3DS'], entry_points={'console_scripts': ['b3ds=b3DS:main',]},)\npython_requires='>=3.7, <4',\ninstall_requires=['pycryptodome']," > setup.py
+#  echo -e "from distutils.core import setup\nsetup(name='b3ds', version='0.1', py_modules=['b3DS'],)\npython_requires='>=3.7, <4',\ninstall_requires=['pycryptodome']," > setup.py
+#  rm -rf build
+#  python setup.py build
+#  python setup.py install --root="$pkgdir" --optimize=1 --skip-build
+#  install -Dm755 b3ds -t "$pkgdir"/usr/bin
+  install -Dm755 b3DS.py "$pkgdir"/usr/bin/b3ds
+}
+
+package_python-3dsdecoder() {
+  pkgdesc='3DS Rom Decoder'
+  depends+=(python-opencv python-numpy)
+  cd 3dsdecoder
+#  perl -pi -e 's|(def rol.*)|def main():\n    pass\n\n\1|gi' 3DSDecoder.py
+#  echo -e '#!/bin/bash\npython /usr/lib/python3.*/site-packages/3DSDecoder.py "$@"' > 3dsdecoder
+#  echo -e "from distutils.core import setup\nsetup(name='3dsdecoder', version='0.1', py_modules=['3DSDecoder'], entry_points={'console_scripts': ['3dsdecoder=3DSDecoder:main',]},)\npython_requires='>=3.7, <4',\ninstall_requires=['cryptography', 'numpy', 'opencv-python']," > setup.py
+#  echo -e "from distutils.core import setup\nsetup(name='3dsdecoder', version='0.1', py_modules=['3DSDecoder'],)\npython_requires='>=3.7, <4',\ninstall_requires=['cryptography', 'numpy', 'opencv-python']," > setup.py
+#  rm -rf build
+#  python setup.py build
+#  python setup.py install --root="$pkgdir" --optimize=1 --skip-build
+#  install -Dm755 3dsdecoder -t "$pkgdir"/usr/bin
+  install -Dm755 3DSDecoder.py "$pkgdir"/usr/bin/3dsdecoder
+}
