diff a/rbdoom3bfg/PKGBUILD b/rbdoom3bfg/PKGBUILD
--- a/rbdoom3bfg/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/rbdoom3bfg/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,62 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=rbdoom3bfg
+pkgver=r1858.f330194d
+pkgrel=1
+pkgdesc='Refined Doom 3 BFG.'
+arch=(x86_64)
+url=https://github.com/RobertBeckebans/RBDOOM-3-BFG
+license=(GPL3)
+depends=(ffmpeg glew openal libpng perl-rename sdl2 xorg-fonts-misc)
+makedepends=(cmake directx-shader-compiler gendesk git
+             glu icoutils inkscape mesa rapidjson zip)
+provides=("$pkgname")
+replaces=(rbdoom3-bfg-git rbdoom-3-bfg)
+conflicts=(rbdoom3-bfg-git rbdoom-3-bfg)
+source=(rbdoom3bfg::git+https://github.com/RobertBeckebans/RBDOOM-3-BFG.git
+        "$pkgname".sh)
+sha256sums=('SKIP'
+         '1c98b34a26c6abb1b5a39aff4a2a2cb7a5f16204a7188be2d455f3f0e3fb845d')
+
+pkgver() {
+  cd "$pkgname"
+  printf "r%s.%s" "$(git rev-list --count HEAD)" "$(git rev-parse --short HEAD)"
+}
+
+prepare() {
+  gendesk -f -n --name='RBDoom3BFG' --comment="${pkgdesc/.}" --custom='Keywords=game;engine;' --exec="/usr/share/$pkgname/$pkgname.sh" --icon="$pkgname" --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname="$pkgname"
+  if [ ! -f "$pkgname.svg" ]; then
+    icotool -i 1 -x "$pkgname"/neo/sys/win32/rc/res/doom.ico -o "$pkgname".png
+    inkscape "$pkgname".png --export-plain-svg --export-filename="$pkgname".svg
+    rm -f "$pkgname".ico "$pkgname".png
+  fi
+  test ! -d build || rm -rf build
+  cd "$pkgname"
+  perl -pi -e 's/.*werror.*//gi' neo/CMakeLists.txt
+  git submodule update --init --recursive
+}
+
+build() {
+#  cd "$pkgname"/neo
+#  ./cmake-linux-release.sh
+#  cmake --build ../build -j$(nproc)
+  cmake -S "$pkgname"/neo -B build \
+    -DCMAKE_INSTALL_PREFIX=/usr \
+    -DUSE_SYSTEM_LIBGLEW=ON \
+    -DUSE_SYSTEM_LIBJPEG=ON \
+    -DUSE_SYSTEM_LIBPNG=ON \
+    -DUSE_SYSTEM_RAPIDJSON=ON \
+    -DUSE_SYSTEM_ZLIB=ON \
+    -DUSE_PRECOMPILED_HEADERS=OFF \
+    -Wno-dev
+  cmake --build build -j$(nproc)
+}
+
+package() {
+  install -dm 755 "$pkgdir"/usr/{bin,share/{"$pkgname",applications,icons/hicolor/scalable/apps}}
+  install -m 755 build/RBDoom3BFG "$pkgdir"/usr/share/"$pkgname/$pkgname"
+  install -m 755 "$pkgname".sh "$pkgdir"/usr/share/"$pkgname"/
+  install -m 644 "$pkgname".desktop "$pkgdir"/usr/share/applications/
+  install -m 644 "$pkgname".svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  ln -rsf "$pkgdir"/usr/share/"$pkgname/$pkgname" "$pkgdir"/usr/bin/
+}
diff a/rbdoom3bfg/rbdoom3bfg.sh b/rbdoom3bfg/rbdoom3bfg.sh
--- a/rbdoom3bfg/rbdoom3bfg.sh	1969-12-31 17:00:00.000000000 -0700
+++ b/rbdoom3bfg/rbdoom3bfg.sh	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,61 @@
+#!/bin/bash
+gmdir="$HOME/games/doom3"
+gmcfg="$HOME/.local/share/rbdoom3bfg"
+gmtgt="base/_common.resources"
+gmprm=""
+glnch="rbdoom3bfg"
+
+gconf () {
+test -d "$gmcfg" || mkdir -p "$gmcfg"
+echo -e "gamepath= $gmdir\ngamemod= base\ngameparm= $gmprm" | tee "$gmcfg/basedir"
+}
+
+gstart () {
+echo "$glnch +set fs_basepath $gmbdr $gmgnm $gmpar" | xargs -i -r sh -c "{}"
+}
+
+gtest () {
+gmbse="$(grep -Pi '^gamepath=.*' "$gmcfg/basedir" | head -1)"
+gmgme="$(grep -Pi '^gamemod=.*' "$gmcfg/basedir" | head -1)"
+gmgpm="$(grep -Pi '^gameparm=.*' "$gmcfg/basedir" | head -1)"
+gmbdr="${gmbse#gamepath= }"
+gmgnm="${gmgme#gamemod= }"
+gmpar="${gmgpm#gameparm= }"
+}
+
+test -f "$gmcfg/basedir" || gconf
+gtest
+if [ -z "$gmbdr" ]; then
+  gconf
+  gtest
+fi
+if [ -d "$gmbdr" ]; then
+  if [ ! -f "$gmbdr/$gmtgt" ]; then
+    gmchk="$(find -H "$gmbdr" -mindepth 2 -maxdepth 2 -type f -regextype posix-extended -iregex "$gmbdr/$gmtgt" -printf '%p\n' | grep -Pic "$gmbdr/$gmtgt")"
+    if [ $gmchk -eq 1 ]; then
+      echo -e "File system is case sensitive\nFile/folder names must be lowercase\n Rename subdirs of\n  $gmbdr\n    containing\n  $gmtgt?\nRequires perl-rename command" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      test $? -eq 101 && find -H "$gmbdr" -depth -exec perl-rename 's|([^/]*\Z)|lc($1)|e' {} +
+    elif [ $gmchk -gt 1 ]; then
+      echo -e "File system is case sensitive\nFile/folder names must be lowercase\n Cannot rename subdirs of\n  $gmbdr\n    containing\n  $gmtgt\n\nBecause more than one exists" | xmessage -file - -buttons Ok:0 -center -timeout 30
+    else
+      echo -e "Config:\n $gmcfg/basedir\n  $gmbse\n found, but not\n  $gmgme\n ($gmbdr/$gmtgt)\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      test $? -eq 101 && gconf
+    fi
+  else
+    if [ ! -d "$gmbdr/$gmgnm" ]; then
+      echo -e "Config:\n $gmcfg/basedir\n  $gmbse\n found, but not\n  $gmgme\n ($gmbdr/$gmgnm)\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      test $? -eq 101 && gconf
+    else
+      if [ -z "$(echo "$gmgnm" | grep -Pio '\w+')" ]; then
+        gmgnm=""
+        gstart
+      else
+        gmgnm="+set game $gmgnm"
+        gstart
+      fi
+    fi
+  fi
+else
+  echo -e "Config:\n $gmcfg/basedir\n  $gmbse\n not found\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+  test $? -eq 101 && gconf
+fi
