diff a/rclone-browserng/PKGBUILD b/rclone-browserng/PKGBUILD
--- a/rclone-browserng/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/rclone-browserng/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,43 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=rclone-browserng
+pkgver=2.0.0.r243.ge5871fa
+pkgrel=1
+pkgdesc='Simple cross-platform GUI for rclone.'
+url=https://github.com/SysAdminDoc/RcloneBrowserNG
+arch=(x86_64)
+license=('MIT')
+depends=(hicolor-icon-theme qt6-base rclone)
+makedepends=(cmake)
+source=("$pkgname"::git+https://github.com/SysAdminDoc/RcloneBrowserNG.git)
+provides=("$pkgname")
+conflicts=("$pkgname")
+sha256sums=('SKIP')
+
+pkgver() {
+  cd "$pkgname"
+  printf "%s.r%s.g%s" "$(cat VERSION)" "$(git rev-list "$(git rev-list --tags --no-walk --max-count=1)"..HEAD --count)" "$(git rev-parse --short HEAD)"
+#  git describe --long --abbrev=7 --tags | sed 's/^v//;s/_/./;s/\([^-]*-g\)/r\1/;s/-/./g'
+}
+
+prepare() {
+  test ! -d "build" || rm -rf build
+  perl -pi -e 's/ -Werror//gi' "$pkgname"/src/CMakeLists.txt
+  perl -pi -e 's/termargs.*fullargs/termArgs << QString("%1 %2").arg(GetRclone(), fullArgs.join(" "))/gi' "$pkgname"/src/main_window.cpp
+}
+
+build() {
+  cmake -S "$pkgname" -B build \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DBUILD_TESTING=OFF \
+    -DCMAKE_SKIP_RPATH=ON \
+    -DCMAKE_INSTALL_PREFIX=/usr
+  cmake --build build -j$(nproc)
+}
+
+package() {
+  DESTDIR="$pkgdir" cmake --build build --target install
+  mv "$pkgdir"/usr/bin/rclone-browser "$pkgdir"/usr/bin/"$pkgname"
+  perl -pi -e 's/(?<=exec=rclone-browser)/ng/gi' "$pkgdir"/usr/share/applications/*.desktop
+  install -Dm644 "$pkgname"/LICENSE "$pkgdir"/usr/share/licenses/"$pkgname"/LICENSE
+}
