diff a/rtl8761bu_fw/PKGBUILD b/rtl8761bu_fw/PKGBUILD
--- a/rtl8761bu_fw/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/rtl8761bu_fw/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,18 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=rtl8761bu_fw
+pkgver=1.9.1040.1025
+pkgrel=1
+pkgdesc="RTL8761BU HCI 6.0 firmware."
+arch=('any')
+license=('unknown')
+source=(https://catalog.s.download.windowsupdate.com/d/msdownload/update/driver/drvs/2026/03/003d840c-ad58-4fb1-a195-8b2c93f50f4b_745f28875680612c239c83ca7e9cef2d13c0fe9d.cab)
+sha256sums=('70d23f723063804ea52f23631f9a5bd39471d963c1049077f15ac5b425b1144f')
+
+pkgver() {
+  grep -Pio "driverver.*\b.*\d+\.[[:alnum:]\.]+" Rtkfilter.inf | grep -Pio '\d+\.[[:alnum:]\.]+'
+}
+
+package() {
+  install -Dm644 rtl8761b_mp_chip_bt40_fw_asic_rom_patch_new.dat "$pkgdir"/usr/lib/firmware/rtl_bt/"$pkgname".bin
+}
