diff a/rtl8821c_fw/PKGBUILD b/rtl8821c_fw/PKGBUILD
--- a/rtl8821c_fw/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/rtl8821c_fw/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,18 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=rtl8821c_fw
+pkgver=16.4033.2312.2502
+pkgrel=1
+pkgdesc="RTL8821C HCI 5.0 firmware."
+arch=('any')
+license=('unknown')
+source=(https://catalog.s.download.windowsupdate.com/d/msdownload/update/driver/drvs/2026/05/9421a5d6-9853-40ff-9653-4ba78e71002e_2bee537880d83935eed2a2142f59b7a5041117e9.cab)
+sha256sums=('7d4cddfe233c572bb904f8e11d0f3f46b24a3b691af19b9165e5db1793b02659')
+
+pkgver() {
+  grep -Pio "driverver.*\b.*\d+\.[[:alnum:]\.]+" Rtkfilter*.inf | grep -Pio '\d+\.[[:alnum:]\.]+'
+}
+
+package() {
+  install -Dm644 rtl8821c_mp_chip_new.dat "$pkgdir"/usr/lib/firmware/rtl_bt/"$pkgname".bin
+}
