diff a/rtl8761bu_fw/PKGBUILD b/rtl8761bu_fw/PKGBUILD
--- a/rtl8761bu_fw/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/rtl8761bu_fw/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,18 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=rtl8761bu_fw
+pkgver=1.9.1051.3016
+pkgrel=1
+pkgdesc="RTL8761BU HCI 5.4 firmware."
+arch=('any')
+license=('unknown')
+source=(https://static.tp-link.com/upload/driver/2025/202503/20250314/UB500_V3_Win10_Win11.zip)
+sha256sums=('66d218ef2a5dc0d0e6d8ab8f310409d7e72be8c84a1010054269d3cc66434c78')
+
+pkgver() {
+  grep -Pio "driverver.*\b.*\d+\.[[:alnum:]\.]+" 'UB500V3/BT/plugins/Driver Files/Driver/Windows_11_64bit/Rtkfilter.inf' | grep -Pio '\d+\.[[:alnum:]\.]+'
+}
+
+package() {
+  install -Dm644 'UB500V3/BT/plugins/Driver Files/Driver/Windows_11_64bit/rtl8761b_mp_chip_bt40_fw_asic_rom_patch_new.dat' "$pkgdir"/usr/lib/firmware/rtl_bt/"$pkgname".bin
+}
