diff a/sdl2_mixer_ext/PKGBUILD b/sdl2_mixer_ext/PKGBUILD
--- a/sdl2_mixer_ext/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/sdl2_mixer_ext/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,48 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=sdl2_mixer_ext
+pkgver=2.8.2
+pkgrel=1
+pkgdesc='Audio mixer library'
+arch=(x86_64)
+url=https://wohlsoft.github.io/SDL-Mixer-X/
+license=(GPL-3.0-or-later)
+depends=(fluidsynth sdl2 libadlmidi libmodplug libgme libvorbis libxmp wavpack opusfile)
+makedepends=(cmake fluidsynth git ninja)
+source=("$pkgname"::git+https://github.com/WohlSoft/SDL-Mixer-X.git)
+sha256sums=('SKIP')
+
+pkgver() {
+  cd "$pkgname"
+  git describe --long --abbrev=7 --tags | sed 's/^v//;s/_/./;s/\([^-]*-g\)/r\1/;s/-/./g'
+}
+
+prepare() {
+  [[ ! -d "build" ]] || rm -rf build
+}
+
+build() {
+  cmake -S "$pkgname" -B build -G Ninja \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_INSTALL_PREFIX=/usr \
+    -DSDL_MIXER_X_SHARED=ON \
+    -DSDL_MIXER_X_STATIC=OFF \
+    -DMIXERX_ENABLE_GPL=ON \
+    -DMIXERX_ENABLE_LGPL=ON \
+    -DUSE_SYSTEM_SDL2=ON \
+    -DUSE_SYSTEM_AUDIO_LIBRARIES=ON \
+    -DUSE_OGG_VORBIS_STB=OFF \
+    -DUSE_DRFLAC=OFF \
+    -DUSE_WAVPACK_DSD=ON \
+    -DUSE_MP3_DRMP3=OFF \
+    -DUSE_MIDI=ON \
+    -DUSE_MIDI_ADLMIDI=ON \
+    -DUSE_MIDI_EDMIDI=OFF \
+    -DUSE_MIDI_FLUIDLITE=OFF \
+    -DUSE_MIDI_FLUIDSYNTH=ON
+  cmake --build build -j$(nproc)
+}
+
+package() {
+  DESTDIR="$pkgdir" cmake --build build --target install
+}
