diff a/si-vulkan/PKGBUILD b/si-vulkan/PKGBUILD
--- a/si-vulkan/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/si-vulkan/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,20 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=si-vulkan
+pkgver=1.0
+pkgrel=1
+pkgdesc="Sea Islands GCN 2 Vulkan enabler."
+license=(GPL)
+arch=(any)
+provides=("$pkgname")
+conflicts=("$pkgname")
+source=(amdgpu.conf
+        radeon.conf)
+sha256sums=('1882f887e0b7edb34d9ccec0b49b50c221c0ce5e654b4831aa8129e75b50b43b'
+            'de25fcb18709f042e39bcf5175d4aba4cdcfaf01ef09b0bdde91621c97ff8cdd')
+
+package() {
+  install -dm 755 "$pkgdir"/etc/modprobe.d
+  install -m 644 amdgpu.conf "$pkgdir"/etc/modprobe.d/
+  install -m 644 radeon.conf "$pkgdir"/etc/modprobe.d/
+}
diff a/si-vulkan/amdgpu.conf b/si-vulkan/amdgpu.conf
index e69de29..feb733a 100644
--- a/si-vulkan/amdgpu.conf	1969-12-31 17:00:00.000000000 -0700
+++ b/si-vulkan/amdgpu.conf	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,2 @@
+options amdgpu si_support=0
+options amdgpu cik_support=1
diff a/si-vulkan/radeon.conf b/si-vulkan/radeon.conf
index e69de29..feb733a 100644
--- a/si-vulkan/radeon.conf	1969-12-31 17:00:00.000000000 -0700
+++ b/si-vulkan/radeon.conf	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,2 @@
+options radeon si_support=0
+options radeon cik_support=0

