diff a/solaar/PKGBUILD b/solaar/PKGBUILD
--- a/solaar/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/solaar/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,53 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=solaar
+pkgver=1.1.9.r30.g8dd8c8b
+pkgrel=1
+pkgdesc="Device manager for Logitech's Unifying receiver peripherals"
+url=https://pwr.github.com/Solaar
+license=(GPL2)
+arch=(any)
+provides=("$pkgname" python-hid-parser python-hid-parser-git)
+conflicts=("$pkgname" python-hid-parser python-hid-parser-git)
+depends=(gtk3 libnotify python-dbus python-evdev python-gobject
+         python-psutil python-yaml python-pyudev python-xlib)
+optdepends=('libappindicator-gtk3: tray icon support')
+makedepends=(git)
+source=("$pkgname"::git+https://github.com/pwr-Solaar/Solaar.git)
+sha256sums=('SKIP')
+
+pkgver() {
+  cd "$pkgname"
+  git describe --long --abbrev=7 --tags | sed 's/\([^-]*-g\)/r\1/;s/-/./g'
+#  printf "%s.r%s.%s" $(git describe --long --abbrev=7 --tags | sed 's/-rc[^-]*//;s/\([^-]*-g\)/\1/;s/-/ /g')
+}
+
+prepare() {
+  cd "$pkgname"
+  git submodule update --init --recursive
+  python setup.py clean
+}
+
+build() {
+  cd "$pkgname"
+  python setup.py build
+}
+
+package() {
+  cd "$pkgname"
+  tools/po-compile.sh
+  python setup.py install --skip-build \
+    --optimize=1 \
+    --prefix=/usr \
+    --root="$pkgdir"/
+  # udev
+  rm -rf "$pkgdir"/usr/share/"$pkgname"/udev-rules.d
+  install -vDm 644 rules.d/42-logitech-unify-permissions.rules \
+    "$pkgdir"/usr/lib/udev/rules.d/42-logitech-unify-permissions.rules
+  # autostart
+  install -vDm 644 share/autostart/"$pkgname".desktop \
+    "$pkgdir"/etc/xdg/autostart/"$pkgname".desktop
+  # docs
+  install -vDm 644 {CHANGELOG.md,README.md} \
+    -t "$pkgdir"/usr/share/doc/"$pkgname"/
+}
