diff a/wiiudownloader/PKGBUILD b/wiiudownloader/PKGBUILD
--- a/wiiudownloader/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/wiiudownloader/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,47 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=wiiudownloader
+pkgver=1.20.r7.g841fa19
+pkgrel=1
+pkgdesc='Download encrypted wiiu files from official servers.'
+arch=(x86_64)
+url=https://github.com/Xpl0itU/WiiUDownloader
+license=(GPL2)
+depends=(cairo glibc gtk3)
+makedepends=(git go python)
+provides=("$pkgname")
+conflicts=("$pkgname" "$pkgname"-git)
+source=("$pkgname"::git+$url.git)
+md5sums=('SKIP')
+
+prepare() {
+  gendesk -f -n --name=WiiUDownloader --comment="${pkgdesc%.*}" --custom='Keywords=game;engine;' --exec="$pkgname" --icon="$pkgname" --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname="$pkgname"
+  cd "$pkgname"
+  test ! -d "build" || rm -rf build
+  mkdir -p build
+}
+
+
+pkgver() {
+  cd "$pkgname"
+  git describe --long --abbrev=7 --tags | sed 's/^v//i;s/_/./;s/\([^-]*-g\)/r\1/;s/-/./g'
+}
+
+build() {
+  cd "$pkgname"
+  export CGO_CPPFLAGS="${CPPFLAGS}"
+  export CGO_CFLAGS="${CFLAGS}"
+  export CGO_CXXFLAGS="${CXXFLAGS}"
+  export CGO_LDFLAGS="${LDFLAGS}"
+  export GOFLAGS="-buildmode=pie -trimpath -mod=readonly -modcacherw"
+  python3 grabTitles.py
+#  go build cmd/WiiUDownloader/main.go cmd/WiiUDownloader/darkMode.go cmd/WiiUDownloader/mainwindow.go cmd/WiiUDownloader/utils.go
+  go build -o build/wiiudownloader -ldflags="-linkmode=external -X main.version=$pkgver" ./cmd/WiiUDownloader
+}
+
+package() {
+  install -dm 755 "$pkgdir"/usr/{bin,share/{applications,pixmaps}}
+  install -m 755 "$pkgname"/build/"$pkgname" "$pkgdir"/usr/bin/"$pkgname"
+  install -m 644 "$pkgname"/data/WiiUDownloader.png "$pkgdir"/usr/share/pixmaps/"$pkgname".png
+  install -m 644 "$pkgname".desktop "$pkgdir"/usr/share/applications/
+}
