diff a/wstart/PKGBUILD b/wstart/PKGBUILD
--- a/wstart/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/wstart/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,35 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=wstart
+pkgver=0.2
+pkgrel=1
+pkgdesc="Wine Starter."
+url=https://github.com/noabody
+arch=(x86_64)
+license=(Unlicense)
+depends=(glibc)
+makedepends=(nim)
+provides=("$pkgname")
+conflicts=(python-"$pkgname")
+noextract=(master.tar.gz)
+source=(https://github.com/noabody/unibuild/archive/refs/heads/master.tar.gz
+        "$pkgname".patch)
+sha256sums=('SKIP'
+            'c4c14c032ed5e37f92b932932f6bec8db9fa9662f68fee6efd9b9e236448aa22')
+
+prepare() {
+  tar -xzf master.tar.gz -C "$srcdir" --strip-components=2 --wildcards "*/data/$pkgname.nim"
+  patch -Np1 -i "$pkgname".patch
+}
+
+build() {
+  cd "$srcdir"
+  nimble install
+  nim c -d:release -d:ssl --opt:size "$pkgname".nim
+}
+
+package() {
+  cd "$srcdir"
+  install -Dm755 "$pkgname" -t "$pkgdir"/usr/bin
+#  rm -rf "$srcdir" "$startdir"/master.tar.gz
+}
diff a/wstart/wstart.patch b/wstart/wstart.patch
index e69de29..e223692 100755
--- a/wstart/wstart.patch	1969-12-31 17:00:00.000000000 -0700
+++ b/wstart/wstart.patch	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,15 @@
+diff a/wstart.nimble b/wstart.nimble
+index e69de29..e223692 100755
+--- a/wstart.nimble	1969-12-31 17:00:00.000000000 -0700
++++ b/wstart.nimble	1969-12-31 17:00:00.000000000 -0700
+@@ -0,0 +1,10 @@
++# wstart native Nim build
++version       = "0.2.0"
++author        = "wstart"
++description   = "Wine Starter"
++license       = "unlicense"
++srcDir        = "."
++bin           = @["wstart"]
++
++requires "nim >= 2.0.0"
++requires "https://github.com/RePRGM/PEFile"
