diff a/wstart/PKGBUILD b/wstart/PKGBUILD
--- a/wstart/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/wstart/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,32 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=wstart
+pkgver=0.1
+pkgrel=1
+pkgdesc="Wine Starter."
+url=https://github.com/noabody
+arch=(x86_64)
+license=(Unlicense)
+depends=(glibc)
+makedepends=(nim)
+provides=("$pkgname")
+conflicts=(python-"$pkgname")
+noextract=(master.tar.gz)
+source=(https://github.com/noabody/unibuild/archive/refs/heads/master.tar.gz)
+sha256sums=('SKIP')
+
+prepare() {
+  tar -xzf master.tar.gz -C "$srcdir" --strip-components=2 --wildcards "*/data/$pkgname.nim" "*/data/$pkgname.nimble"
+}
+
+build() {
+  cd "$srcdir"
+  nimble install
+  nim c -d:release -d:ssl --opt:size "$pkgname".nim
+}
+
+package() {
+  cd "$srcdir"
+  install -Dm755 "$pkgname" -t "$pkgdir"/usr/bin
+  rm -rf "$srcdir" "$startdir"/master.tar.gz
+}
