diff a/xash3d/PKGBUILD b/xash3d/PKGBUILD
--- a/xash3d/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/xash3d/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,85 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=xash3d
+pkgver=0.19.2.r56.g725fd84b
+pkgrel=1
+pkgdesc='Custom Gold Source Engine built from scratch.'
+arch=(x86_64)
+url=http://xash.su
+license=(GPL2)
+depends=(freetype2 fontconfig hicolor-icon-theme
+         perl-rename sdl2 xorg-fonts-misc)
+makedepends=(cmake freetype2 gendesk git glu icoutils inkscape python2)
+#options=(!strip)
+source=(
+  "$pkgname"::git+https://github.com/FWGS/"$pkgname".git#branch=master
+  valve::git+https://github.com/FWGS/hlsdk-"$pkgname".git#commit=6af5c3f472f1efc6c65369689499b3431ce0650b
+  gearbox::git+https://github.com/FWGS/hlsdk-"$pkgname".git#branch=opfor
+  bshift::git+https://github.com/FWGS/hlsdk-"$pkgname".git#commit=c7fe134222c45575391ebac095d153c4e2bbbbc7
+  xash-extras::git+https://github.com/FWGS/xash-extras.git#branch=master
+  https://raw.githubusercontent.com/FWGS/"$pkgname"-android-project/master/makepak.py
+  "$pkgname".sh
+)
+provides=($pkgname)
+conflicts=(
+  $pkgname-git
+  $pkgname-hlsdk
+)
+sha256sums=('SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            '0f08d22f7f382b990f58f3fb4b2e26c24a6f2f833ed3b95d8875056c9beef43a')
+
+pkgver() {
+  cd "$pkgname"
+  git describe --long --tags | sed 's/^v//;s/\([^-]*-g\)/r\1/;s/-/./g'
+}
+
+prepare() {
+  gendesk -f -n --name='Xash3D' --comment="${pkgdesc/.}" --custom='Keywords=game;engine;' --exec="/usr/share/$pkgname/$pkgname.sh" --icon="$pkgname" --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname="$pkgname"
+  test -f "$pkgname.svg" || inkscape "$pkgname"/game_launch/icon-xash-material.png --export-plain-svg --export-filename="$pkgname".svg
+  cd "$pkgname"
+  git submodule update --init --recursive
+  for _mod in "$pkgname" valve gearbox bshift; do
+    if [[ -d "$srcdir/$_mod/build" ]]; then
+      rm -rf "$srcdir/$_mod"/build
+    fi
+  done
+}
+
+build() {
+  python2 makepak.py xash-extras extras.pak
+  cmake -S "$pkgname" -B "$pkgname"/build \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DXASH_DOWNLOAD_DEPENDENCIES=OFF \
+    -DXASH_SDL=ON \
+    -DXASH_64BIT=ON
+  make -C "$pkgname"/build -j4
+  for _mod in valve-hl gearbox-opfor bshift; do
+  cmake -S "${_mod%-*}" -B "${_mod%-*}"/build \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DSERVER_LIBRARY_NAME="${_mod#*-}" \
+    -D64BIT=ON
+  make -C "${_mod%-*}"/build -j4
+  done
+}
+
+package() {
+  install -dm 755 "$pkgdir"/usr/{bin,share/{"$pkgname"/{lib,bshift/{dlls,cl_dlls},gearbox/{dlls,cl_dlls},valve/{dlls,cl_dlls}},applications,icons/hicolor/scalable/apps}}
+  install -m 755 "$pkgname"/build/game_launch/"$pkgname" "$pkgdir"/usr/share/"$pkgname"/
+  install -m 644 "$pkgname"/build/{engine,mainui}/lib*.so "$pkgdir"/usr/share/"$pkgname"/lib/
+  install -m 644 valve/build/cl_dll/client*64.so "$pkgdir"/usr/share/"$pkgname"/valve/cl_dlls/client64.so
+  install -m 644 valve/build/dlls/hl*64.so "$pkgdir"/usr/share/"$pkgname"/valve/dlls/hl64.so
+  install -m 644 bshift/build/cl_dll/client*64.so "$pkgdir"/usr/share/"$pkgname"/bshift/cl_dlls/client64.so
+  install -m 644 bshift/build/dlls/bshift*64.so "$pkgdir"/usr/share/"$pkgname"/bshift/dlls/bshift64.so
+  install -m 644 gearbox/build/cl_dll/client*64.so "$pkgdir"/usr/share/"$pkgname"/gearbox/cl_dlls/client64.so
+  install -m 644 gearbox/build/dlls/opfor*64.so "$pkgdir"/usr/share/"$pkgname"/gearbox/dlls/opfor64.so
+  install -m 644 extras.pak "$pkgdir"/usr/share/"$pkgname"/
+  install -m 755 "$pkgname".sh "$pkgdir"/usr/share/"$pkgname"/
+  install -m 644 "$pkgname".svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  install -m 644 "$pkgname".desktop "$pkgdir"/usr/share/applications/
+  ln -rsf "$pkgdir"/usr/share/"$pkgname/$pkgname" "$pkgdir"/usr/bin/
+}
diff a/xash3d/xash3d.sh b/xash3d/xash3d.sh
--- a/xash3d/xash3d.sh	1969-12-31 17:00:00.000000000 -0700
+++ b/xash3d/xash3d.sh	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,76 @@
+#!/bin/bash
+gmdir="$HOME/games/half-life"
+gmcfg="$HOME/.config/xash3d"
+gmtgt="valve/halflife.wad"
+gmprm="-console"
+glnch="xash3d"
+
+gconf () {
+test -d "$gmcfg" || mkdir -p "$gmcfg"
+echo -e "gamepath= $gmdir\ngamemod= valve\ngameparm= $gmprm" | tee "$gmcfg/basedir"
+}
+
+gstart () {
+for value in valve-hl gearbox-opfor bshift; do
+  gmdll=${value#*-}
+  value=${value%-*}
+  if [ -d "$gmbdr/$value" ]; then
+    etest="$(test -f "$gmbdr/$value/liblist.gam" && grep -Pio "^gamedll_linux.*" "$gmbdr/$value/liblist.gam")"
+    if [ -n "$etest" ]; then
+      test -z "$(echo "$etest" | grep -Pio "^gamedll_linux \"dlls/"$gmdll"64.so\"")" && perl -pi -e "s|(^gamedll_linux).*|\1 \"dlls/"$gmdll"64.so\"|i" "$gmbdr/$value/liblist.gam"
+    else
+      echo "gamedll_linux \"dlls/"$gmdll"64.so\"" >> "$gmbdr/$value/liblist.gam"
+    fi
+    test -h "$gmbdr/$value/cl_dlls/client_amd64.so" || ln -sf /usr/share/xash3d/$value/cl_dlls/client*64.so $gmbdr/$value/cl_dlls/client64.so
+    test -h "$gmbdr/$value/dlls/"$gmdll"_amd64.so" || ln -sf /usr/share/xash3d/$value/dlls/"$gmdll"*64.so $gmbdr/$value/dlls/"$gmdll"64.so
+  fi
+done
+test -h "$gmbdr/valve/extras.pak" || ln -sf /usr/share/xash3d/extras.pak "$gmbdr"/valve/
+echo "LD_LIBRARY_PATH=/usr/share/xash3d/lib:$LD_LIBRARY_PATH XASH3D_BASEDIR=$gmbdr $glnch $gmgnm $gmpar" | xargs -i -r sh -c "{}"
+}
+
+gtest () {
+gmbse="$(grep -Pi '^gamepath=.*' "$gmcfg/basedir" | head -1)"
+gmgme="$(grep -Pi '^gamemod=.*' "$gmcfg/basedir" | head -1)"
+gmgpm="$(grep -Pi '^gameparm=.*' "$gmcfg/basedir" | head -1)"
+gmbdr="${gmbse#gamepath= }"
+gmgnm="${gmgme#gamemod= }"
+gmpar="${gmgpm#gameparm= }"
+}
+
+test -f "$gmcfg/basedir" || gconf
+gtest
+if [ -z "$gmbdr" ]; then
+  gconf
+  gtest
+fi
+if [ -d "$gmbdr" ]; then
+  if [ ! -f "$gmbdr/$gmtgt" ]; then
+    gmchk="$(find -H "$gmbdr" -mindepth 2 -maxdepth 2 -type f -regextype posix-extended -iregex "$gmbdr/$gmtgt" -printf '%p\n' | grep -Pic "$gmbdr/$gmtgt")"
+    if [ $gmchk -eq 1 ]; then
+      echo -e "File system is case sensitive\nFile/folder names must be lowercase\n Rename subdirs of\n  $gmbdr\n    containing\n  $gmtgt?\nRequires perl-rename command" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      test $? -eq 101 && find -H "$gmbdr" -depth -exec perl-rename 's|([^/]*\Z)|lc($1)|e' {} +
+    elif [ $gmchk -gt 1 ]; then
+      echo -e "File system is case sensitive\nFile/folder names must be lowercase\n Cannot rename subdirs of\n  $gmbdr\n    containing\n  $gmtgt\n\nBecause more than one exists" | xmessage -file - -buttons Ok:0 -center -timeout 30
+    else
+      echo -e "Config:\n $gmcfg/basedir\n  $gmbse\n found, but not\n  $gmgme\n ($gmbdr/$gmtgt)\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      test $? -eq 101 && gconf
+    fi
+  else
+    if [ ! -d "$gmbdr/$gmgnm" ]; then
+      echo -e "Config:\n $gmcfg/basedir\n  $gmbse\n found, but not\n  $gmgme\n ($gmbdr/$gmgnm)\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      test $? -eq 101 && gconf
+    else
+      if [ -z "$(echo "$gmgnm" | grep -Pio '\w+')" ]; then
+        gmgnm=""
+        gstart
+      else
+        gmgnm="-game $gmgnm"
+        gstart
+      fi
+    fi
+  fi
+else
+  echo -e "Config:\n $gmcfg/basedir\n  $gmbse\n not found\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+  test $? -eq 101 && gconf
+fi
