diff a/yabasanshiro/PKGBUILD b/yabasanshiro/PKGBUILD
--- a/yabasanshiro/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/yabasanshiro/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,28 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=yabasanshiro
+pkgver=2.12.0
+pkgrel=1
+pkgdesc='A Sega Saturn emulator. Qt5 port'
+arch=(x86_64)
+url=http://www.uoyabause.org
+license=(GPL)
+depends=(libsm libxkbcommon-x11 libxrandr
+         mini18n openal qt5-multimedia sdl2)
+makedepends=(snapd)
+conflicts=("$pkgname")
+provides=("$pkgname")
+
+prepare() {
+  snap download "$pkgname"
+  unsquashfs -d "$pkgname" "$pkgname"*.snap
+}
+
+package() {
+  cd "$pkgname"/usr
+  install -dm 755 "$pkgdir"/usr/{bin,share/{applications,pixmaps}}
+  install -m 755 bin/"$pkgname" "$pkgdir"/usr/bin/
+  install -m 644 share/applications/"$pkgname".desktop "$pkgdir"/usr/share/applications/
+  install -m 644 share/pixmaps/"$pkgname".png "$pkgdir"/usr/share/pixmaps/
+  perl -pi -e 's|usr.*pixmaps/||g' "$pkgdir"/usr/share/applications/"$pkgname".desktop
+}
